---
title: "Genomic characterization and clinical outcomes of patients with peritoneal metastases from colorectal cancer"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_float: true
editor_options: 
  chunk_output_type: console
---

# Overview 

*This code represents the analyses performed for the manuscript "Genomic characterization and clinical outcomes of patients with peritoneal metastases from colorectal cancer."*

Authors: Samantha Brown, Jessica A. Lavery, Hannah E. Fuchs, Katherine S. Panageas

```{r, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE)

library(gtsummary)
library(genieBPC)
library(gnomeR)
library(survival)
library(survminer)
library(tidyverse)
library(ggconsort)
library(ggsurvfit)
library(viridis)
library(patchwork)
library(cbioportalR)

set_synapse_credentials()
theme_gtsummary_compact()
```

```{r}
## read in CRC v2.0-public clinical data
crc_v2_0 <- pull_data_synapse(cohort = "CRC", version = "v2.0-public")

## save each dataset to global env
pt_char_2_0 <- crc_v2_0$CRC_v2.0$pt_char
ca_dx_index_2_0_pre <- crc_v2_0$CRC_v2.0$ca_dx_index %>% 
  select(-contains(c("abdomen", "pelvis", "lung")))
ca_dx_non_index_2_0 <- crc_v2_0$CRC_v2.0$ca_dx_non_index
ca_drugs_2_0 <- crc_v2_0$CRC_v2.0$ca_drugs
cpt_2_0 <- crc_v2_0$CRC_v2.0$cpt
prissmm_imaging_2_0 <- crc_v2_0$CRC_v2.0$prissmm_imaging
prissmm_md_2_0 <- crc_v2_0$CRC_v2.0$prissmm_md
prissmm_pathology_2_0 <- crc_v2_0$CRC_v2.0$prissmm_pathology
tm_2_0 <- crc_v2_0$CRC_v2.0$tumor_marker
```

```{r}
## read in CRC v2.0-public genomic data from cBioPortal
cbioportalR::set_cbioportal_db("https://genie-private.cbioportal.org/api")

mutations_extended_2_0 <- get_mutations_by_study("crc_public_genie_bpc")
fusions_2_0 <- get_fusions_by_study("crc_public_genie_bpc")
cna_2_0 <- get_cna_by_study("crc_public_genie_bpc")
```

# Change file path

```{r}
# hist_file <- [insert file path to Excel file downloaded from: "https://seer.cancer.gov/icd-o-3/sitetype.icdo3.d20220429.xlsx"]

hist_codebook <- read_csv(hist_file$path) %>%
  janitor::clean_names() %>%
  mutate(temp_hist_behav = str_remove(histology_behavior, "/.*")) %>%
  select(histology_behavior,
         temp_hist_behav,
         histology_behavior_description) %>%
  distinct()
```

# Derive peritoneal and lung metastases

```{r}
ca_mets_dx_1 <- ca_dx_index_2_0_pre %>%
  filter(stage_dx == "Stage IV") %>%
  select(record_id,
         cohort,
         ca_seq,
         ca_dmets_yn,
         starts_with("ca_first_dmets")) %>%
  pivot_longer(
    cols = starts_with("ca_first_dmets"),
    names_to = "mets_number",
    values_to = "site_mets",
    # dont drop NAs, want 0 if pt had no mets
    values_drop_na = FALSE
  ) %>%
  mutate(
    dmets_dx_lung = case_when(
      ca_dmets_yn == "Yes" &
        word(site_mets, 1) %in% c("C34.0", "C34.1", "C34.2", "C34.3", "C34.8", "C34.9") ~ 1,
      TRUE ~ 0
    ),
    dmets_dx_peritoneum = case_when(
      ca_dmets_yn == "Yes" &
        word(site_mets, 1) %in% c("C48.1", "C48.2", "C56.9", "C57.4", "F20") ~ 1,
      TRUE ~ 0
    )
  ) %>%
  rowwise() %>%
  ungroup()

ca_mets_dx <- ca_mets_dx_1 %>%
  group_by(record_id, cohort, ca_seq) %>%
  summarize_at(vars(starts_with("dmets_dx")), max) %>%
  mutate_at(vars(starts_with("dmets_dx")), .funs = list(function(x) {
    case_when(x == 1 ~ "Yes",
              x == 0 ~ "No")
  })) %>%
  ungroup()
```

```{r}
red_cap_site <-
  c(
    "C15.2 Abdominal esophagus",
    "C15.5 Lower third of esophagus",
    "C16.0 Cardia NOS",
    "C16.1 Fundus of stomach",
    "C16.2 Body of stomach",
    "C16.3 Gastric antrum",
    "C16.4 Pylorus",
    "C16.5 Lesser curvature of stomach NOS",
    "C16.6 Greater curvature of stomach NOS",
    "C16.8 Overlapping lesion of stomach",
    "C16.9 Stomach NOS",
    "C17.0 Duodenum",
    "C17.1 Jejunum",
    "C17.2 Ileum",
    "C17.3 Meckel diverticulum",
    "C17.8 Overlapping lesion of small intestine",
    "C17.9 Small intestine NOS",
    "C18.0 Cecum",
    "C18.1 Appendix",
    "C18.2 Ascending colon",
    "C18.3 Hepatic flexure of colon",
    "C18.4 Transverse colon",
    "C18.5 Splenic flexure of colon",
    "C18.6 Descending colon",
    "C18.7 Sigmoid colon",
    "C18.8 Overlapping lesion of colon",
    "C18.9 Colon NOS",
    "C19.9 Rectosigmoid junction",
    "C20.9 Rectum NOS",
    "C21.0 Anus NOS",
    "C21.1 Anal canal",
    "C21.2 Cloacogenic zone",
    "C21.8 Overlapping lesion of rectum anus and anal canal",
    "C22.1 intrahepatic bile duct",
    "C23.9 Gallbladder",
    "C24.0 Extrahepatic bile duct",
    "C24.1 Ampulla of Vater",
    "C24.8 Overlapping lesion of biliary tract",
    "C24.9 Biliary tract NOS",
    "C25.0 Head of pancreas",
    "C25.1 Body of pancreas",
    "C25.2 Tail of pancreas",
    "C25.3 Pancreatic duct",
    "C25.4 Islets of Langerhans",
    "C25.7 Other specified parts of pancreas",
    "C25.8 Overlapping lesion of pancreas",
    "C25.9 Pancreas NOS",
    "C26.0 Intestinal tract NOS",
    "C26.8 Overlapping lesion of digestive system",
    "C26.9 Gastrointestinal tract NOS",
    "C34.0 Main bronchus",
    "C34.1 Upper lobe lung",
    "C34.2 Middle lobe lung",
    "C34.3 Lower lobe lung",
    "C34.8 Overlapping lesion of lung",
    "C34.9 Lung NOS",
    "C42.2 Spleen",
    "C47.4 Peripheral nerves and autonomic nervous system of abdomen",
    "C47.5 Peripheral nerves and autonomic nervous system of pelvis",
    "C47.6 Peripheral nerves and autonomic nervous system of trunk NOS",
    "C48.0 Retroperitoneum",
    "C48.1 Specified parts of peritoneum",
    "C48.2 Peritoneum NOS",
    "C48.8 Overlapping lesion of retroperitoneum and peritoneum",
    "C49.4 Connective Subcutaneous and other soft tissues of abdomen",
    "C49.5 Connective Subcutaneous and other soft tissues of pelvis",
    "C49.6 Connective Subcutaneous and other soft tissues of trunk NOS",
    "C51.0 Labium majus",
    "C51.1 Labium minus",
    "C51.2 Clitoris",
    "C51.8 Overlapping lesion of vulva",
    "C51.9 Vulva NOS",
    "C52.9 Vagina NOS",
    "C53.0 Endocervix",
    "C53.1 Exocervix",
    "C53.8 Overlapping lesion of cervix uteri",
    "C53.9 Cervix uteri",
    "C54.0 Isthmus uteri",
    "C54.1 Endometrium",
    "C54.2 Myometrium",
    "C54.3 Fundus uteri",
    "C54.8 Overlapping lesion of corpus uteri",
    "C54.9 Corpus uteri",
    "C55.9 Uterus NOS",
    "C56.9 Ovary",
    "C57.0 Fallopian tube",
    "C57.1 Broad ligament",
    "C57.2 Round ligament",
    "C57.3 Parametrium",
    "C57.4 Uterine adnexa",
    "C57.7 Other specified parts of female genital organs",
    "C57.8 Overlapping lesion of female genital organs",
    "C57.9 Female genital tract NOS",
    "C58.9 Placenta",
    "C60.0 Prepuce",
    "C60.1 Glans penis",
    "C60.2 Body of penis",
    "C60.8 Overlapping lesion of penis",
    "C60.9 Penis NOS",
    "C61.9 Prostate gland",
    "C62.0 Undescended testis",
    "C62.1 Descended testis",
    "C62.9 Testis NOS",
    "C63.0 Epididymis",
    "C63.1 Spermatic cord",
    "C63.2 Scrotum NOS",
    "C63.7 Other specified parts of male genital organs",
    "C63.8 Overlapping lesion of male genital organs",
    "C63.9 Male genital organs NOS",
    "C64.9 Kidney NOS",
    "C65.9 Renal pelvis",
    "C66.9 Ureter",
    "C67.0 Trigone of bladder",
    "C67.1 Dome of bladder",
    "C67.2 Lateral wall of bladder",
    "C67.3 Anterior wall of bladder",
    "C67.4 Posterior wall of bladder",
    "C67.5 Bladder neck",
    "C67.6 Ureteric orifice",
    "C67.7 Urachus",
    "C67.8 Overlapping lesion of bladder",
    "C67.9 Bladder NOS",
    "C68.0 Urethra",
    "C68.1 Paraurethral gland",
    "C68.8 Overlapping lesion of urinary organs",
    "C68.9 Urinary system NOS",
    "C74.0 Cortex of adrenal gland",
    "C74.1 Medulla of adrenal gland",
    "C74.9 Adrenal gland NOS",
    "C75.5 Aortic body and other paraganglia",
    "C76.2 Abdomen NOS",
    "C76.3 Pelvis NOS",
    "C77.2 Intra-abdominal lymph nodes",
    "C77.5 Pelvic lymph nodes",
    "F20 Peritoneal Fluid/Ascites",
    "F40 Urine"
  )

mets_site_group <- c(
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "pelvis",
  "abdomen",
  "abdomen",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "thorax",
  "thorax",
  "thorax",
  "thorax",
  "thorax",
  "thorax",
  "abdomen",
  "abdomen",
  "pelvis",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "pelvis",
  "abdomen",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "abdomen",
  "abdomen",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "pelvis",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "abdomen",
  "pelvis",
  "abdomen",
  "pelvis",
  "abdomen",
  "pelvis"
)

crc_classification <- c(
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Local/Regional",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Local/Regional",
  "Local/Regional",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Local/Regional",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Distant",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Local/Regional",
  "Distant",
  "Distant"
)

mets_classification1 <- data.frame(red_cap_site,
                                   mets_site_group,
                                   crc_classification) %>%
  mutate(icdo3 = word(red_cap_site, 1),
         icdo3_no_dec = str_remove(icdo3, "\\."))
```

```{r}
## identify ICD-O3 codes for peritoneum mets
define_peritoneum_mets <- mets_classification1 %>%
  filter(
    grepl("specified parts of peritoneum", str_to_lower(red_cap_site)) |
      grepl("peritoneum nos", str_to_lower(red_cap_site)) |
      grepl("uterine adnexa", str_to_lower(red_cap_site)) |
      grepl("ovar", str_to_lower(red_cap_site)) | 
      grepl("peritoneal fluid", str_to_lower(red_cap_site))
  ) %>% 
  mutate(mets_site_group = "peritoneum")

define_peritoneum_mets_vs_abdomen_vs_pelvis <-
  bind_rows(mets_classification1,
            define_peritoneum_mets) %>%
  rename(mets_site_group_pre = mets_site_group) %>%
  filter(mets_site_group_pre %in% c("abdomen", "pelvis", "peritoneum")) %>%
  mutate(mets_site_group_unique = case_when(
    !(icdo3 %in% c("C48.1", "C48.2", "C56.9", "C57.4", "F20")) ~ paste0(mets_site_group_pre, "_without_pm"),
    TRUE ~ "peritoneum"
  )) %>%
  group_by(across(-contains("mets_site_"))) %>%
  summarize(mets_site_group = unique(mets_site_group_unique),
            .groups = "drop")

define_lung_mets <- mets_classification1 %>%
  filter(grepl("C34.0|C34.1|C34.2|C34.3|C34.8|C34.9", red_cap_site)) %>%
  mutate(mets_site_group = "lung")

# append mets_classification dataframe with PM defined
mets_classification <- bind_rows(
  mets_classification1 %>% 
    filter(!(mets_site_group %in% c("abdomen", "pelvis"))),
  define_peritoneum_mets_vs_abdomen_vs_pelvis,
  define_lung_mets
)
```

```{r, include = FALSE}
# check PM and lung mets groupings
distinct(define_peritoneum_mets, red_cap_site, mets_site_group)
distinct(define_lung_mets, red_cap_site, mets_site_group)
```

```{r}
ca_dx_index <- ca_dx_index_2_0_pre %>%
  filter(redcap_ca_index == "Yes") %>% 
  group_by(cohort, record_id) %>% 
  slice(which.min(ca_seq))
```

```{r}
ca_dx_scans_pre <- left_join(ca_dx_index,
  prissmm_imaging_2_0,
  by = c("cohort", "record_id")
) %>%
  # scans post-diagnosis only
  filter(
    image_scan_int > dob_ca_dx_days,
    image_ca == "Yes, the Impression states or implies there is evidence of cancer"
  ) %>%
  select(cohort, record_id, ca_seq, dob_ca_dx_days, stage_dx, scan_number, image_scan_int, starts_with("image_casite")) %>%
  pivot_longer(
    cols = starts_with("image_casite"),
    names_to = "number",
    values_to = "scan_ca_site",
    values_drop_na = TRUE
  ) %>%
  # extract ICDO3 code for scan site with cancer
  mutate(scan_ca_site_icdo3 = str_remove(word(scan_ca_site, 1), "\\."))

# filter on distant or indeterminate mets (ignore local/regional)
ca_dx_scans <- left_join(ca_dx_scans_pre,
  mets_classification,
  by = c("scan_ca_site_icdo3" = "icdo3_no_dec")
) %>%
  mutate(mets_classification = case_when(
    cohort == "CRC" ~ crc_classification
  )) %>%
  # filter to distant only
  mutate(keep_flg = case_when(
    mets_classification == "Distant" ~ 1,
    mets_classification == "Indeterminate" ~ 1,
    TRUE ~ 0
  )) %>%
  filter(keep_flg == 1) %>%
  select(-dob_ca_dx_days)
```

```{r}
ca_dx_scans_mets <- left_join(
  ca_dx_index %>%
    filter(redcap_ca_index == "Yes") %>%
    select(cohort, record_id, ca_seq, stage_dx, dob_ca_dx_days),
  ca_dx_scans %>%
    filter(mets_classification == "Distant"),
  by = c("cohort", "record_id", "ca_seq", "stage_dx")
)
```

```{r}
ca_dx_scans_mets_first_distant <- ca_dx_scans_mets %>%
  group_by(cohort, record_id, ca_seq) %>%
  slice_min(image_scan_int, with_ties = FALSE) %>%
  ungroup()

ca_dx_scans_mets_first_distant_site <- ca_dx_scans_mets %>%
  group_by(cohort, record_id, ca_seq, mets_site_group) %>%
  slice_min(image_scan_int, with_ties = FALSE) %>%
  ungroup() %>%
  mutate(mets_in_group = case_when(
    !is.na(mets_classification) ~ 1,
    TRUE ~ 0
  )) %>%
  pivot_wider(
    id_cols = c(cohort, record_id, ca_seq),
    names_from = mets_site_group,
    names_prefix = "distant_mets_scan_",
    values_from = mets_in_group,
    values_fill = 0
  ) %>%
  janitor::clean_names()

# date of first distant met by site
ca_dx_scans_mets_first_distant_site_dt <- ca_dx_scans_mets %>%
  filter(mets_classification == "Distant") %>%
  # pick the first scan for each site_group
  group_by(cohort, record_id, ca_seq, mets_site_group) %>%
  slice_min(image_scan_int, with_ties = FALSE) %>%
  ungroup() %>%
  mutate(mets_in_group = case_when(
    !is.na(mets_classification) ~ 1,
    TRUE ~ 0
  )) %>%
  pivot_wider(
    id_cols = c(cohort, record_id, ca_seq),
    names_from = mets_site_group,
    names_prefix = "tt_distant_mets_scan_",
    values_from = image_scan_int
  ) %>%
  janitor::clean_names()

ca_dx_scans_mets_distant_redcap_repeat_instance <-
  ca_dx_scans_mets %>%
  group_by(cohort, record_id, ca_seq, mets_site_group) %>%
  summarize(
    scan_numbers = paste0(scan_number, collapse = ", "),
    .groups = "drop"
  ) %>%
  pivot_wider(
    id_cols = c(cohort, record_id, ca_seq),
    names_from = mets_site_group,
    names_prefix = "scan_scan_instance_",
    values_from = scan_numbers
  ) %>%
  janitor::clean_names() %>%
  select(-scan_scan_instance_na)
```

```{r}
path_long_spec_site <- prissmm_pathology_2_0 %>%
  select(
    cohort,
    record_id,
    path_proc_number,
    path_proc_int,
    path_rep_number,
    # path_rep_int,
    starts_with("path_ca_type")
  ) %>%
  pivot_longer(
    cols = c(starts_with("path_ca_type")),
    names_to = "specimen_number",
    names_prefix = "path_ca_type",
    values_to = "specimen_site",
    values_drop_na = FALSE
  ) %>%
  mutate(specimen_site_grouped = case_when(
    specimen_site %in% c("Non Small Cell Lung Cancer", "Lung Cancer, NOS") ~ "NSCLC",
    specimen_site %in% c("Colon Cancer", "Colon/Rectum Cancer", "Rectum and Rectosigmoid Cancer") ~ "CRC",
    specimen_site %in% c("Breast Cancer", "Breast Sarcoma") ~ "BrCa",
    specimen_site %in% c("Pancreatic Cancer") ~ "Pancreas",
    specimen_site %in% c("Bladder Cancer", "Renal Pelvis Cancer") ~ "Bladder",
    specimen_site %in% c("Prostate Cancer") ~ "Prostate"
  ))
```

```{r}
# transpose path_ca yes/no flag of cancer for each specimen on the path report
path_long_ca <- prissmm_pathology_2_0 %>%
  select(
    cohort,
    record_id,
    # redcap_repeat_instance,
    path_proc_number,
    path_proc_int,
    path_rep_number,
    # path_rep_int,
    starts_with("path_ca"),
    -starts_with("path_ca_inv"),
    -starts_with("path_ca_type"),
    -starts_with("path_ca_hist")
  ) %>%
  pivot_longer(
    cols = c(starts_with("path_ca")),
    names_to = "specimen_number",
    names_prefix = "path_ca",
    values_to = "path_ca_flag",
    values_drop_na = FALSE
  )

path_long_icdo3 <- prissmm_pathology_2_0 %>%
  select(
    cohort,
    record_id,
    path_proc_number,
    path_proc_int,
    path_rep_number,
    starts_with("path_site")
  ) %>%
  pivot_longer(
    cols = c(starts_with("path_site")),
    names_to = "specimen_number",
    names_prefix = "path_site",
    values_to = "path_icdo3",
    values_drop_na = FALSE
  ) %>%
  mutate(path_ca_site_icdo3 = str_remove(word(path_icdo3, 1), "\\."))

path_long <- plyr::join_all(
  list(path_long_spec_site,
       path_long_ca,
       path_long_icdo3),
  by = c(
    "cohort",
    "record_id",
    "path_proc_number",
    "path_proc_int",
    "path_rep_number",
    "specimen_number"
  ),
  type = "left"
) %>%
  filter(path_ca_flag == "Yes", !is.na(specimen_site_grouped))
```

```{r}
ca_dx_path_pre <- left_join(
  ca_dx_index %>%
    select(
      cohort,
      record_id,
      dob_ca_dx_days,
      ca_seq,
      stage_dx
    ),
  path_long,
  by = c("cohort", "record_id")
) %>%
  filter(path_proc_int > dob_ca_dx_days)

ca_dx_path <- left_join(ca_dx_path_pre,
  mets_classification,
  by = c("path_ca_site_icdo3" = "icdo3_no_dec")
) %>%
  mutate(mets_classification = case_when(
    cohort == "CRC" ~ crc_classification
  )) %>%
  mutate(keep_flg = case_when(
    mets_classification == "Distant" ~ 1,
    mets_classification == "Indeterminate" ~ 1,
    TRUE ~ 0
  )) %>%
  filter(keep_flg == 1) %>%
  select(-dob_ca_dx_days) %>% 
  ungroup()
```

```{r}
ca_dx_path_mets <- left_join(
  ca_dx_index_2_0_pre %>%
    filter(redcap_ca_index == "Yes") %>%
    select(cohort, record_id, ca_seq, stage_dx),
  ca_dx_path %>%
    filter(mets_classification == "Distant"),
  by = c("cohort", "record_id", "ca_seq", "stage_dx")
)
```

```{r}
ca_dx_path_mets_first_distant <- ca_dx_path_mets %>%
  group_by(cohort, record_id, ca_seq) %>% 
  slice_min(path_proc_int, with_ties = FALSE) %>%
  ungroup()

ca_dx_path_mets_first_distant_site <- ca_dx_path_mets %>%
  filter(mets_classification == "Distant") %>%
  group_by(cohort, record_id, ca_seq, mets_site_group) %>%
  slice_min(path_proc_int, with_ties = FALSE) %>%
  ungroup() %>%
  mutate(mets_in_group = case_when(!is.na(mets_classification) ~ 1,
                                   TRUE ~ 0)) %>%
  pivot_wider(
    id_cols = c(cohort, record_id, ca_seq),
    names_from = mets_site_group,
    names_prefix = "distant_mets_path_",
    values_from = mets_in_group,
    values_fill = 0
  ) %>%
  janitor::clean_names()

ca_dx_path_mets_first_distant_site_dt <- ca_dx_path_mets %>%
  group_by(cohort, record_id, ca_seq, mets_site_group) %>%
  slice_min(path_proc_int, with_ties = FALSE) %>%
  ungroup() %>%
  mutate(mets_in_group = case_when(!is.na(mets_classification) ~ 1,
                                   TRUE ~ 0)) %>%
  pivot_wider(
    id_cols = c(cohort, record_id, ca_seq),
    names_from = mets_site_group,
    names_prefix = "tt_distant_mets_path_",
    values_from = path_proc_int
  ) %>%
  janitor::clean_names()

ca_dx_path_mets_distant_redcap_repeat_instance <- ca_dx_path_mets %>%
  mutate(path_proc_rep = paste0("proc ", path_proc_number, " rep ", path_rep_number)) %>% 
  group_by(cohort, record_id, ca_seq, mets_site_group) %>%
  summarize(
    path_proc_reps = paste0(path_proc_rep, collapse = ", "),
    .groups = "drop"
  ) %>%
  pivot_wider(
    id_cols = c(cohort, record_id, ca_seq),
    names_from = mets_site_group,
    names_prefix = "path_proc_reps_instance_",
    values_from = path_proc_reps
  ) %>%
  janitor::clean_names() %>%
  select(-path_proc_reps_instance_na)
```

```{r}
pts_mets_coh_path <- ca_dx_path_mets %>%
  mutate(
    numeric_mets_distant = case_when(mets_classification == "Distant" ~ 1,
                                     TRUE ~ 0),
    numeric_mets_distant_indet = case_when(
      mets_classification %in% c("Distant", "Indeterminate") ~ 1,
      TRUE ~ 0
    )
  ) %>%
  group_by(cohort, record_id, ca_seq, stage_dx) %>%
  summarize(
    any_distant = max(numeric_mets_distant),
    any_distant_indet = max(numeric_mets_distant_indet),
    .groups = "drop"
  )
```

```{r}
pts_mets_coh_path_distant <- plyr::join_all(
  list(
    pts_mets_coh_path %>%
      filter(any_distant == 1),
    ca_dx_path_mets_first_distant_site,
    ca_dx_path_mets_first_distant_site_dt,
    ca_dx_path_mets_distant_redcap_repeat_instance
  ),
  type = "left",
  by = c("cohort", "record_id", "ca_seq")
)
```

```{r}
ca_mets_dx_1 <- ca_dx_index %>%
  filter(stage_dx == "Stage IV", ca_dmets_yn == "Yes") %>%
  select(record_id,
         cohort,
         ca_seq,
         ca_dmets_yn,
         starts_with("ca_first_dmets")) %>%
  pivot_longer(
    cols = starts_with("ca_first_dmets"),
    names_to = "mets_number",
    values_to = "site_mets",
    values_drop_na = TRUE
  ) %>%
  mutate(ca_dmets_icdo3 = str_remove(word(site_mets, 1), "\\."))

ca_mets_dx_2 <- left_join(ca_mets_dx_1,
  mets_classification %>% select(-contains("red")),
  by = c("ca_dmets_icdo3" = "icdo3_no_dec")
) %>%
  mutate(mets_classification = case_when(
    cohort == "CRC" ~ crc_classification
  )) %>%
  mutate(keep_flg = case_when(
    mets_classification == "Distant" ~ 1,
    mets_classification == "Indeterminate" ~ 1,
    TRUE ~ 0
  )) %>%
  filter(keep_flg == 1)

ca_mets_dx_3 <- left_join(
  ca_dx_index %>%
    filter(stage_dx == "Stage IV", ca_dmets_yn == "Yes") %>%
    select(cohort, record_id, ca_seq),
  ca_mets_dx_2,
  by = c("cohort", "record_id", "ca_seq")
) %>%
  mutate(mets_in_group = case_when(!is.na(mets_classification) ~ 1,
                                   TRUE ~ 0)) %>%
  distinct(cohort, record_id, ca_seq, mets_site_group, .keep_all = TRUE) %>%
  pivot_wider(
    id_cols = c(cohort, record_id, ca_seq),
    names_from = mets_site_group,
    names_prefix = "distant_mets_dx_",
    values_from = mets_in_group,
    values_fill = 0
  ) %>%
  janitor::clean_names()

ca_mets_dx_3_time <- left_join(
  ca_dx_index_2_0_pre %>%
    filter(redcap_ca_index == "Yes") %>%
    filter(stage_dx == "Stage IV", ca_dmets_yn == "Yes") %>%
    select(cohort, record_id, ca_seq, dob_ca_dx_days),
  ca_mets_dx_2,
  by = c("cohort", "record_id", "ca_seq")
) %>%
  mutate(tt_mets = case_when(
    !is.na(mets_classification) ~ dob_ca_dx_days,
    TRUE ~ as.numeric(NA)
  )) %>%
  distinct(cohort, record_id, ca_seq, mets_site_group, .keep_all = TRUE) %>%
  pivot_wider(
    id_cols = c(cohort, record_id, ca_seq),
    names_from = mets_site_group,
    names_prefix = "tt_distant_mets_dx_",
    values_from = tt_mets
  ) %>%
  janitor::clean_names()
```

```{r}
presence_of_each_mets <- left_join(
  ca_dx_index %>%
    select(cohort, record_id, ca_seq, dob_ca_dx_days, stage_dx),
  plyr::join_all(
    list(
      ca_dx_scans_mets_first_distant_site,
      ca_dx_path_mets_first_distant_site,
      ca_mets_dx_3
    ),
    by = c("cohort", "record_id", "ca_seq"),
    type = "full"
  ),
  by = c("cohort", "record_id", "ca_seq")
) %>%
  pivot_longer(
    cols = starts_with("distant_mets_"),
    names_to = "distant_mets",
    names_prefix = "distant_mets_",
    values_to = "values",
    values_drop_na = TRUE
  ) %>%
  mutate(body_part = case_when(
    !grepl("abdomen|pelvis|peritoneum", distant_mets) ~ word(distant_mets, 2, sep = "_"),
    grepl("abdomen|pelvis|peritoneum", distant_mets) ~ gsub("scan_", "", gsub("path_", "", gsub(
      "dx_", "", distant_mets
    )))
  )) %>%
  group_by(
    cohort,
    record_id,
    ca_seq,
    stage_dx,
    dob_ca_dx_days,
    body_part
  ) %>%
  summarize(dmets = max(values, na.rm = TRUE),
            .groups = "drop") %>%
  pivot_wider(names_from = body_part,
              names_prefix = "dmets_",
              values_from = dmets) %>%
  select(-dmets_na)
```

```{r}
time_to_each_mets_from_dx <- left_join(
  ca_dx_index %>%
    select(cohort, record_id, ca_seq, dob_ca_dx_days),
  plyr::join_all(
    list(
      ca_dx_scans_mets_first_distant_site_dt,
      ca_dx_path_mets_first_distant_site_dt,
      ca_mets_dx_3_time
    ),
    by = c("cohort", "record_id", "ca_seq"),
    type = "full"
  ),
  by = c("cohort", "record_id", "ca_seq")
) %>%
  pivot_longer(
    cols = starts_with("tt_distant"),
    names_to = "dob_distant_mets",
    names_prefix = "tt_distant_mets_",
    values_to = "values",
    values_drop_na = TRUE
  ) %>%
  mutate(time_from_dx_days = values - dob_ca_dx_days) %>%
  mutate(body_part = case_when(
    !grepl("abdomen|pelvis|peritoneum", dob_distant_mets) ~ word(dob_distant_mets, 2, sep = "_"),
    grepl("abdomen|pelvis|peritoneum", dob_distant_mets) ~ gsub("scan_", "", gsub(
      "path_", "", gsub("dx_", "", dob_distant_mets)
    ))
  )) %>%
  group_by(cohort, record_id, ca_seq, body_part) %>%
  summarize(
    tt_distant_met_days = min(time_from_dx_days, na.rm = TRUE),
    tt_distant_met_mos = tt_distant_met_days / 30.4,
    tt_distant_met_yrs = tt_distant_met_days / 365.25,
    .groups = "drop"
  ) %>%
  pivot_wider(
    names_from = body_part,
    names_prefix = "dx_dmets_",
    values_from = c(tt_distant_met_days, tt_distant_met_mos, tt_distant_met_yrs)
  ) %>%
  rename_with( ~ paste0(
    "dx_to_dmets_",
    str_remove(., "tt_distant_met_days_dx_dmets_"),
    "_days"
  ),
  .cols = starts_with("tt_distant_met_days_")) %>%
  rename_with( ~ paste0(
    "dx_to_dmets_",
    str_remove(., "tt_distant_met_mos_dx_dmets_"),
    "_mos"
  ),
  .cols = starts_with("tt_distant_met_mos_")) %>%
  rename_with( ~ paste0(
    "dx_to_dmets_",
    str_remove(., "tt_distant_met_yrs_dx_dmets_"),
    "_yrs"
  ),
  .cols = starts_with("tt_distant_met_yrs_"))
```

```{r}
site_first_mets <- bind_rows(
  ca_dx_scans_mets %>%
    filter(mets_classification == "Distant") %>%
    rename(time_to_distant_mets = image_scan_int),
  ca_dx_path_mets %>%
    filter(mets_classification == "Distant") %>%
    rename(time_to_distant_mets = path_proc_int)
) %>% 
  group_by(cohort, record_id, ca_seq) %>%
  slice_min(order_by = time_to_distant_mets, with_ties = TRUE) %>%
  distinct(cohort,
           record_id,
           ca_seq,
           time_to_distant_mets,
           mets_site_group) %>%
  arrange(cohort,
          record_id,
          ca_seq,
          time_to_distant_mets,
          mets_site_group) %>%
  summarize(first_distant_mets = paste0(mets_site_group, collapse = "; "),
            .groups = "drop")
```

```{r}
mets_summary <- plyr::join_all(
  list(
    presence_of_each_mets,
    time_to_each_mets_from_dx,
    ca_dx_path_mets_distant_redcap_repeat_instance,
    ca_dx_scans_mets_distant_redcap_repeat_instance
  ),
  by = c("cohort", "record_id", "ca_seq"),
  type = "left"
) %>%
  rowwise() %>%
  select(
    cohort,
    record_id,
    ca_seq,
    dob_ca_dx_days,
    stage_dx,
    contains("dx_to_dmets_"),
    starts_with("dob_to_dmets"),
    contains("abdomen"),
    contains("pelvis"),
    contains("peritoneum"),
    contains("lung")
  ) 
```

## End derivation of PM and lung mets

```{r}
ca_dx_index_2_0 <- ca_dx_index_2_0_pre %>%
  left_join(
    .,
    mets_summary %>%
      select(
        cohort,
        record_id,
        ca_seq,
        contains("abdomen"),
        contains("pelvis"),
        contains("peritoneum"),
        contains("lung")
      ),
    by = c("cohort", "record_id", "ca_seq")
  ) %>% 
  left_join(
    ., ca_mets_dx, by = c("cohort", "record_id", "ca_seq")
  )
```

```{r, include = FALSE}
## define mCRC cohort
crc_metastatic_pre <- ca_dx_index_2_0 %>% 
  group_by(record_id) %>%
  slice(which.min(ca_seq)) %>%
  ungroup() %>%
  filter(pfs_cohort %in% c("Stage IV", "Stage I-III with Distant Mets"))
```

```{r}
## define PM in stage IV pts
crc_metastatic_pre_stg_iv1 <- crc_metastatic_pre %>%
  filter(stage_dx == "Stage IV") %>%
  mutate(
    any_peritoneal_met = case_when(
      dx_to_dmets_peritoneum_days == 0 ~ "PM",
      dx_to_dmets_peritoneum_days != 0 ~ "No PM",
      is.na(dx_to_dmets_peritoneum_days) ~ "No PM"
    )
  )

crc_metastatic_pre_stg_iv <- crc_metastatic_pre_stg_iv1 %>%
  left_join(
    .,
    crc_metastatic_pre %>%
      filter(stage_dx == "Stage IV") %>%
      select(
        cohort,
        record_id,
        stage_dx,
        starts_with("ca_first_dmets"),
        starts_with("dx_to_dmets")
      ) %>%
      select(-ends_with("mos"),-ends_with("yrs")) %>%
      pivot_longer(
        cols = starts_with("dx_to_dmets"),
        names_to = "dmets",
        names_prefix = "dmets_to_dx_",
        values_to = "days_from_dx",
        values_drop_na = TRUE
      ) %>%
      filter(round(days_from_dx, 0) == 0) %>%
      mutate(
        dmets_short = str_replace_all(dmets, "dx_to_dmets_", ""),
        dmets_short = str_replace_all(dmets_short, "_days", "")
      ) %>%
      group_by(cohort, record_id, stage_dx, days_from_dx) %>%
      summarize(
        dmets_location = paste(dmets_short, collapse = ", "),
        .groups = "drop"
      ) %>%
      mutate(
        pm_and_other_mets_pre = case_when(
          dmets_location == "peritoneum" ~ "PM only",
          dmets_location != "peritoneum" &
            grepl("peritoneum", dmets_location) ~ "PM and others"
        )
      ) %>%
      select(cohort, record_id, pm_and_other_mets_pre),
    by = c("cohort", "record_id")
  ) %>%
  mutate(pm_and_other_mets = case_when(
    !is.na(pm_and_other_mets_pre) ~ pm_and_other_mets_pre,
    is.na(pm_and_other_mets_pre) ~ any_peritoneal_met
  ))
```

```{r}
## define PM in stage I-III pts who become metastatic
crc_metastatic_pre_stg_i_iii <- crc_metastatic_pre %>%
  filter(pfs_cohort == "Stage I-III with Distant Mets") %>%
  left_join(
    .,
    crc_metastatic_pre %>%
      filter(pfs_cohort == "Stage I-III with Distant Mets") %>%
      select(record_id, starts_with("dx_to_dmets")) %>%
      select(-ends_with("mos"),-ends_with("yrs")) %>%
      pivot_longer(
        cols = c(starts_with("dx_to_dmets"),-dx_to_dmets_days),
        names_to = "dmets",
        names_prefix = "dmets_to_dx_",
        values_to = "days_from_dx",
        values_drop_na = TRUE
      ) %>%
      mutate(
        dmets_short = str_replace_all(dmets, "dx_to_dmets_", ""),
        dmets_short = str_replace_all(dmets_short, "_days", ""),
        days_from_dx = round(days_from_dx, 0)
      ) %>%
      filter(days_from_dx == dx_to_dmets_days) %>%
      group_by(record_id, dx_to_dmets_days) %>%
      summarize(
        dmets_location = paste(dmets_short, collapse = ", "),
        .groups = "drop"
      ) %>%
      mutate(
        pm_and_other_mets_pre = case_when(
          dmets_location == "peritoneum" ~ "PM only",
          dmets_location != "peritoneum" &
            grepl("peritoneum", dmets_location) ~ "PM and others"
        )
      ) %>%
      select(record_id, pm_and_other_mets_pre),
    by = "record_id"
  ) %>%
  mutate(
    pm_and_other_mets = case_when(
      !is.na(pm_and_other_mets_pre) ~ pm_and_other_mets_pre,
      TRUE ~ "No PM"
    ),
    any_peritoneal_met = case_when(
      !is.na(pm_and_other_mets_pre) ~ "PM",
      is.na(pm_and_other_mets_pre) ~ "No PM"
    )
  )
```

```{r}
dmets_at_dx_stg_iv_df <- crc_metastatic_pre %>%
  filter(stage_dx == "Stage IV") %>%
  select(cohort, record_id, stage_dx, starts_with("dx_to_dmets")) %>%
  select(-ends_with("mos"), -ends_with("yrs"), -dx_to_dmets_days) %>%
  pivot_longer(
    cols = starts_with("dx_to_dmets"),
    names_to = "dmets",
    names_prefix = "dmets_to_dx_",
    values_to = "days_from_dx"
  ) %>%
  mutate(
    dmets_short = str_replace_all(dmets, "dx_to_dmets_", ""),
    dmets_short = str_replace_all(dmets_short, "_days", ""),
    days_from_dx = round(days_from_dx, 0)
  ) %>%
  filter(days_from_dx == 0) %>%
  select(-dmets, -days_from_dx) %>% 
  mutate(value = 1) %>% 
  pivot_wider(names_from = dmets_short,
              values_from = value) %>% 
  mutate(across(-c("cohort", "record_id", "stage_dx"), replace_na, 0)) %>% 
  rename_at(vars(-c("cohort", "record_id", "stage_dx")), function(x) paste0("any_", x, "_mets_at_dx")) %>% 
  select(record_id, starts_with("any_"))
```

```{r}
dmets_at_met_stg_i_iii_df <- crc_metastatic_pre %>%
  filter(pfs_cohort == "Stage I-III with Distant Mets") %>% 
  select(cohort, record_id, stage_dx, starts_with("dx_to_dmets")) %>%
  select(-ends_with("mos"), -ends_with("yrs")) %>%
  pivot_longer(
    cols = c(starts_with("dx_to_dmets"), -dx_to_dmets_days),
    names_to = "dmets",
    names_prefix = "dmets_to_dx_",
    values_to = "days_from_dx"
  ) %>%
  mutate(
    dmets_short = str_replace_all(dmets, "dx_to_dmets_", ""),
    dmets_short = str_replace_all(dmets_short, "_days", ""),
    days_from_dx = round(days_from_dx, 0)
  ) %>%
  ## filter on first site of distant mets
  filter(days_from_dx == dx_to_dmets_days) %>% 
  select(-dmets, -days_from_dx, -dx_to_dmets_days) %>% 
  mutate(value = 1) %>% 
  pivot_wider(names_from = dmets_short,
              values_from = value) %>% 
  mutate(across(-c("cohort", "record_id", "stage_dx"), replace_na, 0)) %>% 
  rename_at(vars(-c("cohort", "record_id", "stage_dx")), function(x) paste0("any_", x, "_mets_at_dx")) %>% 
  select(record_id, starts_with("any_"))
```

```{r, include = FALSE}
## derive MMR status
mmr_status_df <- prissmm_pathology_2_0 %>%
  filter(mmr_testing == "Yes") %>% 
  select(record_id, mmr_testing, starts_with("mmr_result")) %>%
  pivot_longer(
    cols = c(starts_with("mmr_result")),
    names_to = "result_no",
    values_to = "mmr_result_long",
    values_drop_na = TRUE
  ) %>%
  select(-result_no) %>%
  filter(mmr_result_long != "") %>%
  mutate(
    mmr_d = case_when(
      mmr_result_long == "Loss of nuclear expression of one or more MMR proteins; deficient mismatch repair" ~ 1,
      TRUE ~ 0
    ),
    mmr_p = case_when(
      mmr_result_long == "No loss of nuclear expression of MMR proteins: No evidence of deficient mismatch repair (low probability of MSI-H): proficient" ~ 1,
      TRUE ~ 0
    ),
    mmr_indet = case_when(mmr_result_long == "Indeterminate/Not stated" ~ 1,
                          TRUE ~ 0)
  ) %>%
  group_by(record_id, mmr_testing) %>%
  summarize(
    any_mmr_d = max(mmr_d),
    any_mmr_p = max(mmr_p),
    any_mmr_indet = max(mmr_indet),
    .groups = "drop"
  ) 

## derive MSI status
msi_status_df <- prissmm_pathology_2_0 %>%
  filter(msi_testing == "Yes") %>% 
  select(record_id, msi_testing, starts_with("msi_result")) %>%
  pivot_longer(
    cols = c(starts_with("msi_result")),
    names_to = "result_no",
    values_to = "msi_result_long",
    values_drop_na = TRUE
  ) %>%
  select(-result_no) %>%
  mutate(
    msi_h = case_when(grepl("MSI-H", msi_result_long) ~ 1,
                      TRUE ~ 0),
    msi_l = case_when(grepl("MSI-L", msi_result_long) ~ 1,
                      TRUE ~ 0),
    mss = case_when(grepl("MSS", msi_result_long) ~ 1,
                    TRUE ~ 0)
  ) %>%
  group_by(record_id, msi_testing) %>%
  summarize(
    any_msi_h = max(msi_h),
    any_msi_l = max(msi_l),
    any_mss = max(mss),
    .groups = "drop"
  )
```

```{r, include = FALSE}
## classify histologies
hist_tbl <- crc_metastatic_pre %>%
  select(ca_hist_adeno_squamous, naaccr_histology_cd) %>%
  unique() %>%
  mutate(naaccr_histology_cd = as.character(naaccr_histology_cd)) %>%
  left_join(
    .,
    hist_codebook %>% select(temp_hist_behav, histology_behavior_description),
    by = c("naaccr_histology_cd" = "temp_hist_behav")
  ) %>%
  mutate(
    histology_derived = case_when(
      histology_behavior_description %in% c("Mucinous adenocarcinoma", "Signet ring cell carcinoma", "Adenocarcinoma") ~ histology_behavior_description,
      ca_hist_adeno_squamous == "Adenocarcinoma" ~ ca_hist_adeno_squamous,
      !is.na(ca_hist_adeno_squamous) & !(histology_behavior_description %in% c("Mucinous adenocarcinoma", "Signet ring cell carcinoma")) & (ca_hist_adeno_squamous != "Adenocarcinoma") ~ "Other",
      is.na(ca_hist_adeno_squamous) ~ as.character(NA)
    )
  ) %>% 
  select(naaccr_histology_cd,
         ca_hist_adeno_squamous,
         histology_derived) %>%
  unique()
```

```{r, include = FALSE}
sample_panels1 <- crc_metastatic_pre %>%
  select(record_id, ca_seq) %>%
  left_join(.,
            cpt_2_0,
            by = c("record_id", "ca_seq")) %>%
  select(cpt_seq_assay_id, cpt_genie_sample_id) %>%
  rename(panel_id = cpt_seq_assay_id,
         sample_id = cpt_genie_sample_id)

## use CRC v2.0-public data
gen_dat_tbl1 <-
  create_gene_binary(
    mutation = mutations_extended_2_0,
    fusion = fusions_2_0,
    cna = cna_2_0,
    samples = sample_panels1$sample_id,
    specify_panel = sample_panels1
  )

gen_dat_kras_braf_nras <- gen_dat_tbl1 %>% 
  rename(cpt_genie_sample_id = sample_id) %>% 
  left_join(., crc_metastatic_pre %>% 
    select(record_id, ca_seq) %>% 
    left_join(., 
              cpt_2_0,
              by = c("record_id", "ca_seq")) %>% 
    select(record_id, cpt_genie_sample_id, dx_cpt_rep_days), 
    by = c("cpt_genie_sample_id")) %>% 
  ## slice on first CPT report
  group_by(record_id) %>% 
  slice(which.min(dx_cpt_rep_days)) %>% 
  ungroup() %>% 
  group_by(record_id) %>% 
  summarize(first_cpt_kras = max(KRAS),
            first_cpt_braf = max(BRAF),
            first_cpt_nras = max(NRAS),
            first_cpt_tp53 = max(TP53),
            first_cpt_apc = max(APC),
            first_cpt_pik3ca = max(PIK3CA),
            .groups = "drop") 

gen_dat_kras_braf_nras_any_cpt <- gen_dat_tbl1 %>% 
  rename(cpt_genie_sample_id = sample_id) %>% 
  left_join(., crc_metastatic_pre %>% 
    select(record_id, ca_seq) %>% 
    left_join(., 
              cpt_2_0,
              by = c("record_id", "ca_seq")) %>% 
    select(record_id, cpt_genie_sample_id, dx_cpt_rep_days), 
    by = c("cpt_genie_sample_id")) %>% 
  group_by(record_id) %>% 
  summarize(any_cpt_kras = max(KRAS),
            any_cpt_braf = max(BRAF),
            any_cpt_nras = max(NRAS),
            any_cpt_tp53 = max(TP53),
            any_cpt_apc = max(APC),
            any_cpt_pik3ca = max(PIK3CA),
            .groups = "drop") 
```

```{r, include = FALSE}
tbl1_cohort <- bind_rows(crc_metastatic_pre_stg_iv,
                         crc_metastatic_pre_stg_i_iii) %>%
  mutate(naaccr_histology_cd = as.character(naaccr_histology_cd)) %>%
  left_join(., pt_char_2_0, by = c("cohort", "record_id")) %>%
  left_join(.,
            bind_rows(dmets_at_dx_stg_iv_df, dmets_at_met_stg_i_iii_df),
            by = "record_id") %>%
  left_join(., mmr_status_df, by = "record_id") %>%
  left_join(., msi_status_df, by = "record_id") %>%
  left_join(.,
            hist_tbl,
            by = c("naaccr_histology_cd", "ca_hist_adeno_squamous")) %>%
  left_join(., gen_dat_kras_braf_nras, by = "record_id") %>%
  left_join(., gen_dat_kras_braf_nras_any_cpt, by = "record_id") %>%
  mutate(
    crc_type_derived = case_when(
      crc_type %in% c("Right colon", "Left colon", "Rectal") ~ crc_type,
      crc_type == "Other" ~ as.character(NA),
      TRUE ~ as.character(NA)
    ),
    crc_side_derived = case_when(
      crc_type %in% c("Left colon", "Rectal") ~ "Left",
      crc_type == "Right colon" ~ "Right",
      crc_type == "Other" ~ as.character(NA),
      TRUE ~ as.character(NA)
    ),
    mmr_testing_derived = case_when(mmr_testing == "Yes" ~ "Yes",
                                    is.na(mmr_testing) ~ "No"),
    mmr_status_derived = case_when(
      is.na(mmr_testing) ~ as.character(NA),
      mmr_testing == "Yes" &
        any_mmr_d == 1 & any_mmr_p == 0 ~ "dMMR",
      mmr_testing == "Yes" &
        any_mmr_d == 0 & any_mmr_p == 1 ~ "pMMR",
      mmr_testing == "Yes" &
        any_mmr_d == 1 & any_mmr_p == 1 ~ "Non-concordant",
      mmr_testing == "Yes" &
        any_mmr_d == 0 &
        any_mmr_p == 0 & any_mmr_indet == 1 ~ as.character(NA)
    ),
    msi_testing_derived = case_when(msi_testing == "Yes" ~ "Yes",
                                    is.na(msi_testing) ~ "No"),
    msi_status_derived = case_when(
      is.na(msi_testing) ~ as.character(NA),
      msi_testing == "Yes" &
        any_msi_h == 1 & any_msi_l == 0 & any_mss == 0 ~ "MSI-H",
      msi_testing == "Yes" &
        any_msi_h == 0 & (any_msi_l == 1  | any_mss == 1) ~ "MSI-L/MSS",
      msi_testing == "Yes" &
        any_msi_h == 1 & (any_msi_l == 1 | any_mss == 1) ~ "Non-concordant"
    ),
    hist_grade_derived = case_when(
      ca_grade %in% c("I", "II") ~ "I/II",
      ca_grade %in% c("III", "IV") ~ "III/IV",
      TRUE ~ as.character(NA)
    ),
    first_cpt_kras_braf_nras = case_when(
      first_cpt_kras == 1 |
        first_cpt_braf == 1 |
        first_cpt_nras == 1 ~ 1,
      TRUE ~ 0
    ),
    any_cpt_kras_braf_nras = case_when(any_cpt_kras == 1 |
                                         any_cpt_braf == 1 |
                                         any_cpt_nras == 1 ~ 1,
                                       TRUE ~ 0)
  ) %>%
  left_join(., 
    cpt_2_0 %>%
      group_by(record_id, ca_seq) %>%
      ## filter on first cancer panel test
      slice(which.min(dx_cpt_rep_mos)) %>%
      ungroup(),
    by = c("record_id", "ca_seq")
  ) %>%
  group_by(record_id) %>%
  slice(which.min(ca_seq)) %>%
  ungroup() %>%
  mutate(cpt_post_os = case_when(
    os_dx_status == 1 & cpt_report_post_death == 1 ~ 1,
    os_dx_status == 0 & cpt_report_post_last_alive == 1 ~ 1,
    TRUE ~ 0)) %>% 
  filter(!is.na(pm_and_other_mets),
         ## filter out stg IV pt without mets at dx
         record_id != "GENIE-VICC-311749")
```

```{r, include = FALSE}
concurrent_dx_record_id <-
  bind_rows(ca_dx_index_2_0, 
            ca_dx_non_index_2_0) %>% 
  group_by(record_id) %>% 
  arrange(record_id, (dob_ca_dx_days)) %>% 
  ## merge on prev dx date
  mutate(prev_dob_ca_dx_days = lag(dob_ca_dx_days)) %>% 
  ungroup() %>% 
  ## define concurrent dx as dx within 15 days
  mutate(concurrent_dx = case_when(dob_ca_dx_days - prev_dob_ca_dx_days <= 15 ~ 1,
                                   TRUE ~ 0)) %>% 
  filter(redcap_ca_index == "Yes", 
         concurrent_dx == 1) %>% 
  select(record_id) %>% 
  unlist()

concurrent_dx <- 
  bind_rows(ca_dx_index_2_0,
            ca_dx_non_index_2_0) %>% 
  filter(record_id %in% c(concurrent_dx_record_id)) %>% 
  select(record_id, redcap_ca_index, stage_dx, dob_ca_dx_days) %>%
  group_by(record_id) %>% 
  summarize(stage_dx_concurrent = paste0(stage_dx, collapse = ", "),
            dob_ca_dx_days_concurrent = paste0(dob_ca_dx_days, collapse = ", "),
            .groups = "drop") %>% 
  filter(grepl("Stage IV", stage_dx_concurrent))
```

```{r}
## define regimen-line
regimen_order_df <- bind_rows(
  left_join(
    tbl1_cohort %>% filter(pfs_cohort == "Stage IV"),
    ca_drugs_2_0 %>%
      select(
        cohort,
        record_id,
        ca_seq,
        regimen_number,
        regimen_drugs,
        dx_reg_start_int,
        dx_reg_start_int_mos,
        os_g_status,
        tt_os_g_mos,
        pfs_i_g_status,
        tt_pfs_i_g_mos,
        pfs_m_g_status,
        tt_pfs_m_g_mos,
        pfs_i_or_m_g_status,
        tt_pfs_i_or_m_g_mos,
        pfs_i_and_m_g_status,
        tt_pfs_i_and_m_g_mos
      ),
    by = c("record_id", "ca_seq")
  ) %>%
    filter(dx_reg_start_int_mos >= 0),
  ## for stg i-iii pts with distant mets, filter on regimens that began after pt became metastatic
  left_join(
    tbl1_cohort %>% filter(pfs_cohort == "Stage I-III with Distant Mets"),
    ca_drugs_2_0 %>%
      select(
        cohort,
        record_id,
        ca_seq,
        regimen_number,
        regimen_drugs,
        dx_reg_start_int_mos,
        pfs_i_g_status,
        tt_pfs_i_g_mos,
        pfs_m_g_status,
        tt_pfs_m_g_mos,
        pfs_i_or_m_g_status,
        tt_pfs_i_or_m_g_mos,
        pfs_i_and_m_g_status,
        tt_pfs_i_and_m_g_mos
      ),
    by = c("record_id", "ca_seq")
  ) %>%
    ## filter on regimens started on/after first met
    filter(dx_reg_start_int_mos >= dx_to_dmets_mos)
) %>%
  group_by(record_id, ca_seq) %>%
  arrange(record_id, ca_seq, regimen_number) %>%
  mutate(regimen_line_derived = 1:n()) %>%
  ungroup() %>% 
  left_join(., genieBPC::regimen_abbreviations, by = "regimen_drugs") %>%
  mutate(
    regimen_drugs_w_abbrev_pre = case_when(
      !is.na(abbreviation) ~ paste0(abbreviation, " (", regimen_drugs, ")"),
      is.na(abbreviation) ~ regimen_drugs
    ),
    ## Replace any "-" with a space to prep for concatenating
    regimen_drugs_w_abbrev = str_replace(regimen_drugs_w_abbrev_pre, "-", " "),
    regimen_group_enrique = case_when(
      regimen_drugs %in% c(
        "Capecitabine", 
        "Fluorouracil", 
        "Fluorouracil, Leucovorin Calcium"
        ) ~ "5FU/Capecitabine",
      regimen_drugs %in% c(
        "Bevacizumab, Capecitabine",
        "Bevacizumab, Fluorouracil, Leucovorin Calcium"
      ) ~ "5FU/Capecitabine + Bevacizumab",
      regimen_drugs == "Capecitabine, Oxaliplatin" ~ "FOLFOX/XELOX",
      regimen_drugs %in% c(
        "Bevacizumab, Capecitabine, Oxaliplatin",
        "Bevacizumab, Capecitabine, Fluorouracil, Leucovorin Calcium, Oxaliplatin"
        ) ~ "FOLFOX/XELOX + Bevacizumab",
      regimen_drugs %in% c(
        "Fluorouracil, Leucovorin Calcium, Oxaliplatin, Panitumumab",
        "Cetuximab, Fluorouracil, Leucovorin Calcium, Oxaliplatin"
      ) ~ "FOLFOX + antiEGFR",
      regimen_drugs %in% c(
        "Pembrolizumab"
      ) ~ "Anti-PD(L1)",
      regimen_drugs %in% c(
        "Cetuximab"
      ) ~ "Anti-EGFR",
      regimen_drugs %in% c(
        "Cetuximab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium",
        "Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium, Panitumumab"
      ) ~ "FOLFIRI + anti-EGFR",
      ## FOLFOX + Bev
      regimen_drugs == "Bevacizumab, Fluorouracil, Leucovorin Calcium, Oxaliplatin" ~  "FOLFOX + Bevacizumab",
      ## FOLFIRI + Bev
      regimen_drugs == "Bevacizumab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium" ~  "FOLFIRI + Bevacizumab",
      ## FOLFOXIRI + Bev
      regimen_drugs == "Bevacizumab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium, Oxaliplatin" ~  "FOLFOXIRI + Bevacizumab",
      !is.na(abbreviation) ~ abbreviation,
      TRUE ~ regimen_drugs
    )
  )
```

```{r}
## identify patients with no first-line regimen
consort_no_approved_first_line_regimen <- tbl1_cohort %>%
  group_by(record_id) %>%
  slice(which.min(ca_seq)) %>%
  ungroup() %>%
  filter((cpt_report_post_death == 0 |
            is.na(cpt_report_post_death)),
         (cpt_report_post_last_alive == 0 |
             is.na(cpt_report_post_last_alive))) %>%
  filter(!is.na(any_peritoneal_met)) %>%
  left_join(
    .,
    regimen_order_df %>%
      select(record_id,
             ca_seq,
             regimen_line_derived,
             regimen_drugs) %>%
      filter(
        regimen_line_derived == 1
      ),
    by = c("record_id", "ca_seq")
  ) %>%
  mutate(no_approved_1l_reg = case_when(
    regimen_line_derived == 1 & 
        regimen_drugs %in% c(
          ## FOLFOX
          "Fluorouracil, Leucovorin Calcium, Oxaliplatin",
          ## FOLFIRI
          "Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium",
          ## FOLFOXIRI
          "Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium, Oxaliplatin",
          ## FOLFOX + Bev
          "Bevacizumab, Fluorouracil, Leucovorin Calcium, Oxaliplatin",
          ## FOLFIRI + Bev
          "Bevacizumab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium",
          ## FOLFOXIRI + Bev
          "Bevacizumab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium, Oxaliplatin",
          ## FOLFOX + Cetuximab
          "Cetuximab, Fluorouracil, Leucovorin Calcium, Oxaliplatin",
          ## FOLFIRI + Cetuximab
          "Cetuximab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium",
          ## FOLFOXIRI + Cetuximab
          "Bevacizumab, Capecitabine, Oxaliplatin",
          "Bevacizumab, Capecitabine, Fluorouracil, Leucovorin Calcium, Oxaliplatin",
          "Capecitabine, Oxaliplatin",
          "Capecitabine, Fluorouracil, Leucovorin Calcium, Oxaliplatin",
          "Cetuximab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium, Oxaliplatin",
          "Cetuximab, Fluorouracil, Leucovorin Calcium, Oxaliplatin",
          "Cetuximab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium",
          "Fluorouracil, Irinotecan Hydrochloride",
          "Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium, Panitumumab",
          "Fluorouracil, Leucovorin Calcium, Oxaliplatin, Panitumumab"
        ) ~ 1,
    TRUE ~ as.numeric(NA)
  )) %>% 
  filter(is.na(no_approved_1l_reg)) %>%
  select(record_id) %>%
  mutate(look_regimen_line = "No 1L combination/approved therapy")

consort_cpt_after_pfs <- tbl1_cohort %>%
  group_by(record_id) %>%
  slice(which.min(ca_seq)) %>%
  ungroup() %>%
  filter((cpt_report_post_death == 0 |
            is.na(cpt_report_post_death)),
         (cpt_report_post_last_alive == 0 |
             is.na(cpt_report_post_last_alive))) %>%
  filter(!is.na(any_peritoneal_met)) %>%
  left_join(
    .,
    regimen_order_df %>%
      select(record_id,
             ca_seq,
             regimen_line_derived,
             regimen_drugs,
             dx_reg_start_int_mos,
             tt_pfs_i_and_m_g_mos) %>%
      filter(regimen_line_derived == 1),
    by = c("record_id", "ca_seq")
  ) %>%
  mutate(
    ## calculate time from regimen start to CPT report
    reg_start_cpt_rep_mos_pre = dx_cpt_rep_mos - dx_reg_start_int_mos
  ) %>% 
  select(record_id, reg_start_cpt_rep_mos_pre, tt_pfs_i_and_m_g_mos) %>% 
  filter(reg_start_cpt_rep_mos_pre >= tt_pfs_i_and_m_g_mos) %>% 
  select(record_id) %>% 
  mutate(cpt_after_pfs_i_and_m_g = 1)
```

```{r}
tbl1_cohort <- tbl1_cohort %>% 
  labelled::set_variable_labels(
    institution.x = "Institution",
      pfs_cohort = "Metastatic cohort",
      age_dx = "Age at diagnosis",
      naaccr_sex_code = "Sex",
      crc_type_derived = "Tumor location",
      crc_side_derived = "Tumor sidedness",
      any_lung_mets_at_dx = "Lung metastases at diagnosis",
      any_liver_mets_at_dx = "Liver metastases at advanced diagnosis",
      any_brain_mets_at_dx = "Brain metastases at advanced diagnosis",
      any_bone_mets_at_dx = "Bone metastases at advanced diagnosis",
      hist_grade_derived = "Histological grade",
      histology_derived = "Histology",
      mmr_testing_derived = "MMR testing",
      mmr_status_derived = "MMR status",
      msi_testing_derived = "MSI testing",
      msi_status_derived = "MSI status",
      first_cpt_kras = "KRAS mutation (first CPT)",
      first_cpt_braf = "BRAF mutation (first CPT)",
      first_cpt_nras = "NRAS mutation (first CPT)",
      first_cpt_kras_braf_nras = "Any KRAS/BRAF/NRAS mutation (first CPT)",
      first_cpt_tp53 = "TP53 mutation (first CPT)",
      first_cpt_apc = "APC mutation (first CPT)",
      first_cpt_pik3ca = "PIK3CA mutation (first CPT)",
      any_cpt_kras = "KRAS mutation (any CPT)",
      any_cpt_braf = "BRAF mutation (any CPT)",
      any_cpt_nras = "NRAS mutation (any CPT)",
      any_cpt_kras_braf_nras = "Any KRAS/BRAF/NRAS mutation (any CPT)",
      any_cpt_tp53 = "TP53 mutation (any CPT)",
      any_cpt_apc = "APC mutation (any CPT)",
      any_cpt_pik3ca = "PIK3CA mutation (any CPT)",
      cpt_post_os = "CPT after death/last follow-up"
  )
```

# Consort diagram

```{r}
all_crc_pts <- 
  ca_dx_index_2_0 %>% 
  left_join(., 
            pt_char_2_0,
            by = c("cohort", "record_id")) %>% 
  left_join(., 
            cpt_2_0 %>% 
              group_by(cohort, record_id, ca_seq) %>%
              slice(which.min(dx_cpt_rep_mos)) %>% 
              ungroup(),
            by = c("cohort", "record_id", "ca_seq")) %>% 
  group_by(cohort, record_id) %>% 
  slice(which.min(ca_seq)) %>% 
  ungroup() %>% 
  left_join(., tbl1_cohort %>% select(record_id, pm_and_other_mets), by = "record_id") %>% 
  mutate(adv_disease_cpt_mos_pre = case_when(pfs_cohort == "Stage IV" ~ dx_cpt_rep_mos,
                                             pfs_cohort == "Stage I-III with Distant Mets" ~ dx_cpt_rep_mos - dx_to_dmets_mos),
         adv_disease_cpt_mos = case_when(adv_disease_cpt_mos_pre < 0 ~ 0,
                                         adv_disease_cpt_mos_pre >= 0 ~ adv_disease_cpt_mos_pre)) %>% 
  select(-adv_disease_cpt_mos_pre) %>% 
  left_join(., consort_no_approved_first_line_regimen, by = "record_id") %>% 
  left_join(., consort_cpt_after_pfs, by = "record_id")

all_crc_pts %>% 
  cohort_start("Overall population") %>% 
  cohort_define(
    metastatic_pts = .full %>% filter(record_id %in% c(tbl1_cohort %>% select(record_id) %>% unlist())),
    metastatic_pts_pm_only = metastatic_pts %>% filter(pm_and_other_mets == "PM only"),
    metastatic_pts_pm_and_others = metastatic_pts %>% filter(pm_and_other_mets == "PM and others"),
    metastatic_pts_no_pm = metastatic_pts %>% filter(pm_and_other_mets == "No PM" | is.na(pm_and_other_mets)),
    excluded_non_met_stg_i_iii = .full %>% filter((stage_dx_iv == "Stage I-III" & is.na(pfs_cohort)) | record_id == "GENIE-VICC-311749"),
    excluded_os_analysis = metastatic_pts %>% filter(tt_os_adv_mos < adv_disease_cpt_mos),
    excluded_os_analysis_cpt = metastatic_pts %>% filter(tt_os_adv_mos < adv_disease_cpt_mos),
    os_analysis_pts = anti_join(metastatic_pts, excluded_os_analysis, by = "record_id"),
    os_analysis_pm_only = os_analysis_pts %>%
      filter(pm_and_other_mets == "PM only"),
    os_analysis_pm_and_others = os_analysis_pts %>%
      filter(pm_and_other_mets == "PM and others"),
    os_analysis_no_pm = os_analysis_pts %>%
      filter(pm_and_other_mets == "No PM"),
    excluded_pfs_analysis = os_analysis_pts %>% filter(look_regimen_line == "No 1L combination/approved therapy" | cpt_after_pfs_i_and_m_g == 1),
    excluded_pfs_analysis_cpt_after_pfs = os_analysis_pts %>% filter(cpt_after_pfs_i_and_m_g == 1),
    excluded_pfs_analysis_no_regimen = os_analysis_pts %>% filter(look_regimen_line == "No 1L combination/approved therapy"),
    pfs_analysis_pts = anti_join(os_analysis_pts, excluded_pfs_analysis, by = "record_id"),
    pfs_analysis_pm_only = pfs_analysis_pts %>% filter(pm_and_other_mets == "PM only"),
    pfs_analysis_pm_and_others = pfs_analysis_pts %>% filter(pm_and_other_mets == "PM and others"),
    pfs_analysis_no_pm = pfs_analysis_pts %>% filter(pm_and_other_mets == "No PM")
  ) %>% 
  cohort_label(
    .full = "Overall population",
    metastatic_pts = "Clinical and molecular analysis population",
    metastatic_pts_pm_only = "PM only",
    metastatic_pts_pm_and_others = "PM and others",
    metastatic_pts_no_pm = "No PM",
    excluded_non_met_stg_i_iii = "Non-metastatic stage I-III CRC",
    excluded_os_analysis = "Patients excluded from OS analysis",
    excluded_os_analysis_cpt = "Patients with CPT after OS",
    os_analysis_pts = "OS analysis",
    os_analysis_pm_only = "PM only",
    os_analysis_pm_and_others = "PM and others",
    os_analysis_no_pm = "No PM",
    excluded_pfs_analysis = "Patients excluded from PFS analysis",
    excluded_pfs_analysis_cpt_after_pfs = "Patients with CPT after PFS",
    excluded_pfs_analysis_no_regimen = "Patients w/o 1L approved/combo trt",
    pfs_analysis_pts = "PFS analysis",
    pfs_analysis_pm_only = "PM only",
    pfs_analysis_pm_and_others = "PM and others",
    pfs_analysis_no_pm = "No PM"
  ) -> study_cohorts

study_cohorts %>%
  consort_box_add(
    name = "full", x = 8, y = 100, label = cohort_count_adorn(., .full)
  ) %>% 
  consort_box_add(
    "excluded_non_met_stg_i_iii", 10, 85, cohort_count_adorn(., excluded_non_met_stg_i_iii)
  ) %>% 
  consort_box_add(
    "metastatic_pts", 8, 70, glue::glue(
      '{cohort_count_adorn(study_cohorts, metastatic_pts)}<br>
       {cohort_count_adorn(study_cohorts, metastatic_pts_pm_only)}<br>
       {cohort_count_adorn(study_cohorts, metastatic_pts_pm_and_others)}<br>
       {cohort_count_adorn(study_cohorts, metastatic_pts_no_pm)}
      ')
  ) %>% 
  consort_box_add(
    "excluded_os_analysis", 10, 48, glue::glue(
      '{cohort_count_adorn(., excluded_os_analysis)}<br>
       {cohort_count_adorn(study_cohorts, excluded_os_analysis_cpt)}<br>'
    )  
  ) %>% 
  consort_box_add("os_analysis_pts", 8, 40, glue::glue(
      '{cohort_count_adorn(study_cohorts, os_analysis_pts)}<br>
       {cohort_count_adorn(study_cohorts, os_analysis_pm_only)}<br>
       {cohort_count_adorn(study_cohorts, os_analysis_pm_and_others)}<br>
       {cohort_count_adorn(study_cohorts, os_analysis_no_pm)}
      ')
    ) %>% 
  consort_box_add(
    "excluded_pfs_analysis", 10, 20, glue::glue(
      '{cohort_count_adorn(., excluded_pfs_analysis)}<br>
       {cohort_count_adorn(study_cohorts, excluded_pfs_analysis_cpt_after_pfs)}<br>
       {cohort_count_adorn(study_cohorts, excluded_pfs_analysis_no_regimen)}<br>'
    )  
  ) %>% 
  consort_box_add(
    "pfs_analysis_pts", 8, 15, glue::glue(
      '{cohort_count_adorn(., pfs_analysis_pts)}<br>
       {cohort_count_adorn(study_cohorts, pfs_analysis_pm_only)}<br>
       {cohort_count_adorn(study_cohorts, pfs_analysis_pm_and_others)}<br>
       {cohort_count_adorn(study_cohorts, pfs_analysis_no_pm)}<br>'
    )  
  ) %>% 
  consort_arrow_add(
    start = "full",
    start_side = "bottom",
    end = "metastatic_pts",
    end_side = "top"
  ) %>% 
   consort_arrow_add(
    end = "excluded_non_met_stg_i_iii", 
    end_side = "left", 
    start_x = 8, 
    start_y = 85
  ) %>% 
  consort_arrow_add(
    start = "metastatic_pts",
    start_side = "bottom",
    end = "os_analysis_pts",
    end_side = "top"
  ) %>% 
  consort_arrow_add(
    end = "excluded_os_analysis",
    end_side = "left",
    start_x = 8,
    start_y = 48
  ) %>% 
  consort_arrow_add(
    start = "os_analysis_pts",
    start_side = "bottom",
    end = "pfs_analysis_pts",
    end_side = "top"
  ) %>% 
  consort_arrow_add(
    end = "excluded_pfs_analysis",
    end_side = "left",
    start_x = 8,
    start_y = 20
  ) -> study_consort

consort_diagram <- study_consort %>% 
  ggplot() +
  geom_consort() + 
  theme_consort(margin_h = 15, margin_v = 2)

consort_diagram
```

# Full mCRC cohort: patient characteristics by presence of PM

```{r}
min_cpt_seq_yr <-
  tbl1_cohort %>%
  summarize(min(cpt_seq_date)) %>%
  unlist()

max_cpt_seq_yr <-
  tbl1_cohort %>%
  summarize(max(cpt_seq_date)) %>%
  unlist()
```

Years of genomic sequencing: (`r min_cpt_seq_yr`, `r max_cpt_seq_yr`)

```{r}
tbl1_cohort %>%
  select(
    institution.x,
    pfs_cohort,
    age_dx,
    naaccr_sex_code,
    race_ethnicity,
    crc_type_derived,
    crc_side_derived,
    any_lung_mets_at_dx,
    any_liver_mets_at_dx,
    any_brain_mets_at_dx,
    any_bone_mets_at_dx,
    hist_grade_derived,
    histology_derived,
    msi_status_derived,
    mmr_status_derived,
    cpt_post_os,
    any_peritoneal_met
  ) %>%
  tbl_summary(
    by = any_peritoneal_met
  ) %>%
  add_overall() %>%
  add_p(include = c(-cpt_post_os, -msi_status_derived)) %>%
  bold_p() %>%
  separate_p_footnotes()
```

```{r}
tbl1_cohort %>%
  select(
    institution.x,
    pfs_cohort,
    age_dx,
    naaccr_sex_code,
    race_ethnicity,
    crc_type_derived,
    crc_side_derived,
    any_lung_mets_at_dx,
    any_liver_mets_at_dx,
    any_brain_mets_at_dx,
    any_bone_mets_at_dx,
    hist_grade_derived,
    histology_derived,
    msi_status_derived,
    mmr_status_derived,
    cpt_post_os,
    pm_and_other_mets
  ) %>%
  tbl_summary(
    by = pm_and_other_mets
  ) %>%
  add_overall()
```

# Full mCRC cohort: patient characteristics by stage group & presence/absence of PM

```{r}
tbl1_cohort %>%
  select(
    institution.x,
    pfs_cohort,
    age_dx,
    naaccr_sex_code,
    race_ethnicity,
    crc_type_derived,
    crc_side_derived,
    any_lung_mets_at_dx,
    any_liver_mets_at_dx,
    any_brain_mets_at_dx,
    any_bone_mets_at_dx,
    hist_grade_derived,
    histology_derived,
    msi_status_derived,
    mmr_status_derived,
    cpt_post_os,
    any_peritoneal_met
  ) %>%
  tbl_strata(
    strata = pfs_cohort,
    .tbl_fun =
      ~ .x %>%
      tbl_summary(
        by = any_peritoneal_met
      ) %>%
      add_p(include = c(-cpt_post_os, -msi_status_derived)) %>%
      bold_p() %>% 
      separate_p_footnotes(),
    .header = "**{strata}**, N = {n}"
  )
```

# Logistic regression models for baseline patient characteristics associated with PM

### Univariable models

```{r}
tbl_pm_univ <- tbl1_cohort %>%
  mutate(
    any_peritoneal_met_num = case_when(any_peritoneal_met == "PM" ~ 1,
                                       any_peritoneal_met == "No PM" ~ 0),
    naaccr_sex_code = relevel(factor(naaccr_sex_code), ref = "Male")
  ) %>%
  select(
    institution.x,
    pfs_cohort,
    age_dx,
    naaccr_sex_code,
    race_ethnicity,
    crc_type_derived,
    crc_side_derived,
    any_lung_mets_at_dx,
    any_liver_mets_at_dx,
    any_brain_mets_at_dx,
    any_bone_mets_at_dx,
    hist_grade_derived,
    histology_derived,
    msi_status_derived,
    mmr_status_derived,
    # cpt_post_os,
    any_peritoneal_met_num
  ) %>%
  tbl_uvregression(
    method = glm,
    y = any_peritoneal_met_num,
    method.args = list(family = binomial),
    exponentiate = TRUE,
    hide_n = TRUE
  ) %>% 
  add_global_p() %>% 
  bold_p()

tbl_pm_univ
```

### Multivariable models

```{r}
model_pm_multi <- glm(
  any_peritoneal_met_num ~ pfs_cohort + naaccr_sex_code + crc_type_derived + histology_derived + any_lung_mets_at_dx + any_liver_mets_at_dx,
  tbl1_cohort %>%
    mutate(
      any_peritoneal_met_num = case_when(any_peritoneal_met == "PM" ~ 1,
                                         any_peritoneal_met == "No PM" ~ 0),
      naaccr_sex_code = relevel(factor(naaccr_sex_code), ref = "Male")
    ),
  family = binomial(link = "logit")
)

## check for collinearity
performance::check_collinearity(model_pm_multi)
## all VIFs < 2 --> low correlation

tbl_pm_multi <- model_pm_multi %>%
  tbl_regression(exponentiate = TRUE) %>% 
  add_global_p() %>% 
  bold_p()

tbl_pm_multi
```

# Full mCRC cohort: genomic data available

```{r}
tbl1_cohort %>% 
  select(record_id, pfs_cohort, ca_seq) %>% 
    left_join(., 
              cpt_2_0,
              by = c("record_id", "ca_seq")) %>% 
  group_by(record_id, pfs_cohort) %>% 
  summarize(n_samples_available = n(), .groups = "drop") %>% 
  select(pfs_cohort, n_samples_available) %>% 
  tbl_summary(by = pfs_cohort, label = list(n_samples_available = "N genomic samples available per patient")) %>% 
  add_overall()
```

**All CPTs:**

```{r}
tbl1_cohort %>%
  select(record_id, pfs_cohort, ca_seq) %>%
  left_join(.,
            cpt_2_0,
            by = c("record_id", "ca_seq")) %>%
  mutate(
    cpt_seq_assay_id =
      case_when(
        cpt_seq_assay_id != "" ~ cpt_seq_assay_id,
        cpt_seq_assay_id == "" ~ as.character(NA)
      )
  ) %>%
  select(pfs_cohort, sample_type, cpt_seq_assay_id) %>%
  tbl_summary(
    by = pfs_cohort,
    label = list(sample_type = "Sample type",
                 cpt_seq_assay_id = "Sequencing Assay ID")
  ) %>% 
  add_overall()
```

&nbsp;

**First CPT only:**

```{r}
supplemental_tbl4 <- tbl1_cohort %>%
  select(record_id, pfs_cohort, ca_seq, dx_to_dmets_mos) %>%
  left_join(.,
            cpt_2_0,
            by = c("record_id", "ca_seq")) %>%
  group_by(record_id) %>%
  slice(which.min(ca_seq)) %>%
  slice(which.min(dx_cpt_rep_mos)) %>%
  ungroup() %>% 
  mutate(
    cpt_seq_assay_id =
      case_when(
        cpt_seq_assay_id != "" ~ cpt_seq_assay_id,
        cpt_seq_assay_id == "" ~ as.character(NA)
      ),
    adv_disease_cpt_mos_pre = case_when(pfs_cohort == "Stage IV" ~ dx_cpt_rep_mos,
                                        pfs_cohort == "Stage I-III with Distant Mets" ~ dx_cpt_rep_mos - dx_to_dmets_mos)
  ) %>%
  select(pfs_cohort, sample_type, cpt_seq_assay_id, adv_disease_cpt_mos_pre) %>%
  tbl_summary(
    by = pfs_cohort,
    label = list(sample_type = "Sample type",
                 cpt_seq_assay_id = "Sequencing Assay ID",
                 adv_disease_cpt_mos_pre = "Months from advanced diagnosis to CPT report")
  ) %>%
  add_overall()

supplemental_tbl4
```

# Full mCRC cohort: genomic characteristics

```{r}
sample_panels <- tbl1_cohort %>%
  select(record_id, ca_seq) %>%
  left_join(.,
    cpt_2_0,
    by = c("record_id", "ca_seq")
  ) %>%
  group_by(record_id, ca_seq) %>% 
  slice(which.min(dx_cpt_rep_mos)) %>% 
  ungroup() %>% 
  select(cpt_seq_assay_id, cpt_genie_sample_id) %>%
  rename(
    panel_id = cpt_seq_assay_id,
    sample_id = cpt_genie_sample_id
  )

sample_panels_with_groups <- sample_panels %>% 
  left_join(., tbl1_cohort %>%
              select(record_id, ca_seq, any_peritoneal_met, pm_and_other_mets, stage_dx_iv) %>%
              left_join(.,
                        cpt_2_0,
                        by = c("record_id", "ca_seq")
              ) %>%
              group_by(record_id, ca_seq) %>% 
              slice(which.min(dx_cpt_rep_mos)) %>% 
              ungroup() %>% 
              select(cpt_genie_sample_id, any_peritoneal_met, pm_and_other_mets, stage_dx_iv),
            by = c("sample_id" = "cpt_genie_sample_id"))

sample_ids_any_pm <- sample_panels_with_groups %>% 
  filter(any_peritoneal_met == "PM") %>% 
  select(sample_id)

sample_ids_no_pm <- sample_panels_with_groups %>% 
  filter(any_peritoneal_met == "No PM") %>% 
  select(sample_id)

sample_ids_pm_only <- sample_panels_with_groups %>% 
  filter(pm_and_other_mets == "PM only") %>% 
  select(sample_id)

sample_ids_pm_others <- sample_panels_with_groups %>% 
  filter(pm_and_other_mets == "PM and others") %>% 
  select(sample_id)
```

```{r}
gen_dat_all_pts <- gnomeR::create_gene_binary(
    mutation = mutations_extended_2_0,
    fusion = fusions_2_0,
    cna = cna_2_0,
    samples = sample_panels$sample_id,
    specify_panel = sample_panels
  )
```

```{r}
top_freq_genes_all_pts <- gen_dat_all_pts %>%
  summarize_by_gene() %>%
  pivot_longer(cols = c(-sample_id),
               names_to = "gene",
               values_to = "alteration") %>%
  mutate(n_samples = n_distinct(sample_id)) %>%
  group_by(gene) %>%
  summarize(
    n_alterations = sum(alteration, na.rm = TRUE),
    n_samples_gene = sum(!is.na(alteration)),
    pct_samples_gene = n_samples_gene / n_samples,
    pct_alterations = n_alterations / n_samples_gene,
    .groups = "drop"
  ) %>%
  unique() %>%
  arrange(desc(pct_alterations)) %>%
  filter(pct_samples_gene >= 0.20) %>% 
  filter(pct_alterations >= 0.07) %>% 
  select(gene)

gen_dat_all_pts_top_7pct <- gen_dat_all_pts %>% 
  summarize_by_gene() %>% 
  select(sample_id, all_of(c(top_freq_genes_all_pts %>% select(gene) %>% unlist() %>% as.vector())), MED12, NRAS)
```

*The genes shown below reflect genes altered in at least 7% of samples and included on at least 20% of sample panels.*

```{r}
gen_dat_all_pts_top_7pct %>% 
  left_join(., tbl1_cohort %>% select(cpt_genie_sample_id, any_peritoneal_met, pm_and_other_mets), 
            by = c("sample_id" = "cpt_genie_sample_id")) %>% 
  select(-sample_id, -pm_and_other_mets) %>% 
  tbl_summary(by = any_peritoneal_met,
              sort = everything() ~ "frequency") %>% 
  add_overall() %>% 
  add_p() %>% 
  add_q() %>% 
  separate_p_footnotes()
```

# Full mCRC cohort: genomic characteristics by alteration type

*The genes shown below reflect genes altered in at least 7% of samples and included on at least 20% of sample panels.*

```{r}
gen_dat_all_pts %>% 
  select(sample_id, contains(c(top_freq_genes_all_pts %>% select(gene) %>% unlist() %>% as.vector()))) %>% 
  left_join(., tbl1_cohort %>% select(cpt_genie_sample_id, any_peritoneal_met, pm_and_other_mets), 
            by = c("sample_id" = "cpt_genie_sample_id")) %>% 
  select(-sample_id, -pm_and_other_mets) %>% 
  tbl_summary(by = any_peritoneal_met,
              sort = everything() ~ "frequency") %>% 
  add_overall() %>% 
  add_p() %>% 
  add_q() 
```

# Comutations in KRAS/BRAF/NRAS

Samples with alterations in at least 2 of KRAS, BRAF, NRAS

```{r}
gen_dat_all_pts %>% 
  summarize_by_gene() %>% 
  select(sample_id, KRAS, BRAF, NRAS) %>% 
  rowwise() %>% 
  mutate(look = sum(across(c(KRAS, BRAF, NRAS)))) %>% 
  filter(look > 1) %>% 
  left_join(., tbl1_cohort %>% 
              select(cpt_genie_sample_id, any_peritoneal_met, pm_and_other_mets),
            by = c("sample_id" = "cpt_genie_sample_id")) %>% 
  select(contains("met")) %>% 
  tbl_summary()
```

# BRAF V600E

```{r}
tbl1_cohort %>%
  select(record_id,
         cpt_genie_sample_id,
         any_peritoneal_met,
         pm_and_other_mets) %>%
  mutate(
    braf_v600e_pre = case_when(
      cpt_genie_sample_id %in% c(
        mutations_extended_2_0 %>%
          filter(hugoGeneSymbol == "BRAF", proteinChange == "V600E") %>%
          select(sampleId) %>%
          unlist()
      ) ~ 1,
      TRUE ~ 0
    ),
    braf_tested = case_when(
      cpt_genie_sample_id %in% c(
        gen_dat_all_pts %>%
          summarize_by_gene() %>%
          select(sample_id, BRAF) %>%
          mutate(braf_tested = case_when(!is.na(BRAF) ~ 1,
                                         is.na(BRAF) ~ 0)) %>%
          filter(braf_tested == 1) %>%
          select(sample_id) %>%
          unlist()
      ) ~ 1,
      TRUE ~ 0
    ),
    braf_v600e = case_when(
      braf_tested == 1 & braf_v600e_pre == 1 ~ 1,
      braf_tested == 1 & braf_v600e_pre == 0 ~ 0,
      braf_tested == 0 ~ as.numeric(NA)
    )
  ) %>%
  select(braf_v600e, any_peritoneal_met) %>%
  tbl_summary(by = any_peritoneal_met,
              label = list(braf_v600e = "BRAF V600E")) %>%
  add_p(test = list(all_categorical() ~ "fisher.test"))
```

# Overall survival

*Note: for patients with multiple cancer diagnoses, OS analyses are estimated from the first indication of advanced disease according to the first index cancer diagnosis.*

```{r}
## look at # of patients with CPT report after death/last follow up
nrow_cpt_after_os <- 
  tbl1_cohort %>% 
  filter(cpt_report_post_death == 1 | cpt_report_post_last_alive == 1) %>% 
  nrow()
```

Criteria for exclusion from OS analyses

- `r nrow_cpt_after_os` patients were excluded from survival analyses because their cancer panel test report was returned after death/last follow-up.

```{r}
os_cohort <-
  tbl1_cohort %>%
  filter((cpt_report_post_death == 0 | is.na(cpt_report_post_death)),
         (cpt_report_post_last_alive == 0 | is.na(cpt_report_post_last_alive))) %>% 
  filter(!is.na(any_peritoneal_met)) %>% 
  mutate(adv_disease_cpt_mos_pre = case_when(pfs_cohort == "Stage IV" ~ dx_cpt_rep_mos,
                                         pfs_cohort == "Stage I-III with Distant Mets" ~ dx_cpt_rep_mos - dx_to_dmets_mos),
         adv_disease_cpt_mos = case_when(adv_disease_cpt_mos_pre < 0 ~ 0,
                                         adv_disease_cpt_mos_pre >= 0 ~ adv_disease_cpt_mos_pre )) %>% 
  select(-adv_disease_cpt_mos_pre) %>% 
  mutate(
    kras_any_pm = case_when(
      first_cpt_kras == 1 ~ paste0("KRASmut; ", any_peritoneal_met),
      first_cpt_kras == 0 ~ paste0("KRASwt; ", any_peritoneal_met)
    ),
    braf_any_pm = case_when(
      first_cpt_braf == 1 ~ paste0("BRAFmut; ", any_peritoneal_met),
      first_cpt_braf == 0 ~ paste0("BRAFwt; ", any_peritoneal_met)
    ),
    nras_any_pm = case_when(
      first_cpt_nras == 1 ~ paste0("NRASmut; ", any_peritoneal_met),
      first_cpt_nras == 0 ~ paste0("NRASwt; ", any_peritoneal_met)
    ),
    ras_braf_any_pm = case_when(
      first_cpt_kras_braf_nras == 1 ~ paste0("RAS/BRAFmut; ", any_peritoneal_met),
      first_cpt_kras_braf_nras == 0 ~ paste0("RAS/BRAFwt; ", any_peritoneal_met)
    ),
    apc_any_pm = case_when(
      first_cpt_apc == 1 ~ paste0("APCmut; ", any_peritoneal_met),
      first_cpt_apc == 0 ~ paste0("APCwt; ", any_peritoneal_met)
    ),
    pik3ca_any_pm = case_when(
      first_cpt_pik3ca == 1 ~ paste0("PIK3CAmut; ", any_peritoneal_met),
      first_cpt_pik3ca == 0 ~ paste0("PIK3CAwt; ", any_peritoneal_met)
    ),
    tp53_any_pm = case_when(
      first_cpt_tp53 == 1 ~ paste0("TP53mut; ", any_peritoneal_met),
      first_cpt_tp53 == 0 ~ paste0("TP53wt; ", any_peritoneal_met)
    )
  )
```

```{r}
supplemental_tbl5 <- os_cohort %>%
  select(
    institution.x,
    pfs_cohort,
    age_dx,
    naaccr_sex_code,
    race_ethnicity,
    crc_type_derived,
    crc_side_derived,
    any_lung_mets_at_dx,
    any_liver_mets_at_dx,
    any_brain_mets_at_dx,
    any_bone_mets_at_dx,
    hist_grade_derived,
    histology_derived,
    msi_status_derived,
    mmr_status_derived,
    cpt_post_os,
    any_peritoneal_met
  ) %>%
  tbl_summary(
    by = any_peritoneal_met
  ) %>%
  add_overall() 

supplemental_tbl5
```

# OS by presence/absence of PM

**Median survival in OS cohort**

```{r}
survfit(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ 1,
    data = os_cohort %>% 
      filter(tt_os_adv_mos > adv_disease_cpt_mos)
  ) %>% 
  tbl_survfit(probs = 0.5,
              estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)) 
```

**All patients**

```{r}
tbl_os_n_pm <- os_cohort %>%
  select(any_peritoneal_met) %>%
  tbl_summary(label = list(any_peritoneal_met = "Presence of PM at advanced disease"),
              statistic = list(everything() ~ "{n}")) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA) 

tbl_os_n_event_pm <- os_cohort %>%
  filter(os_adv_status == 1) %>% 
  select(any_peritoneal_met) %>%
  tbl_summary(label = list(any_peritoneal_met = "Presence of PM at advanced disease"),
              statistic = list(everything() ~ "{n}")) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA) 

tbl_med_os_pm <-
  survfit(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ any_peritoneal_met,
    data = os_cohort
  ) %>%
  tbl_survfit(probs = .5,
              label = list(any_peritoneal_met = "Presence of PM at advanced disease"),
              label_header = "**Median overall survival (months)**",
              estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4))

left_trunc_os_by_any_pm <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ any_peritoneal_met,
    data = os_cohort
  ))

logrank_pval_os_by_any_pm <- left_trunc_os_by_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge(list(tbl_os_n_pm, tbl_os_n_event_pm, tbl_med_os_pm)) %>% 
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "Presence of PM at advanced disease",
    footnote = paste0("Left truncation adjusted log-rank p-value: ", round(logrank_pval_os_by_any_pm, 3))
  )
```

```{r}
figure_2a <- survfit2(Surv(time = adv_disease_cpt_mos, time2 = tt_os_adv_mos, event = os_adv_status) ~ any_peritoneal_met, 
         data = os_cohort %>% 
           filter(tt_os_adv_mos > adv_disease_cpt_mos)) %>%
  ggsurvfit(theme = list(theme_classic(), 
                         theme(legend.position = "top"))
            ) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  scale_x_continuous(limits = c(0, 120), 
                     breaks = seq(0, 120, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B")) +
  labs(x = "Time from advanced disease (months)",
       y = "Probability of advanced disease",
       title = "A. OS from diagnosis of advanced disease: PM vs. no PM")

figure_2a
```

```{r}
survfit2(Surv(time = adv_disease_cpt_mos, time2 = tt_os_adv_mos, event = os_adv_status) ~ any_peritoneal_met + pfs_cohort, 
         data = os_cohort) %>%
  ggsurvfit(theme = list(theme_classic(), 
                         theme(legend.position = "right"))
            ) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  scale_x_continuous(limits = c(0, 120), 
                     breaks = seq(0, 120, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B", "#4C8B2B")) +
  labs(x = "Time from advanced disease (months)",
       y = "Probability of advanced disease",
       title = "Overall survival from advanced disease")
```

**Stage IV**

```{r}
tbl_os_n_pm_stg_iv <- os_cohort %>%
  filter(pfs_cohort == "Stage IV") %>% 
  select(any_peritoneal_met) %>%
  tbl_summary(label = list(any_peritoneal_met = "Presence of PM at advanced disease"),
              statistic = list(everything() ~ "{n}")) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA) 

tbl_os_n_event_pm_stg_iv <- os_cohort %>%
  filter(pfs_cohort == "Stage IV") %>% 
  filter(os_adv_status == 1) %>% 
  select(any_peritoneal_met) %>%
  tbl_summary(label = list(any_peritoneal_met = "Presence of PM at advanced disease"),
              statistic = list(everything() ~ "{n}")) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA) 

tbl_med_os_pm_stg_iv <-
  survfit(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ any_peritoneal_met,
    data = os_cohort %>% filter(pfs_cohort == "Stage IV") 
  ) %>%
  tbl_survfit(probs = .5,
              label = list(any_peritoneal_met = "Presence of PM at advanced disease"),
              label_header = "**Median overall survival (months)**",
              estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4))

left_trunc_os_by_any_pm_stg_iv <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ any_peritoneal_met,
    data = os_cohort %>% filter(pfs_cohort == "Stage IV") 
  ))

logrank_pval_os_by_any_pm_stg_iv <- left_trunc_os_by_any_pm_stg_iv$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge(list(tbl_os_n_pm_stg_iv, tbl_os_n_event_pm_stg_iv, tbl_med_os_pm_stg_iv)) %>% 
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "Presence of PM at advanced disease",
    footnote = paste0("Left truncation adjusted log-rank p-value: ", round(logrank_pval_os_by_any_pm_stg_iv, 3))
  )
```

```{r}
figure_2c <- survfit2(Surv(time = adv_disease_cpt_mos, time2 = tt_os_adv_mos, event = os_adv_status) ~ any_peritoneal_met, 
         data = os_cohort %>% filter(pfs_cohort == "Stage IV")) %>%
  ggsurvfit(theme = list(theme_classic(), 
                         theme(legend.position = "top"))
            ) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  scale_x_continuous(limits = c(0, 120), 
                     breaks = seq(0, 120, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B")) +
  labs(x = "Time from advanced disease (months)",
       y = "Probability of advanced disease",
       title = "OS from stage IV diagnosis: PM vs. No PM")

figure_2c
```

**Stage I-III patients with distant mets**

```{r}
tbl_os_n_pm_stg_i_iii <- os_cohort %>%
  filter(pfs_cohort == "Stage I-III with Distant Mets") %>% 
  select(any_peritoneal_met) %>%
  tbl_summary(label = list(any_peritoneal_met = "Presence of PM at advanced disease"),
              statistic = list(everything() ~ "{n}")) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA) 

tbl_os_n_event_pm_stg_i_iii <- os_cohort %>%
  filter(pfs_cohort == "Stage I-III with Distant Mets") %>% 
  filter(os_adv_status == 1) %>% 
  select(any_peritoneal_met) %>%
  tbl_summary(label = list(any_peritoneal_met = "Presence of PM at advanced disease"),
              statistic = list(everything() ~ "{n}")) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA) 

tbl_med_os_pm_stg_i_iii <-
  survfit(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ any_peritoneal_met,
    data = os_cohort %>% filter(pfs_cohort == "Stage I-III with Distant Mets") 
  ) %>%
  tbl_survfit(probs = .5,
              label = list(any_peritoneal_met = "Presence of PM at advanced disease"),
              label_header = "**Median overall survival (months)**",
              estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4))

left_trunc_os_by_any_pm_stg_i_iii <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ any_peritoneal_met,
    data = os_cohort %>% filter(pfs_cohort == "Stage I-III with Distant Mets") 
  ))

logrank_pval_os_by_any_pm_stg_i_iii <- left_trunc_os_by_any_pm_stg_i_iii$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge(list(tbl_os_n_pm_stg_i_iii, tbl_os_n_event_pm_stg_i_iii, tbl_med_os_pm_stg_i_iii)) %>% 
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "Presence of PM at advanced disease",
    footnote = paste0("Left truncation adjusted log-rank p-value: ", round(logrank_pval_os_by_any_pm_stg_i_iii, 3))
  )
```

```{r}
figure_2d <- survfit2(Surv(time = adv_disease_cpt_mos, time2 = tt_os_adv_mos, event = os_adv_status) ~ any_peritoneal_met, 
         data = os_cohort %>% filter(pfs_cohort == "Stage I-III with Distant Mets",
                                     tt_os_adv_mos > adv_disease_cpt_mos)) %>%
  ggsurvfit(theme = list(theme_classic(), 
                         theme(legend.position = "top"))
            ) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  scale_x_continuous(limits = c(0, 120), 
                     breaks = seq(0, 120, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B")) +
  labs(x = "Time from advanced disease (months)",
       y = "Probability of advanced disease",
       title = "OS from diagnosis of advanced disease among Stage I-III with distant metastasis: PM vs. No PM")

figure_2d
```

# OS by PM only/PM & others/no PM

**All patients**

```{r}
tbl_os_n_pm_and_others <- os_cohort %>%
  select(pm_and_other_mets) %>%
  # mutate(pm_and_other_mets = as.character(pm_and_other_mets)) %>%
  tbl_summary(label = list(pm_and_other_mets = "Presence of PM at advanced disease"),
              statistic = list(everything() ~ "{n}")) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_n_event_pm_and_others <- os_cohort %>%
  filter(os_adv_status == 1) %>%
  select(pm_and_other_mets) %>%
  tbl_summary(label = list(pm_and_other_mets = "Presence of PM at advanced disease"),
              statistic = list(everything() ~ "{n}")) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_med_os_pm_and_others <-
  survfit(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ pm_and_other_mets,
    data = os_cohort
  ) %>%
  tbl_survfit(
    probs = .5,
    label = list(pm_and_other_mets = "Presence of PM at advanced disease"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_by_pm_others <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ pm_and_other_mets,
    data = os_cohort
  ))

logrank_pval_os_by_pm_others <- left_trunc_os_by_pm_others$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge(
  list(
    tbl_os_n_pm_and_others,
    tbl_os_n_event_pm_and_others,
    tbl_med_os_pm_and_others
  )
) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "Presence of PM at advanced disease",
    footnote = paste0("Left truncation adjusted log-rank p-value: ", logrank_pval_os_by_pm_others)
  )
```

```{r}
figure_2b <- survfit2(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ pm_and_other_mets,
  data = os_cohort %>% 
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  scale_x_continuous(limits = c(0, 120), 
                     breaks = seq(0, 120, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B")) +
  labs(x = "Time from advanced disease (months)",
       y = "Probability of overall survival",
       title = "OS from diagnosis of advanced disease: PM only vs. PM and others vs. No PM")

figure_2b
```

**Stage IV patients**

```{r}
tbl_os_n_pm_and_others_stg_iv <- os_cohort %>%
  filter(pfs_cohort == "Stage IV") %>% 
  select(pm_and_other_mets) %>%
  tbl_summary(label = list(pm_and_other_mets = "Presence of PM at advanced disease"),
              statistic = list(everything() ~ "{n}")) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_n_event_pm_and_others_stg_iv <- os_cohort %>%
  filter(pfs_cohort == "Stage IV") %>% 
  filter(os_adv_status == 1) %>%
  select(pm_and_other_mets) %>%
  tbl_summary(label = list(pm_and_other_mets = "Presence of PM at advanced disease"),
              statistic = list(everything() ~ "{n}")) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_med_os_pm_and_others_stg_iv <-
  survfit(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ pm_and_other_mets,
    data = os_cohort %>% filter(pfs_cohort == "Stage IV")
  ) %>%
  tbl_survfit(
    probs = .5,
    label = list(pm_and_other_mets = "Presence of PM at advanced disease"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_by_pm_others_stg_iv <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ pm_and_other_mets,
    data = os_cohort %>% filter(pfs_cohort == "Stage IV") 
  ))

logrank_pval_os_by_pm_others_stg_iv <- left_trunc_os_by_pm_others_stg_iv$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge(
  list(
    tbl_os_n_pm_and_others_stg_iv,
    tbl_os_n_event_pm_and_others_stg_iv,
    tbl_med_os_pm_and_others_stg_iv
  )
) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "Presence of PM at advanced disease",
    footnote = paste0("Left truncation adjusted log-rank p-value: ", logrank_pval_os_by_pm_others_stg_iv, 3)
  )
```

```{r}
survfit2(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ pm_and_other_mets,
  data = os_cohort %>% filter(pfs_cohort == "Stage IV")
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  scale_x_continuous(limits = c(0, 120), 
                     breaks = seq(0, 120, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B")) +
  labs(x = "Time from advanced disease (months)",
       y = "Probability of overall survival",
       title = "Overall survival from advanced disease")
```

**Stage I-III patients with distant mets**

```{r}
tbl_os_n_pm_and_others_stg_i_iii <- os_cohort %>%
  filter(pfs_cohort == "Stage I-III with Distant Mets") %>% 
  select(pm_and_other_mets) %>%
  tbl_summary(label = list(pm_and_other_mets = "Presence of PM at advanced disease"),
              statistic = list(everything() ~ "{n}")) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_n_event_pm_and_others_stg_i_iii <- os_cohort %>%
  filter(pfs_cohort == "Stage I-III with Distant Mets") %>% 
  filter(os_adv_status == 1) %>%
  select(pm_and_other_mets) %>%
  tbl_summary(label = list(pm_and_other_mets = "Presence of PM at advanced disease"),
              statistic = list(everything() ~ "{n}")) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_med_os_pm_and_others_stg_i_iii <-
  survfit(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ pm_and_other_mets,
    data = os_cohort %>% filter(pfs_cohort == "Stage I-III with Distant Mets")
  ) %>%
  tbl_survfit(
    probs = .5,
    label = list(pm_and_other_mets = "Presence of PM at advanced disease"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_by_pm_others_stg_i_iii <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ pm_and_other_mets,
    data = os_cohort %>% filter(pfs_cohort == "Stage I-III with Distant Mets") 
  ))

logrank_pval_os_by_pm_others_stg_i_iii <- left_trunc_os_by_pm_others_stg_i_iii$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge(
  list(
    tbl_os_n_pm_and_others_stg_i_iii,
    tbl_os_n_event_pm_and_others_stg_i_iii,
    tbl_med_os_pm_and_others_stg_i_iii
  )
) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "Presence of PM at advanced disease",
    footnote = paste0("Left truncation adjusted log-rank p-value: ", round(logrank_pval_os_by_pm_others_stg_i_iii, 3))
  )
```

```{r}
survfit2(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ pm_and_other_mets,
  data = os_cohort %>% filter(pfs_cohort == "Stage I-III with Distant Mets")
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  scale_x_continuous(limits = c(0, 120), 
                     breaks = seq(0, 120, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B")) +
  labs(x = "Time from advanced disease (months)",
       y = "Probability of overall survival",
       title = "Overall survival from advanced disease")
```

# OS by tumor location

```{r}
survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ crc_type_derived,
  data = os_cohort %>% 
    filter(!is.na(crc_type_derived),
           tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )
```

```{r}
survfit2(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ crc_type_derived,
  data = os_cohort %>% 
    filter(!is.na(crc_type_derived),
           tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  scale_x_continuous(limits = c(0, 120), 
                     breaks = seq(0, 120, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B")) +
  labs(x = "Time from advanced disease (months)",
       y = "Probability of overall survival",
       title = "Overall survival from advanced disease")
```

&nbsp;

```{r}
survfit2(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ crc_type_derived + any_peritoneal_met,
  data = os_cohort %>% 
    filter(!is.na(crc_type_derived),
           tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  scale_x_continuous(limits = c(0, 120),
                     breaks = seq(0, 120, by = 12)) +
  labs(x = "Time from advanced disease (months)",
       y = "Probability of overall survival",
       title = "Overall survival from advanced disease")
```

# OS among KRAS/BRAF/NRASmut patients

```{r}
tbl_n_kras_any_pm <- os_cohort %>%
  filter(first_cpt_kras== 1,
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "KRASmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_kras_any_pm <- os_cohort %>%
  filter(os_adv_status == 1, first_cpt_kras == 1,
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "KRASmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_med_kras_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ any_peritoneal_met ,
  data = os_cohort %>%
    filter(first_cpt_kras == 1,
           tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "KRASmut: Presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_kras_any_pm <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ any_peritoneal_met,
    data = os_cohort %>% filter(first_cpt_kras == 1,
                                tt_os_adv_mos > adv_disease_cpt_mos) 
  ))

logrank_pval_os_kras_any_pm <- left_trunc_os_kras_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_os_med_kras_any_pm <-
  tbl_merge(list(
    tbl_n_kras_any_pm,
    tbl_n_event_kras_any_pm,
    tbl_os_med_kras_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "KRASmut: Presence of PM",
    footnote = paste0("KRASmut p-value: ", round(logrank_pval_os_kras_any_pm, 3))
  )
```

```{r}
tbl_n_braf_any_pm <- os_cohort %>%
  filter(first_cpt_braf == 1,
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "BRAFmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_braf_any_pm <- os_cohort %>%
  filter(os_adv_status == 1, first_cpt_braf == 1,
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "BRAFmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_med_braf_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ any_peritoneal_met ,
  data = os_cohort %>%
    filter(first_cpt_braf == 1,
           tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "BRAFmut: Presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_braf_any_pm <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ any_peritoneal_met,
    data = os_cohort %>% filter(first_cpt_braf == 1,
                                tt_os_adv_mos > adv_disease_cpt_mos) 
  ))

logrank_pval_os_braf_any_pm <- left_trunc_os_braf_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_os_med_braf_any_pm <-
  tbl_merge(list(
    tbl_n_braf_any_pm,
    tbl_n_event_braf_any_pm,
    tbl_os_med_braf_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "BRAFmut: Presence of PM",
    footnote = paste0("BRAFmut p-value: ", round(logrank_pval_os_braf_any_pm, 3))
  )
```

```{r}
tbl_n_nras_any_pm <- os_cohort %>%
  filter(first_cpt_nras == 1,
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "NRASmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_nras_any_pm <- os_cohort %>%
  filter(os_adv_status == 1, first_cpt_nras == 1,
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "NRASmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_med_nras_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ any_peritoneal_met ,
  data = os_cohort %>%
    filter(first_cpt_nras == 1,
           tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "NRASmut: Presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_nras_any_pm <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ any_peritoneal_met,
    data = os_cohort %>% filter(first_cpt_nras == 1,
                                tt_os_adv_mos > adv_disease_cpt_mos) 
  ))

logrank_pval_os_nras_any_pm <- left_trunc_os_nras_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_os_med_nras_any_pm <-
  tbl_merge(list(
    tbl_n_nras_any_pm,
    tbl_n_event_nras_any_pm,
    tbl_os_med_nras_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "NRASmut: Presence of PM",
    footnote = paste0("NRASmut p-value: ", round(logrank_pval_os_nras_any_pm, 3))
  )
```

```{r}
tbl_n_ras_braf_any_pm <- os_cohort %>%
  filter(first_cpt_kras_braf_nras == 1,
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "KRAS/BRAF/NRASmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_ras_braf_any_pm <- os_cohort %>%
  filter(os_adv_status == 1, first_cpt_kras_braf_nras == 1,
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "KRAS/BRAF/NRASmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_med_ras_braf_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ any_peritoneal_met ,
  data = os_cohort %>%
    filter(first_cpt_kras_braf_nras == 1,
           tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "KRAS/BRAF/NRASmut: Presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_ras_braf_any_pm <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ any_peritoneal_met,
    data = os_cohort %>% filter(first_cpt_kras_braf_nras == 1,
                                tt_os_adv_mos > adv_disease_cpt_mos) 
  ))

logrank_pval_os_ras_braf_any_pm <- left_trunc_os_ras_braf_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_os_med_ras_braf_any_pm <-
  tbl_merge(
    list(
      tbl_n_ras_braf_any_pm,
      tbl_n_event_ras_braf_any_pm,
      tbl_os_med_ras_braf_any_pm
    )
  ) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "KRAS/BRAF/NRASmut: Presence of PM",
    footnote = paste0("KRAS/BRAF/NRASmut p-value: ", round(logrank_pval_os_ras_braf_any_pm, 3))
  )
```

```{r}
tbl_n_pik3ca_any_pm <- os_cohort %>%
  filter(first_cpt_pik3ca== 1,
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "PIK3CAmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_pik3ca_any_pm <- os_cohort %>%
  filter(os_adv_status == 1, first_cpt_pik3ca == 1,
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "PIK3CAmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_med_pik3ca_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ any_peritoneal_met ,
  data = os_cohort %>%
    filter(first_cpt_pik3ca == 1,
           tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "PIK3CAmut: Presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_pik3ca_any_pm <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ any_peritoneal_met,
    data = os_cohort %>% filter(first_cpt_pik3ca == 1,
                                tt_os_adv_mos > adv_disease_cpt_mos) 
  ))

logrank_pval_os_pik3ca_any_pm <- left_trunc_os_pik3ca_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_os_med_pik3ca_any_pm <-
  tbl_merge(list(
    tbl_n_pik3ca_any_pm,
    tbl_n_event_pik3ca_any_pm,
    tbl_os_med_pik3ca_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "PIK3CAmut: Presence of PM",
    footnote = paste0("PIK3CAmut p-value: ", round(logrank_pval_os_pik3ca_any_pm, 3))
  )
```

```{r}
tbl_n_apc_any_pm <- os_cohort %>%
  filter(first_cpt_apc== 1,
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "APCmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_apc_any_pm <- os_cohort %>%
  filter(os_adv_status == 1, first_cpt_apc == 1,
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "APCmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_med_apc_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ any_peritoneal_met ,
  data = os_cohort %>%
    filter(first_cpt_apc == 1,
           tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "APCmut: Presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_apc_any_pm <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ any_peritoneal_met,
    data = os_cohort %>% filter(first_cpt_apc == 1,
                                tt_os_adv_mos > adv_disease_cpt_mos) 
  ))

logrank_pval_os_apc_any_pm <- left_trunc_os_apc_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_os_med_apc_any_pm <-
  tbl_merge(list(
    tbl_n_apc_any_pm,
    tbl_n_event_apc_any_pm,
    tbl_os_med_apc_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "APCmut: Presence of PM",
    footnote = paste0("APCmut p-value: ", round(logrank_pval_os_apc_any_pm, 3))
  )
```

```{r}
tbl_n_tp53_any_pm <- os_cohort %>%
  filter(first_cpt_tp53== 1,
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "TP53mut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_tp53_any_pm <- os_cohort %>%
  filter(os_adv_status == 1, first_cpt_tp53 == 1,
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "TP53mut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_med_tp53_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ any_peritoneal_met ,
  data = os_cohort %>%
    filter(first_cpt_tp53 == 1,
           tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "TP53mut: Presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_tp53_any_pm <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ any_peritoneal_met,
    data = os_cohort %>% filter(first_cpt_tp53 == 1,
                                tt_os_adv_mos > adv_disease_cpt_mos) 
  ))

logrank_pval_os_tp53_any_pm <- left_trunc_os_tp53_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_os_med_tp53_any_pm <-
  tbl_merge(list(
    tbl_n_tp53_any_pm,
    tbl_n_event_tp53_any_pm,
    tbl_os_med_tp53_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "TP53mut: Presence of PM",
    footnote = paste0("TP53mut p-value: ", round(logrank_pval_os_tp53_any_pm, 3))
  )
```

```{r}
## stack tbl_merge
supplemental_tbl6 <- tbl_stack(
  list(
    tbl_merge_os_med_kras_any_pm,
    tbl_merge_os_med_braf_any_pm,
    tbl_merge_os_med_nras_any_pm,
    tbl_merge_os_med_ras_braf_any_pm,
    tbl_merge_os_med_pik3ca_any_pm,
    tbl_merge_os_med_apc_any_pm,
    tbl_merge_os_med_tp53_any_pm
  )
)

supplemental_tbl6
```

# OS by KRAS/BRAF/NRAS and presence of PM

```{r}
tbl_n_kras_status_any_pm <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(kras_any_pm) %>%
  tbl_summary(
    type = list(kras_any_pm ~ "categorical"),
    label = list(kras_any_pm = "KRAS status by presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_kras_status_any_pm <- os_cohort %>%
  filter(os_adv_status == 1, 
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(kras_any_pm) %>%
  tbl_summary(
    type = list(kras_any_pm ~ "categorical"),
    label = list(kras_any_pm = "KRAS status by presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_med_kras_status_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ kras_any_pm ,
  data = os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(kras_any_pm = "KRAS status by presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_kras_status_any_pm <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ kras_any_pm,
    data = os_cohort %>% filter(tt_os_adv_mos > adv_disease_cpt_mos)
  ))

logrank_pval_os_kras_status_any_pm <- left_trunc_os_kras_status_any_pm$sctest %>%
  as.data.frame() %>%
  rownames_to_column(var = "rowname") %>%
  filter(rowname == "pvalue") %>%
  select(".") %>%
  unlist()

tbl_merge_os_med_kras_status_any_pm <-
  tbl_merge(list(
    tbl_n_kras_status_any_pm,
    tbl_n_event_kras_status_any_pm,
    tbl_os_med_kras_status_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>%
  modify_table_styling(
    columns = label,
    rows = label == "KRAS status by presence of PM",
    footnote = paste0("KRAS p-value: ", round(logrank_pval_os_kras_status_any_pm, 3))
  )
```

```{r}
tbl_n_braf_status_any_pm <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(braf_any_pm) %>%
  tbl_summary(
    type = list(braf_any_pm ~ "categorical"),
    label = list(braf_any_pm = "BRAF status by presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_braf_status_any_pm <- os_cohort %>%
  filter(os_adv_status == 1, 
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(braf_any_pm) %>%
  tbl_summary(
    type = list(braf_any_pm ~ "categorical"),
    label = list(braf_any_pm = "BRAF status by presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_med_braf_status_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ braf_any_pm ,
  data = os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(braf_any_pm = "BRAF status by presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_braf_status_any_pm <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ braf_any_pm,
    data = os_cohort %>% filter(tt_os_adv_mos > adv_disease_cpt_mos)
  ))

logrank_pval_os_braf_status_any_pm <- left_trunc_os_braf_status_any_pm$sctest %>%
  as.data.frame() %>%
  rownames_to_column(var = "rowname") %>%
  filter(rowname == "pvalue") %>%
  select(".") %>%
  unlist()

tbl_merge_os_med_braf_status_any_pm <-
  tbl_merge(list(
    tbl_n_braf_status_any_pm,
    tbl_n_event_braf_status_any_pm,
    tbl_os_med_braf_status_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>%
  modify_table_styling(
    columns = label,
    rows = label == "BRAF status by presence of PM",
    footnote = paste0("BRAF p-value: ", round(logrank_pval_os_braf_status_any_pm, 3))
  )
```

```{r}
tbl_n_nras_status_any_pm <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(nras_any_pm) %>%
  tbl_summary(
    type = list(nras_any_pm ~ "categorical"),
    label = list(nras_any_pm = "NRAS status by presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_nras_status_any_pm <- os_cohort %>%
  filter(os_adv_status == 1, 
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(nras_any_pm) %>%
  tbl_summary(
    type = list(nras_any_pm ~ "categorical"),
    label = list(nras_any_pm = "NRAS status by presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_med_nras_status_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ nras_any_pm ,
  data = os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(nras_any_pm = "NRAS status by presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_nras_status_any_pm <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ nras_any_pm,
    data = os_cohort %>% filter(tt_os_adv_mos > adv_disease_cpt_mos)
  ))

logrank_pval_os_nras_status_any_pm <- left_trunc_os_nras_status_any_pm$sctest %>%
  as.data.frame() %>%
  rownames_to_column(var = "rowname") %>%
  filter(rowname == "pvalue") %>%
  select(".") %>%
  unlist()

tbl_merge_os_med_nras_status_any_pm <-
  tbl_merge(list(
    tbl_n_nras_status_any_pm,
    tbl_n_event_nras_status_any_pm,
    tbl_os_med_nras_status_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>%
  modify_table_styling(
    columns = label,
    rows = label == "NRAS status by presence of PM",
    footnote = paste0("NRAS p-value: ", round(logrank_pval_os_nras_status_any_pm, 3))
  )
```

```{r}
tbl_n_ras_braf_status_any_pm <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(ras_braf_any_pm) %>%
  tbl_summary(
    type = list(ras_braf_any_pm ~ "categorical"),
    label = list(ras_braf_any_pm = "RAS/BRAF status by presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_ras_braf_status_any_pm <- os_cohort %>%
  filter(os_adv_status == 1, 
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(ras_braf_any_pm) %>%
  tbl_summary(
    type = list(ras_braf_any_pm ~ "categorical"),
    label = list(ras_braf_any_pm = "RAS/BRAF status by presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_med_ras_braf_status_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ ras_braf_any_pm ,
  data = os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(ras_braf_any_pm = "RAS/BRAF status by presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_ras_braf_status_any_pm <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ ras_braf_any_pm,
    data = os_cohort %>% filter(tt_os_adv_mos > adv_disease_cpt_mos)
  ))

logrank_pval_os_ras_braf_status_any_pm <- left_trunc_os_ras_braf_status_any_pm$sctest %>%
  as.data.frame() %>%
  rownames_to_column(var = "rowname") %>%
  filter(rowname == "pvalue") %>%
  select(".") %>%
  unlist()

tbl_merge_os_med_ras_braf_status_any_pm <-
  tbl_merge(list(
    tbl_n_ras_braf_status_any_pm,
    tbl_n_event_ras_braf_status_any_pm,
    tbl_os_med_ras_braf_status_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>%
  modify_table_styling(
    columns = label,
    rows = label == "RAS/BRAF status by presence of PM",
    footnote = paste0("RAS/BRAF p-value: ", round(logrank_pval_os_ras_braf_status_any_pm, 3))
  )
```

```{r}
tbl_n_apc_status_any_pm <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(apc_any_pm) %>%
  tbl_summary(
    type = list(apc_any_pm ~ "categorical"),
    label = list(apc_any_pm = "APC status by presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_apc_status_any_pm <- os_cohort %>%
  filter(os_adv_status == 1, 
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(apc_any_pm) %>%
  tbl_summary(
    type = list(apc_any_pm ~ "categorical"),
    label = list(apc_any_pm = "APC status by presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_med_apc_status_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ apc_any_pm ,
  data = os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(apc_any_pm = "APC status by presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_apc_status_any_pm <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ apc_any_pm,
    data = os_cohort %>% filter(tt_os_adv_mos > adv_disease_cpt_mos)
  ))

logrank_pval_os_apc_status_any_pm <- left_trunc_os_apc_status_any_pm$sctest %>%
  as.data.frame() %>%
  rownames_to_column(var = "rowname") %>%
  filter(rowname == "pvalue") %>%
  select(".") %>%
  unlist()

tbl_merge_os_med_apc_status_any_pm <-
  tbl_merge(list(
    tbl_n_apc_status_any_pm,
    tbl_n_event_apc_status_any_pm,
    tbl_os_med_apc_status_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>%
  modify_table_styling(
    columns = label,
    rows = label == "APC status by presence of PM",
    footnote = paste0("APC p-value: ", round(logrank_pval_os_apc_status_any_pm, 3))
  )
```

```{r}
tbl_n_pik3ca_status_any_pm <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(pik3ca_any_pm) %>%
  tbl_summary(
    type = list(pik3ca_any_pm ~ "categorical"),
    label = list(pik3ca_any_pm = "PIK3CA status by presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_pik3ca_status_any_pm <- os_cohort %>%
  filter(os_adv_status == 1, 
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(pik3ca_any_pm) %>%
  tbl_summary(
    type = list(pik3ca_any_pm ~ "categorical"),
    label = list(pik3ca_any_pm = "PIK3CA status by presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_med_pik3ca_status_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ pik3ca_any_pm ,
  data = os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(pik3ca_any_pm = "PIK3CA status by presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_pik3ca_status_any_pm <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ pik3ca_any_pm,
    data = os_cohort %>% filter(tt_os_adv_mos > adv_disease_cpt_mos)
  ))

logrank_pval_os_pik3ca_status_any_pm <- left_trunc_os_pik3ca_status_any_pm$sctest %>%
  as.data.frame() %>%
  rownames_to_column(var = "rowname") %>%
  filter(rowname == "pvalue") %>%
  select(".") %>%
  unlist()

tbl_merge_os_med_pik3ca_status_any_pm <-
  tbl_merge(list(
    tbl_n_pik3ca_status_any_pm,
    tbl_n_event_pik3ca_status_any_pm,
    tbl_os_med_pik3ca_status_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>%
  modify_table_styling(
    columns = label,
    rows = label == "PIK3CA status by presence of PM",
    footnote = paste0("PIK3CA p-value: ", round(logrank_pval_os_pik3ca_status_any_pm, 3))
  )
```

```{r}
tbl_n_tp53_status_any_pm <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(tp53_any_pm) %>%
  tbl_summary(
    type = list(tp53_any_pm ~ "categorical"),
    label = list(tp53_any_pm = "TP53 status by presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_tp53_status_any_pm <- os_cohort %>%
  filter(os_adv_status == 1, 
         tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(tp53_any_pm) %>%
  tbl_summary(
    type = list(tp53_any_pm ~ "categorical"),
    label = list(tp53_any_pm = "TP53 status by presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_med_tp53_status_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ tp53_any_pm ,
  data = os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(tp53_any_pm = "TP53 status by presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_tp53_status_any_pm <-
  summary(coxph(
    Surv(
      time = adv_disease_cpt_mos,
      time2 = tt_os_adv_mos,
      event = os_adv_status
    ) ~ tp53_any_pm,
    data = os_cohort %>% filter(tt_os_adv_mos > adv_disease_cpt_mos)
  ))

logrank_pval_os_tp53_status_any_pm <- left_trunc_os_tp53_status_any_pm$sctest %>%
  as.data.frame() %>%
  rownames_to_column(var = "rowname") %>%
  filter(rowname == "pvalue") %>%
  select(".") %>%
  unlist()

tbl_merge_os_med_tp53_status_any_pm <-
  tbl_merge(list(
    tbl_n_tp53_status_any_pm,
    tbl_n_event_tp53_status_any_pm,
    tbl_os_med_tp53_status_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>%
  modify_table_styling(
    columns = label,
    rows = label == "TP53 status by presence of PM",
    footnote = paste0("TP53 p-value: ", round(logrank_pval_os_tp53_status_any_pm, 3))
  )
```

```{r}
## stack tbl_merge
supplemental_tbl6_v2 <- tbl_stack(
  list(
    tbl_merge_os_med_kras_status_any_pm,
    tbl_merge_os_med_braf_status_any_pm,
    tbl_merge_os_med_nras_status_any_pm,
    tbl_merge_os_med_ras_braf_status_any_pm,
    tbl_merge_os_med_pik3ca_status_any_pm,
    tbl_merge_os_med_apc_status_any_pm,
    tbl_merge_os_med_tp53_status_any_pm
  )
)

supplemental_tbl6_v2
```

# OS from first-line approved combination therapy by presence/absence of PM

```{r}
os_cohort_first_line_folfoxiri_cohort <- os_cohort %>%
  left_join(
    .,
    regimen_order_df %>% select(
      # cohort,
      record_id,
      ca_seq,
      regimen_number,
      regimen_line_derived,
      regimen_drugs,
      dx_reg_start_int_mos,
      os_g_status,
      tt_os_g_mos
    ),
    by = c("record_id", "ca_seq")
  ) %>%
  filter(
    regimen_line_derived == 1,
    regimen_drugs %in% c(
      ## FOLFOX
      "Fluorouracil, Leucovorin Calcium, Oxaliplatin",
      ## FOLFIRI
      "Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium",
      ## FOLFOXIRI
      "Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium, Oxaliplatin",
      ## FOLFOX + Bev
      "Bevacizumab, Fluorouracil, Leucovorin Calcium, Oxaliplatin",
      ## FOLFIRI + Bev
      "Bevacizumab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium",
      ## FOLFOXIRI + Bev
      "Bevacizumab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium, Oxaliplatin",
      ## FOLFOX + Cetuximab
      "Cetuximab, Fluorouracil, Leucovorin Calcium, Oxaliplatin",
      ## FOLFIRI + Cetuximab
      "Cetuximab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium",
      ## FOLFOXIRI + Cetuximab
      
      "Bevacizumab, Capecitabine, Oxaliplatin",
      "Bevacizumab, Capecitabine, Fluorouracil, Leucovorin Calcium, Oxaliplatin",
      "Capecitabine, Oxaliplatin",
      "Capecitabine, Fluorouracil, Leucovorin Calcium, Oxaliplatin",
      "Cetuximab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium, Oxaliplatin",
      "Cetuximab, Fluorouracil, Leucovorin Calcium, Oxaliplatin",
      "Cetuximab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium",
      "Fluorouracil, Irinotecan Hydrochloride",
      "Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium, Panitumumab",
      "Fluorouracil, Leucovorin Calcium, Oxaliplatin, Panitumumab"
    )
  ) %>%
  mutate(## calculate time from regimen start to CPT report
    reg_start_cpt_rep_mos = dx_cpt_rep_mos - dx_reg_start_int_mos)

# os_cohort_first_line_folfoxiri_cohort %>% 
#   select(regimen_drugs) %>% 
#   tbl_summary(label = list(regimen_drugs = "Drug regimens"),
#               sort = everything() ~ "frequency")
```

```{r}
tbl_os_n_folfoxiri_full_cohort <- os_cohort_first_line_folfoxiri_cohort %>% 
  select(any_peritoneal_met) %>% 
  tbl_summary(
    label = list(any_peritoneal_met = "Presence of PM at advanced disease"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_n_event_folfoxiri_full_cohort <- os_cohort_first_line_folfoxiri_cohort %>% 
  filter(os_g_status == 1) %>% 
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "Presence of PM at advanced disease"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA) 

tbl_med_os_folfoxiri_full_cohort <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_os_g_mos,
    event = os_g_status
  ) ~ any_peritoneal_met,
  data = os_cohort_first_line_folfoxiri_cohort) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "Presence of PM at advanced disease"),
    label_header = "**Median OS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_folfoxiri_pm <-
  summary(coxph(
   Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_os_g_mos,
    event = os_g_status
  ) ~ any_peritoneal_met,
  data = os_cohort_first_line_folfoxiri_cohort
  ))

logrank_pval_os_folfoxiri_pm <- left_trunc_os_folfoxiri_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge(
  list(
    tbl_os_n_folfoxiri_full_cohort,
    tbl_os_n_event_folfoxiri_full_cohort,
    tbl_med_os_folfoxiri_full_cohort
  )
) %>%
  modify_spanning_header(everything() ~ NA_character_) %>%
  modify_table_styling(
    columns = label,
    rows = label == "Presence of PM at advanced disease",
    footnote = paste0("Left truncation adjusted logrank p-value: ", round(logrank_pval_os_folfoxiri_pm, 3))
  )
```

```{r}
survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_os_g_mos,
    event = os_g_status
  ) ~ any_peritoneal_met,
  data = os_cohort_first_line_folfoxiri_cohort
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  coord_cartesian(xlim = c(0, 72)) +
  scale_x_continuous(breaks = seq(0, 72, by = 6)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B")) +
  labs(x = "Time from start of first-line regimen (months)",
       y = "Probability of survival",
       title = "Overall survival from start of first-line regimen")
```

# OS from first-line approved combination therapy by PM only/PM & others/no PM

```{r}
tbl_os_n_folfoxiri_full_cohort_pm_others <- os_cohort_first_line_folfoxiri_cohort %>% 
  select(pm_and_other_mets) %>% 
  tbl_summary(
    label = list(pm_and_other_mets = "Presence of PM at advanced disease"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_os_n_event_folfoxiri_full_cohort_pm_others <- os_cohort_first_line_folfoxiri_cohort %>% 
  filter(os_g_status == 1) %>% 
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "Presence of PM at advanced disease"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA) 

tbl_med_os_folfoxiri_full_cohort_pm_others <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_os_g_mos,
    event = os_g_status
  ) ~ pm_and_other_mets,
  data = os_cohort_first_line_folfoxiri_cohort) %>%
  tbl_survfit(
    probs = .5,
    label = list(pm_and_other_mets = "Presence of PM at advanced disease"),
    label_header = "**Median OS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_os_folfoxiri_pm_others <-
  summary(coxph(
   Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_os_g_mos,
    event = os_g_status
  ) ~ pm_and_other_mets,
  data = os_cohort_first_line_folfoxiri_cohort
  ))

logrank_pval_os_folfoxiri_pm_others <- left_trunc_os_folfoxiri_pm_others$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge(
  list(
    tbl_os_n_folfoxiri_full_cohort_pm_others,
    tbl_os_n_event_folfoxiri_full_cohort_pm_others,
    tbl_med_os_folfoxiri_full_cohort_pm_others
  )
) %>%
  modify_spanning_header(everything() ~ NA_character_) %>%
  modify_table_styling(
    columns = label,
    rows = label == "Presence of PM at advanced disease",
    footnote = paste0("Left truncation adjusted logrank p-value: ", round(logrank_pval_os_folfoxiri_pm_others, 3))
  )
```

```{r}
survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_os_g_mos,
    event = os_g_status
  ) ~ pm_and_other_mets,
  data = os_cohort_first_line_folfoxiri_cohort
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  coord_cartesian(xlim = c(0, 72)) +
  scale_x_continuous(limits = c(0, 72), breaks = seq(0, 72, by = 6)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B")) +
  labs(x = "Time from start of first-line regimen (months)",
       y = "Probability of survival",
       title = "Overall survival from start of first-line regimen")
```

# OS: Cox modeling

*The multivariable models below estimates the associations between covariates and OS from advanced disease.*

### Any PM vs. no PM 

```{r}
cox.zph(coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + any_peritoneal_met,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
), global = TRUE)$table

tbl_n_any_pm <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_os_event_any_pm <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos,
         os_adv_status == 1) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**Event N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_os_any_pm_cox <- coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + any_peritoneal_met,
  os_cohort %>% 
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(adv_disease_cpt_mos = "Months from advanced dx to CPT report",
                 any_peritoneal_met = "Presence of PM")
  ) %>% 
  add_global_p() %>% 
  bold_p()

tbl_univ_os_any_pm <-
  tbl_merge(list(
    tbl_n_any_pm,
    tbl_n_os_event_any_pm,
    tbl_univ_os_any_pm_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_os_any_pm
```

### PM only vs. PM and other mets vs. no PM 

```{r}
cox.zph(coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + pm_and_other_mets,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
))

tbl_n_pm_others <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_os_event_pm_others <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos,
         os_adv_status == 1) %>%
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**Event N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_os_pm_and_other_mets_cox <- coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + pm_and_other_mets,
  os_cohort %>% 
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(adv_disease_cpt_mos = "Months from advanced dx to CPT report",
                 pm_and_other_mets = "Presence of PM")
  ) %>% 
  add_global_p() %>% 
  bold_p()

tbl_univ_os_pm_and_other_mets <- 
  tbl_merge(list(
    tbl_n_pm_others,
    tbl_n_os_event_pm_others,
    tbl_univ_os_pm_and_other_mets_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_os_pm_and_other_mets
```

### Stage IV vs. stage I-III with distant mets

```{r}
cox.zph(coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + pfs_cohort,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
))

tbl_n_stage <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(pfs_cohort) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_os_event_stage <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos,
         os_adv_status == 1) %>%
  select(pfs_cohort) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**Event N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_os_stage_cox <- coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + pfs_cohort,
  os_cohort %>% 
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(adv_disease_cpt_mos = "Months from advanced dx to CPT report")
  ) %>% 
  add_global_p() %>% 
  bold_p()

tbl_univ_os_stage <-
    tbl_merge(list(
    tbl_n_stage,
    tbl_n_os_event_stage,
    tbl_univ_os_stage_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_os_stage
```

### Tumor location

```{r}
cox.zph(coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + crc_type_derived,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
))

tbl_n_tumor_location <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(crc_type_derived) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_os_event_tumor_location <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos,
         os_adv_status == 1) %>%
  select(crc_type_derived) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**Event N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_os_tumor_location_cox <- coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + crc_type_derived,
  os_cohort %>% 
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(adv_disease_cpt_mos = "Months from advanced dx to CPT report")
  ) %>% 
  add_global_p() %>% 
  bold_p()

tbl_univ_os_tumor_location <- 
  tbl_merge(list(
    tbl_n_tumor_location,
    tbl_n_os_event_tumor_location,
    tbl_univ_os_tumor_location_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_os_tumor_location
```

### Tumor sidedness

```{r}
cox.zph(coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + crc_side_derived,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
))

tbl_n_tumor_sidedness <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(crc_side_derived) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_os_event_sidedness <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos,
         os_adv_status == 1) %>%
  select(crc_side_derived) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**Event N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_os_tumor_side_cox <- coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + crc_side_derived,
  os_cohort %>% 
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(adv_disease_cpt_mos = "Months from advanced dx to CPT report")
  ) %>% 
  add_global_p() %>% 
  bold_p()

tbl_univ_os_tumor_side <- 
  tbl_merge(list(
    tbl_n_tumor_sidedness,
    tbl_n_os_event_sidedness,
    tbl_univ_os_tumor_side_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_os_tumor_side
```

### MSI status

```{r}
cox.zph(coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + msi_status_derived,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
    mutate(msi_status_derived = relevel(factor(msi_status_derived), ref = "MSI-L/MSS"))
))

tbl_n_msi <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  mutate(msi_status_derived = relevel(factor(
    case_when(
      msi_status_derived == "Non-concordant" ~ as.character(NA),
      TRUE ~ msi_status_derived
    )
  ), ref = "MSI-L/MSS")) %>%
  select(msi_status_derived) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}"),
    label = list(msi_status_derived = "MSI status")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_os_event_msi <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos,
         os_adv_status == 1) %>%
  mutate(msi_status_derived = relevel(factor(
    case_when(
      msi_status_derived == "Non-concordant" ~ as.character(NA),
      TRUE ~ msi_status_derived
    )
  ), ref = "MSI-L/MSS")) %>%
  select(msi_status_derived) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}"),
    label = list(msi_status_derived = "MSI status")
  ) %>%
  modify_header(update = list(stat_0 ~ "**Event N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_os_msi_cox <- coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + msi_status_derived,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
    mutate(msi_status_derived = relevel(factor(
      case_when(
        msi_status_derived == "Non-concordant" ~ as.character(NA),
        TRUE ~ msi_status_derived
      )
    ), ref = "MSI-L/MSS"))
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(
      adv_disease_cpt_mos = "Months from advanced dx to CPT report",
      msi_status_derived = "MSI status"
    )
  ) %>% 
  add_global_p() %>% 
  bold_p()

tbl_univ_os_msi <- 
  tbl_merge(list(
    tbl_n_msi,
    tbl_n_os_event_msi,
    tbl_univ_os_msi_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_os_msi
```

### MMR status

```{r}
cox.zph(coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + mmr_status_derived_factor,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
    mutate(mmr_status_derived_factor = relevel(factor(mmr_status_derived), ref = "pMMR"))
))

tbl_n_mmr <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  mutate(mmr_status_derived_factor = relevel(factor(mmr_status_derived), ref = "pMMR")) %>%
  select(mmr_status_derived_factor) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}"),
    label = list(mmr_status_derived_factor = "MMR status")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_os_event_mmr <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos,
         os_adv_status == 1) %>%
  mutate(mmr_status_derived_factor = relevel(factor(mmr_status_derived), ref = "pMMR")) %>%
  select(mmr_status_derived_factor) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}"),
    label = list(mmr_status_derived_factor = "MMR status")
  ) %>%
  modify_header(update = list(stat_0 ~ "**Event N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_os_mmr_cox <- coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + mmr_status_derived_factor,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
    mutate(mmr_status_derived_factor = relevel(factor(mmr_status_derived), ref = "pMMR"))
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(
      adv_disease_cpt_mos = "Months from advanced dx to CPT report",
      mmr_status_derived_factor = "MMR status"
    )
  ) %>% 
  add_global_p() %>% 
  bold_p()

tbl_univ_os_mmr <- 
  tbl_merge(list(
    tbl_n_mmr,
    tbl_n_os_event_mmr,
    tbl_univ_os_mmr_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_os_mmr
```

### Presence of liver mets

```{r}
cox.zph(coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + any_liver_mets_at_dx,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
))

tbl_n_liver_mets <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_liver_mets_at_dx) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_os_event_liver_mets <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos,
         os_adv_status == 1) %>%
  select(any_liver_mets_at_dx) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**Event N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_os_liver_mets_cox <- coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + any_liver_mets_at_dx,
  os_cohort %>% 
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(adv_disease_cpt_mos = "Months from advanced dx to CPT report")
  ) %>%
  add_global_p() %>% 
  bold_p()

tbl_univ_os_liver_mets <-
  tbl_merge(list(
    tbl_n_liver_mets,
    tbl_n_os_event_liver_mets,
    tbl_univ_os_liver_mets_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_os_liver_mets
```

### Presence of lung mets

```{r}
cox.zph(coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + any_lung_mets_at_dx,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
))

tbl_n_lung_mets <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(any_lung_mets_at_dx) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_os_event_lung_mets <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos,
         os_adv_status == 1) %>%
  select(any_lung_mets_at_dx) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**Event N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_os_lung_mets_cox <- coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + any_lung_mets_at_dx,
  os_cohort %>% 
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(adv_disease_cpt_mos = "Months from advanced dx to CPT report")
  ) %>%
  add_global_p() %>% 
  bold_p()

tbl_univ_os_lung_mets <- 
  tbl_merge(list(
    tbl_n_lung_mets,
    tbl_n_os_event_lung_mets,
    tbl_univ_os_lung_mets_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_os_lung_mets
```

### RAS/BRAF status

```{r}
cox.zph(coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + first_cpt_kras_braf_nras,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
))

tbl_n_ras_braf <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(first_cpt_kras_braf_nras) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_os_event_ras_braf <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos,
         os_adv_status == 1) %>%
  select(first_cpt_kras_braf_nras) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**Event N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_os_ras_braf_cox <- coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + first_cpt_kras_braf_nras,
  os_cohort %>% 
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(adv_disease_cpt_mos = "Months from advanced dx to CPT report")
  ) %>%
  add_global_p() %>% 
  bold_p()

tbl_univ_os_ras_braf <- 
  tbl_merge(list(
    tbl_n_ras_braf,
    tbl_n_os_event_ras_braf,
    tbl_univ_os_ras_braf_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_os_ras_braf
```

### TP53 status

```{r}
cox.zph(coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + first_cpt_tp53,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
))

tbl_n_tp53 <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(first_cpt_tp53) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_os_event_tp53 <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos,
         os_adv_status == 1) %>%
  select(first_cpt_tp53) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**Event N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_os_tp53_cox <- coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + first_cpt_tp53,
  os_cohort %>% 
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(adv_disease_cpt_mos = "Months from advanced dx to CPT report")
  ) %>%
  add_global_p() %>% 
  bold_p()

tbl_univ_os_tp53 <-
  tbl_merge(list(
    tbl_n_tp53,
    tbl_n_os_event_tp53,
    tbl_univ_os_tp53_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_os_tp53
```

### APC status

```{r}
cox.zph(coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + first_cpt_apc,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
))

tbl_n_apc <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
  select(first_cpt_apc) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_os_event_apc <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos,
         os_adv_status == 1) %>%
  select(first_cpt_apc) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**Event N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_os_apc_cox <- coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + first_cpt_apc,
  os_cohort %>% 
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(adv_disease_cpt_mos = "Months from advanced dx to CPT report")
  ) %>%
  add_global_p() %>% 
  bold_p()

tbl_univ_os_apc <-
  tbl_merge(list(
    tbl_n_apc,
    tbl_n_os_event_apc,
    tbl_univ_os_apc_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_os_apc
```

### PIK3CA status

```{r}
cox.zph(coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + first_cpt_pik3ca,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
))

tbl_n_pik3ca <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos) %>%
    select(first_cpt_pik3ca) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_os_event_pik3ca <- os_cohort %>%
  filter(tt_os_adv_mos > adv_disease_cpt_mos,
         os_adv_status == 1) %>%
  select(first_cpt_pik3ca) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**Event N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_os_pik3ca_cox <- coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + first_cpt_pik3ca,
  os_cohort %>% 
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(adv_disease_cpt_mos = "Months from advanced dx to CPT report")
  ) %>%
  add_global_p() %>% 
  bold_p()

tbl_univ_os_pik3ca <-
  tbl_merge(list(
    tbl_n_pik3ca,
    tbl_n_os_event_pik3ca,
    tbl_univ_os_pik3ca_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_os_pik3ca
```

```{r, include = FALSE}
tbl_univ_os_any_pm$table_body <- tbl_univ_os_any_pm$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "adv_disease_cpt_mos")

tbl_univ_os_pm_and_other_mets$table_body <- tbl_univ_os_pm_and_other_mets$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "adv_disease_cpt_mos")

tbl_univ_os_stage$table_body <- tbl_univ_os_stage$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "adv_disease_cpt_mos")

tbl_univ_os_tumor_location$table_body <- tbl_univ_os_tumor_location$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "adv_disease_cpt_mos")

tbl_univ_os_tumor_side$table_body <- tbl_univ_os_tumor_side$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "adv_disease_cpt_mos")

tbl_univ_os_msi$table_body <- tbl_univ_os_msi$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "adv_disease_cpt_mos")

tbl_univ_os_liver_mets$table_body <- tbl_univ_os_liver_mets$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "adv_disease_cpt_mos")

tbl_univ_os_lung_mets$table_body <- tbl_univ_os_lung_mets$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "adv_disease_cpt_mos")

tbl_univ_os_ras_braf$table_body <- tbl_univ_os_ras_braf$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "adv_disease_cpt_mos")

tbl_univ_os_tp53$table_body <- tbl_univ_os_tp53$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "adv_disease_cpt_mos")

tbl_univ_os_apc$table_body <- tbl_univ_os_apc$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "adv_disease_cpt_mos")

tbl_univ_os_pik3ca$table_body <- tbl_univ_os_pik3ca$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "adv_disease_cpt_mos")

tbl_univ_os_cox <- tbl_stack(
  list(
    tbl_univ_os_any_pm,
    tbl_univ_os_pm_and_other_mets,
    tbl_univ_os_stage,
    tbl_univ_os_tumor_location,
    tbl_univ_os_tumor_side,
    tbl_univ_os_msi,
    tbl_univ_os_liver_mets,
    tbl_univ_os_lung_mets,
    tbl_univ_os_ras_braf,
    tbl_univ_os_tp53,
    tbl_univ_os_apc,
    tbl_univ_os_pik3ca
  )
)

tbl_univ_os_cox
```

### Multiple covariates

```{r}
cox.zph(coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + any_peritoneal_met + pfs_cohort + first_cpt_kras_braf_nras + first_cpt_apc + any_liver_mets_at_dx + any_lung_mets_at_dx,
  os_cohort %>%
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
))

tbl_multi_os_cox <- coxph(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ adv_disease_cpt_mos + any_peritoneal_met + pfs_cohort + factor(first_cpt_kras_braf_nras) + factor(first_cpt_apc) + factor(any_liver_mets_at_dx) + factor(any_lung_mets_at_dx),
  os_cohort %>% 
    filter(tt_os_adv_mos > adv_disease_cpt_mos)
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(adv_disease_cpt_mos = "Months from advanced dx to CPT report",
                 any_peritoneal_met = "Presence of PM")
  ) %>%
  add_nevent(location = "level") %>% 
  add_n(location = "level") %>% 
  add_global_p() %>% 
  bold_p()

tbl_multi_os_cox
```

```{r}
## look at # of patients with no first-line regimen
nrow_no_first_line_regimen <- 
  os_cohort %>%
  left_join(
    .,
    regimen_order_df %>%
      select(
        record_id,
        ca_seq,
        regimen_number,
        regimen_line_derived,
        regimen_drugs,
        regimen_group_enrique,
        dx_reg_start_int_mos,
        pfs_i_g_status,
        tt_pfs_i_g_mos,
        pfs_m_g_status,
        tt_pfs_m_g_mos,
        pfs_i_or_m_g_status,
        tt_pfs_i_or_m_g_mos,
        pfs_i_and_m_g_status,
        tt_pfs_i_and_m_g_mos
      ) %>%
      filter(regimen_line_derived == 1),
    by = c("record_id", "ca_seq")
  ) %>% 
  filter(is.na(regimen_line_derived)) %>% 
  nrow()
```

```{r, include = FALSE}
## Create PFS cohort
pfs_surv_cohort <- os_cohort %>%
  left_join(
    .,
    regimen_order_df %>%
      select(
        record_id,
        ca_seq,
        regimen_number,
        regimen_line_derived,
        regimen_drugs,
        regimen_group_enrique,
        dx_reg_start_int_mos,
        pfs_i_g_status,
        tt_pfs_i_g_mos,
        pfs_m_g_status,
        tt_pfs_m_g_mos,
        pfs_i_or_m_g_status,
        tt_pfs_i_or_m_g_mos,
        pfs_i_and_m_g_status,
        tt_pfs_i_and_m_g_mos
      ) %>%
      filter(regimen_line_derived == 1),
    by = c("record_id", "ca_seq")
  ) %>%
  filter(!is.na(regimen_line_derived), regimen_line_derived == 1) %>% 
  mutate(
    ## calculate time from regimen start to CPT report
    reg_start_cpt_rep_mos_pre = dx_cpt_rep_mos - dx_reg_start_int_mos,
    reg_start_cpt_rep_mos = case_when(
      reg_start_cpt_rep_mos_pre < 0 ~ 0,
      reg_start_cpt_rep_mos_pre >= 0 ~ reg_start_cpt_rep_mos_pre
    )
  ) %>%
  select(-reg_start_cpt_rep_mos_pre)
```

# PFS cohort

```{r}
pfs_surv_cohort_first_line_folfoxiri_cohort <- pfs_surv_cohort %>% 
  filter(
    regimen_line_derived == 1,
    regimen_drugs %in% c(
      ## FOLFOX
          "Fluorouracil, Leucovorin Calcium, Oxaliplatin",
          ## FOLFIRI
          "Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium",
          ## FOLFOXIRI
          "Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium, Oxaliplatin",
          ## FOLFOX + Bev
          "Bevacizumab, Fluorouracil, Leucovorin Calcium, Oxaliplatin",
          ## FOLFIRI + Bev
          "Bevacizumab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium",
          ## FOLFOXIRI + Bev
          "Bevacizumab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium, Oxaliplatin",
          ## FOLFOX + Cetuximab
          "Cetuximab, Fluorouracil, Leucovorin Calcium, Oxaliplatin",
          ## FOLFIRI + Cetuximab
          "Cetuximab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium",
          ## FOLFOXIRI + Cetuximab
          "Bevacizumab, Capecitabine, Oxaliplatin",
          "Bevacizumab, Capecitabine, Fluorouracil, Leucovorin Calcium, Oxaliplatin",
          "Capecitabine, Oxaliplatin",
          "Capecitabine, Fluorouracil, Leucovorin Calcium, Oxaliplatin",
          "Cetuximab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium, Oxaliplatin",
          "Cetuximab, Fluorouracil, Leucovorin Calcium, Oxaliplatin",
          "Cetuximab, Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium",
          "Fluorouracil, Irinotecan Hydrochloride",
          "Fluorouracil, Irinotecan Hydrochloride, Leucovorin Calcium, Panitumumab",
          "Fluorouracil, Leucovorin Calcium, Oxaliplatin, Panitumumab"
    )
  ) 
```

```{r}
supplemental_tbl8 <- pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>% 
  select(regimen_drugs) %>% 
  tbl_summary(label = list(regimen_drugs = "Drug regimen"),
              sort = everything() ~ "frequency")

supplemental_tbl8
```

# PFS from first-line combination approved therapy by presence/absence of PM

```{r}
tbl_pfs_n_folfoxiri_full_cohort <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>% 
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "PFS-I-and-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_folfoxiri_full_cohort <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_and_m_g_status == 1) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "PFS-I-and-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_med_pfs_folfoxiri_full_cohort <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ any_peritoneal_met,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "PFS-I-and-M: PM at advanced dx"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_folfoxiri_full_cohort <-
  summary(
    coxph(
      Surv(
        time = reg_start_cpt_rep_mos,
        time2 = tt_pfs_i_and_m_g_mos,
        event = pfs_i_and_m_g_status
      ) ~ any_peritoneal_met,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
      timefix = FALSE
    )
  )

logrank_pval_pfs_folfoxiri_full_cohort <- left_trunc_pfs_folfoxiri_full_cohort$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

med_pfs_i_and_m_g_any_pm <- tbl_merge(
   list(
     tbl_pfs_n_folfoxiri_full_cohort,
     tbl_pfs_n_event_folfoxiri_full_cohort,
     tbl_med_pfs_folfoxiri_full_cohort
   )
 ) %>%
   modify_spanning_header(everything() ~ NA_character_) 

med_pfs_i_and_m_g_any_pm %>% 
    modify_table_styling(
        columns = label,
        rows = label == "PFS-I-and-M: PM at advanced dx",
        footnote = paste0("Left truncation adjusted logrank p-value: ", round(logrank_pval_pfs_folfoxiri_full_cohort, 3))
    )
```

```{r}
figure_3a <- survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ any_peritoneal_met,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  coord_cartesian(xlim = c(0, 72)) + 
  scale_x_continuous(breaks = seq(0, 72, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B", "#9E9E98")) +
  labs(x = "Time from start of 1st line regimen (months)",
       y = "Probability of PFS",
       title = "A. PFS from diagnosis of advanced disease: PM vs. no PM")

figure_3a
```

# Sensitivity analyses: consider other PFS definitions

```{r}
## PFS-I
tbl_pfs_i_n_folfoxiri_full_cohort <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_g_mos > reg_start_cpt_rep_mos) %>% 
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "PFS-I: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_i_n_event_folfoxiri_full_cohort <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_g_status == 1) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "PFS-I: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_med_pfs_i_folfoxiri_full_cohort <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_g_mos,
    event = pfs_i_g_status
  ) ~ any_peritoneal_met,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
    filter(tt_pfs_i_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "PFS-I: PM at advanced dx"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

med_pfs_i_g_any_pm <- tbl_merge(
   list(
     tbl_pfs_i_n_folfoxiri_full_cohort,
     tbl_pfs_i_n_event_folfoxiri_full_cohort,
     tbl_med_pfs_i_folfoxiri_full_cohort
   )
 ) %>%
   modify_spanning_header(everything() ~ NA_character_) 
```

```{r}
## PFS-M
tbl_pfs_m_n_folfoxiri_full_cohort <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_m_g_mos > reg_start_cpt_rep_mos) %>% 
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "PFS-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_m_n_event_folfoxiri_full_cohort <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_m_g_status == 1) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "PFS-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_med_pfs_m_folfoxiri_full_cohort <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_m_g_mos,
    event = pfs_m_g_status
  ) ~ any_peritoneal_met,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
    filter(tt_pfs_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "PFS-M: PM at advanced dx"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

med_pfs_m_g_any_pm <- tbl_merge(
   list(
     tbl_pfs_m_n_folfoxiri_full_cohort,
     tbl_pfs_m_n_event_folfoxiri_full_cohort,
     tbl_med_pfs_m_folfoxiri_full_cohort
   )
 ) %>%
   modify_spanning_header(everything() ~ NA_character_) 
```

```{r}
## PFS-I-or-M
tbl_pfs_i_or_m_n_folfoxiri_full_cohort <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_or_m_g_mos > reg_start_cpt_rep_mos) %>% 
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "PFS-I-or-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_i_or_m_n_event_folfoxiri_full_cohort <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_or_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_or_m_g_status == 1) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "PFS-I-or-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_med_pfs_i_or_m_folfoxiri_full_cohort <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_or_m_g_mos,
    event = pfs_i_or_m_g_status
  ) ~ any_peritoneal_met,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
    filter(tt_pfs_i_or_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "PFS-I-or-M: PM at advanced dx"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

med_pfs_i_or_m_g_any_pm <- tbl_merge(
   list(
     tbl_pfs_i_or_m_n_folfoxiri_full_cohort,
     tbl_pfs_i_or_m_n_event_folfoxiri_full_cohort,
     tbl_med_pfs_i_or_m_folfoxiri_full_cohort
   )
 ) %>%
   modify_spanning_header(everything() ~ NA_character_) 
```

```{r}
## stack tbl_merge
supplemental_tbl9 <- tbl_stack(
  list(
    # med_pfs_i_and_m_g_any_pm,
    med_pfs_i_g_any_pm,
    med_pfs_m_g_any_pm,
    med_pfs_i_or_m_g_any_pm
  )
)

supplemental_tbl9
```

## PFS-I-and-M

```{r}
survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ any_peritoneal_met,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  scale_x_continuous(limits = c(0, 120),
                     breaks = seq(0, 120, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B", "#9E9E98")) +
  labs(x = "Time from initiation of 1st line regimen (months)",
       y = "Probability of PFS",
       title = "PFS-I-and-M from 1st-line approved combination therapy")
```

## PFS-I

```{r}
survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_g_mos,
    event = pfs_i_g_status
  ) ~ any_peritoneal_met,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  scale_x_continuous(limits = c(0, 120),
                     breaks = seq(0, 120, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B", "#9E9E98")) +
  labs(x = "Time from initiation of 1st line regimen (months)",
       y = "Probability of PFS",
       title = "PFS-I from 1st-line approved combination therapy")
```

## PFS-M

```{r}
survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_m_g_mos,
    event = pfs_m_g_status
  ) ~ any_peritoneal_met,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  coord_cartesian(xlim = c(0, 72)) +
  scale_x_continuous(breaks = seq(0, 72, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B", "#9E9E98")) +
  labs(x = "Time from initiation of 1st line regimen (months)",
       y = "Probability of PFS",
       title = "PFS-M from 1st-line approved combination therapy")
```

## PFS-I-or-M

```{r}
survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_or_m_g_mos,
    event = pfs_i_or_m_g_status
  ) ~ any_peritoneal_met,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_or_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  coord_cartesian(xlim = c(0, 72)) +
  scale_x_continuous(breaks = seq(0, 72, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B", "#9E9E98")) +
  labs(x = "Time from initiation of 1st line regimen (months)",
       y = "Probability of PFS",
       title = "PFS-I-or-M from 1st-line approved combination therapy")
```

# PFS from first-line combination approved therapy by presence/absence of PM: stage IV only

```{r}
tbl_pfs_n_folfoxiri_full_cohort_stg_iv <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_cohort == "Stage IV") %>% 
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "PFS-I-and-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_folfoxiri_full_cohort_stg_iv <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_and_m_g_status == 1,
         pfs_cohort == "Stage IV") %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "PFS-I-and-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_med_pfs_folfoxiri_full_cohort_stg_iv <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ any_peritoneal_met,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           pfs_cohort == "Stage IV"),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "PFS-I-and-M: PM at advanced dx"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_folfoxiri_full_cohort_stg_iv <-
  summary(
    coxph(
      Surv(
        time = reg_start_cpt_rep_mos,
        time2 = tt_pfs_i_and_m_g_mos,
        event = pfs_i_and_m_g_status
      ) ~ any_peritoneal_met,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
               pfs_cohort == "Stage IV"),
      timefix = FALSE
    )
  )

logrank_pval_pfs_folfoxiri_full_cohort_stg_iv <- left_trunc_pfs_folfoxiri_full_cohort_stg_iv$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

med_pfs_i_and_m_g_any_pm_stg_iv <- tbl_merge(
   list(
     tbl_pfs_n_folfoxiri_full_cohort_stg_iv,
     tbl_pfs_n_event_folfoxiri_full_cohort_stg_iv,
     tbl_med_pfs_folfoxiri_full_cohort_stg_iv
   )
 ) %>%
   modify_spanning_header(everything() ~ NA_character_) 

med_pfs_i_and_m_g_any_pm_stg_iv %>% 
    modify_table_styling(
        columns = label,
        rows = label == "PFS-I-and-M: PM at advanced dx",
        footnote = paste0("Left truncation adjusted logrank p-value: ", round(logrank_pval_pfs_folfoxiri_full_cohort_stg_iv, 3))
    )
```

```{r}
figure_3c <- survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ any_peritoneal_met,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           pfs_cohort == "Stage IV"),
  timefix = FALSE
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  coord_cartesian(xlim = c(0, 72)) + 
  scale_x_continuous(breaks = seq(0, 72, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B", "#9E9E98")) +
  labs(x = "Time from start of 1st line regimen (months)",
       y = "Probability of PFS",
       title = "PFS from stage IV diagnosis: PM vs. No PM")

figure_3c
```

# PFS from first-line combination approved therapy by presence/absence of PM: stage I-III with distant mets 

```{r}
tbl_pfs_n_folfoxiri_full_cohort_stg_i_iii <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_cohort == "Stage I-III with Distant Mets") %>% 
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "PFS-I-and-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_folfoxiri_full_cohort_stg_i_iii <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_and_m_g_status == 1,
         pfs_cohort == "Stage I-III with Distant Mets") %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "PFS-I-and-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_med_pfs_folfoxiri_full_cohort_stg_i_iii <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ any_peritoneal_met,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           pfs_cohort == "Stage I-III with Distant Mets"),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "PFS-I-and-M: PM at advanced dx"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_folfoxiri_full_cohort_stg_i_iii <-
  summary(
    coxph(
      Surv(
        time = reg_start_cpt_rep_mos,
        time2 = tt_pfs_i_and_m_g_mos,
        event = pfs_i_and_m_g_status
      ) ~ any_peritoneal_met,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
               pfs_cohort == "Stage I-III with Distant Mets"),
      timefix = FALSE
    )
  )

logrank_pval_pfs_folfoxiri_full_cohort_stg_i_iii <- left_trunc_pfs_folfoxiri_full_cohort_stg_i_iii$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

med_pfs_i_and_m_g_any_pm_stg_i_iii <- tbl_merge(
   list(
     tbl_pfs_n_folfoxiri_full_cohort_stg_i_iii,
     tbl_pfs_n_event_folfoxiri_full_cohort_stg_i_iii,
     tbl_med_pfs_folfoxiri_full_cohort_stg_i_iii
   )
 ) %>%
   modify_spanning_header(everything() ~ NA_character_) 

med_pfs_i_and_m_g_any_pm_stg_i_iii %>% 
    modify_table_styling(
        columns = label,
        rows = label == "PFS-I-and-M: PM at advanced dx",
        footnote = paste0("Left truncation adjusted logrank p-value: ", round(logrank_pval_pfs_folfoxiri_full_cohort_stg_i_iii, 3))
    )
```

```{r}
figure_3d <- survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ any_peritoneal_met,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           pfs_cohort == "Stage I-III with Distant Mets"),
  timefix = FALSE
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  coord_cartesian(xlim = c(0, 72)) + 
  scale_x_continuous(breaks = seq(0, 72, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B", "#9E9E98")) +
  labs(x = "Time from start of 1st line regimen (months)",
       y = "Probability of PFS",
       title = "PFS from diagnosis of advanced disease among Stage I-III with distant metastasis: PM vs. No PM")

figure_3d
```

# PFS from first-line combination approved therapy by PM only/PM & others/no PM

```{r}
tbl_pfs_n_folfoxiri_full_cohort_pm_others <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "PFS-I-and-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_folfoxiri_full_cohort_pm_others <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1) %>%
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "PFS-I-and-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_med_pfs_folfoxiri_full_cohort_pm_others <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ pm_and_other_mets,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(pm_and_other_mets = "PFS-I-and-M: PM at advanced dx"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_folfoxiri_full_cohort_pm_others <-
  summary(
    coxph(
      Surv(
        time = reg_start_cpt_rep_mos,
        time2 = tt_pfs_i_and_m_g_mos,
        event = pfs_i_and_m_g_status
      ) ~ pm_and_other_mets,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
      timefix = FALSE
    )
  )

logrank_pval_pfs_folfoxiri_full_cohort_pm_others <-
  left_trunc_pfs_folfoxiri_full_cohort_pm_others$sctest %>%
  as.data.frame() %>%
  rownames_to_column(var = "rowname") %>%
  filter(rowname == "pvalue") %>%
  select(".") %>%
  unlist()

med_pfs_i_and_m_g_pm_others <- tbl_merge(
  list(
    tbl_pfs_n_folfoxiri_full_cohort_pm_others,
    tbl_pfs_n_event_folfoxiri_full_cohort_pm_others,
    tbl_med_pfs_folfoxiri_full_cohort_pm_others
  )
) %>%
  modify_spanning_header(everything() ~ NA_character_) 

med_pfs_i_and_m_g_pm_others %>%
  modify_table_styling(
    columns = label,
    rows = label == "PFS-I-and-M: PM at advanced dx",
    footnote = paste0(
      "Left truncation adjusted logrank p-value: ",
      round(logrank_pval_pfs_folfoxiri_full_cohort_pm_others, 3)
    )
  )
```

```{r}
figure_3b <- survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ pm_and_other_mets,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  coord_cartesian(xlim = c(0, 72)) +
  scale_x_continuous(breaks = seq(0, 72, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B", "#9E9E98")) +
  labs(x = "Time from start of 1st line regimen (months)",
       y = "Probability of PFS",
       title = "PFS from diagnosis of advanced disease: PM only vs. PM and others vs. No PM")

figure_3b
```

# Sensitivity analyses: consider other PFS definitions

```{r}
## PFS-I
tbl_pfs_i_n_folfoxiri_full_cohort_pm_others <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_g_mos > reg_start_cpt_rep_mos) %>% 
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "PFS-I: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_i_n_event_folfoxiri_full_cohort_pm_others <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_g_status == 1) %>%
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "PFS-I: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_med_pfs_i_folfoxiri_full_cohort_pm_others <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_g_mos,
    event = pfs_i_g_status
  ) ~ pm_and_other_mets,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
    filter(tt_pfs_i_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(pm_and_other_mets = "PFS-I: PM at advanced dx"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

med_pfs_i_g_pm_others <- tbl_merge(
   list(
     tbl_pfs_i_n_folfoxiri_full_cohort_pm_others,
     tbl_pfs_i_n_event_folfoxiri_full_cohort_pm_others,
     tbl_med_pfs_i_folfoxiri_full_cohort_pm_others
   )
 ) %>%
   modify_spanning_header(everything() ~ NA_character_) 
```

```{r}
## PFS-M
tbl_pfs_m_n_folfoxiri_full_cohort_pm_others <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_m_g_mos > reg_start_cpt_rep_mos) %>% 
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "PFS-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_m_n_event_folfoxiri_full_cohort_pm_others <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_m_g_status == 1) %>%
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "PFS-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_med_pfs_m_folfoxiri_full_cohort_pm_others <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_m_g_mos,
    event = pfs_m_g_status
  ) ~ pm_and_other_mets,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
    filter(tt_pfs_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(pm_and_other_mets = "PFS-M: PM at advanced dx"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

med_pfs_m_g_pm_others <- tbl_merge(
   list(
     tbl_pfs_m_n_folfoxiri_full_cohort_pm_others,
     tbl_pfs_m_n_event_folfoxiri_full_cohort_pm_others,
     tbl_med_pfs_m_folfoxiri_full_cohort_pm_others
   )
 ) %>%
   modify_spanning_header(everything() ~ NA_character_) 
```

```{r}
## PFS-I-or-M
tbl_pfs_i_or_m_n_folfoxiri_full_cohort_pm_others <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_or_m_g_mos > reg_start_cpt_rep_mos) %>% 
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "PFS-I-or-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_i_or_m_n_event_folfoxiri_full_cohort_pm_others <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_or_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_or_m_g_status == 1) %>%
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "PFS-I-or-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_med_pfs_i_or_m_folfoxiri_full_cohort_pm_others <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_or_m_g_mos,
    event = pfs_i_or_m_g_status
  ) ~ pm_and_other_mets,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
    filter(tt_pfs_i_or_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(pm_and_other_mets = "PFS-I-or-M: PM at advanced dx"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

med_pfs_i_or_m_g_pm_others <- tbl_merge(
   list(
     tbl_pfs_i_or_m_n_folfoxiri_full_cohort_pm_others,
     tbl_pfs_i_or_m_n_event_folfoxiri_full_cohort_pm_others,
     tbl_med_pfs_i_or_m_folfoxiri_full_cohort_pm_others
   )
 ) %>%
   modify_spanning_header(everything() ~ NA_character_) 
```

```{r}
## stack tbl_merge
tbl_stack(
  list(
    med_pfs_i_and_m_g_pm_others,
    med_pfs_i_g_pm_others,
    med_pfs_m_g_pm_others,
    med_pfs_i_or_m_g_pm_others
  )
)
```

## PFS-I-and-M

```{r}
survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ pm_and_other_mets,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  coord_cartesian(xlim = c(0, 72)) +
  scale_x_continuous(breaks = seq(0, 72, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B", "#9E9E98")) +
  labs(x = "Time from initiation of 1st line regimen (months)",
       y = "Probability of PFS",
       title = "PFS-I-and-M from 1st-line approved combination therapy")
```

## PFS-I

```{r}
survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_g_mos,
    event = pfs_i_g_status
  ) ~ pm_and_other_mets,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  coord_cartesian(xlim = c(0, 36)) +
  scale_x_continuous(breaks = seq(0, 36, by = 3)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B", "#9E9E98")) +
  labs(x = "Time from initiation of 1st line regimen (months)",
       y = "Probability of PFS",
       title = "PFS-I from 1st-line approved combination therapy")
```

## PFS-M

```{r}
survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_m_g_mos,
    event = pfs_m_g_status
  ) ~ pm_and_other_mets,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  coord_cartesian(xlim = c(0, 72)) +
  scale_x_continuous(breaks = seq(0, 72, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B", "#9E9E98")) +
  labs(x = "Time from initiation of 1st line regimen (months)",
       y = "Probability of PFS",
       title = "PFS-M from 1st-line approved combination therapy")
```

## PFS-I-or-M

```{r}
survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_or_m_g_mos,
    event = pfs_i_or_m_g_status
  ) ~ pm_and_other_mets,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_or_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  coord_cartesian(xlim = c(0, 72)) +
  scale_x_continuous(breaks = seq(0, 72, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B", "#9E9E98")) +
  labs(x = "Time from initiation of 1st line regimen (months)",
       y = "Probability of PFS",
       title = "PFS-I-or-M from 1st-line approved combination therapy")
```

# PFS from first-line combination approved therapy by PM only/PM & others/no PM: stage IV only

```{r}
tbl_pfs_n_folfoxiri_full_cohort_pm_others_stg_iv <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_cohort == "Stage IV") %>%
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "PFS-I-and-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_folfoxiri_full_cohort_pm_others_stg_iv <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         pfs_cohort == "Stage IV") %>%
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "PFS-I-and-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_med_pfs_folfoxiri_full_cohort_pm_others_stg_iv <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ pm_and_other_mets,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           pfs_cohort == "Stage IV"),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(pm_and_other_mets = "PFS-I-and-M: PM at advanced dx"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_folfoxiri_full_cohort_pm_others_stg_iv <-
  summary(
    coxph(
      Surv(
        time = reg_start_cpt_rep_mos,
        time2 = tt_pfs_i_and_m_g_mos,
        event = pfs_i_and_m_g_status
      ) ~ pm_and_other_mets,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
               pfs_cohort == "Stage IV"),
      timefix = FALSE
    )
  )

logrank_pval_pfs_folfoxiri_full_cohort_pm_others_stg_iv <-
  left_trunc_pfs_folfoxiri_full_cohort_pm_others_stg_iv$sctest %>%
  as.data.frame() %>%
  rownames_to_column(var = "rowname") %>%
  filter(rowname == "pvalue") %>%
  select(".") %>%
  unlist()

med_pfs_i_and_m_g_pm_others_stg_iv <- tbl_merge(
  list(
    tbl_pfs_n_folfoxiri_full_cohort_pm_others_stg_iv,
    tbl_pfs_n_event_folfoxiri_full_cohort_pm_others_stg_iv,
    tbl_med_pfs_folfoxiri_full_cohort_pm_others_stg_iv
  )
) %>%
  modify_spanning_header(everything() ~ NA_character_) 

med_pfs_i_and_m_g_pm_others_stg_iv %>%
  modify_table_styling(
    columns = label,
    rows = label == "PFS-I-and-M: PM at advanced dx",
    footnote = paste0(
      "Left truncation adjusted logrank p-value: ",
      round(logrank_pval_pfs_folfoxiri_full_cohort_pm_others_stg_iv, 3)
    )
  )
```

```{r}
survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ pm_and_other_mets,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           pfs_cohort == "Stage IV"),
  timefix = FALSE
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  coord_cartesian(xlim = c(0, 72)) +
  scale_x_continuous(breaks = seq(0, 72, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B", "#9E9E98")) +
  labs(x = "Time from initiation of 1st line regimen (months)",
       y = "Probability of PFS",
       title = "PFS-I-and-M from 1st-line approved combination therapy (stage IV)")
```

# PFS from first-line combination approved therapy by PM only/PM & others/no PM: stage I-III with distant mets

```{r}
tbl_pfs_n_folfoxiri_full_cohort_pm_others_stg_i_iii <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_cohort == "Stage I-III with Distant Mets") %>%
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "PFS-I-and-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_folfoxiri_full_cohort_pm_others_stg_i_iii <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         pfs_cohort == "Stage I-III with Distant Mets") %>%
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "PFS-I-and-M: PM at advanced dx"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_med_pfs_folfoxiri_full_cohort_pm_others_stg_i_iii <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ pm_and_other_mets,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           pfs_cohort == "Stage I-III with Distant Mets"),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(pm_and_other_mets = "PFS-I-and-M: PM at advanced dx"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_folfoxiri_full_cohort_pm_others_stg_i_iii <-
  summary(
    coxph(
      Surv(
        time = reg_start_cpt_rep_mos,
        time2 = tt_pfs_i_and_m_g_mos,
        event = pfs_i_and_m_g_status
      ) ~ pm_and_other_mets,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
               pfs_cohort == "Stage I-III with Distant Mets"),
      timefix = FALSE
    )
  )

logrank_pval_pfs_folfoxiri_full_cohort_pm_others_stg_i_iii <-
  left_trunc_pfs_folfoxiri_full_cohort_pm_others_stg_i_iii$sctest %>%
  as.data.frame() %>%
  rownames_to_column(var = "rowname") %>%
  filter(rowname == "pvalue") %>%
  select(".") %>%
  unlist()

med_pfs_i_and_m_g_pm_others_stg_i_iii <- tbl_merge(
  list(
    tbl_pfs_n_folfoxiri_full_cohort_pm_others_stg_i_iii,
    tbl_pfs_n_event_folfoxiri_full_cohort_pm_others_stg_i_iii,
    tbl_med_pfs_folfoxiri_full_cohort_pm_others_stg_i_iii
  )
) %>%
  modify_spanning_header(everything() ~ NA_character_) 

med_pfs_i_and_m_g_pm_others_stg_i_iii %>%
  modify_table_styling(
    columns = label,
    rows = label == "PFS-I-and-M: PM at advanced dx",
    footnote = paste0(
      "Left truncation adjusted logrank p-value: ",
      round(logrank_pval_pfs_folfoxiri_full_cohort_pm_others_stg_i_iii, 3)
    )
  )
```

```{r}
survfit2(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ pm_and_other_mets,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           pfs_cohort == "Stage I-III with Distant Mets"),
  timefix = FALSE
) %>%
  ggsurvfit(theme = list(theme_classic(),
                         theme(legend.position = "top"))) +
  add_censor_mark() +
  add_risktable(risktable_stats = c("n.risk")) +
  scale_y_continuous(limits = c(0, 1)) +
  coord_cartesian(xlim = c(0, 72)) +
  scale_x_continuous(breaks = seq(0, 72, by = 12)) +
  scale_color_manual(values = c("#007CBA", "#DF4602", "#83276B", "#9E9E98")) +
  labs(x = "Time from initiation of 1st line regimen (months)",
       y = "Probability of PFS",
       title = "PFS-I-and-M from 1st-line approved combination therapy (stage I-III with distant mets)")
```

# PFS by RAS/BRAFmut pts

```{r}
tbl_n_kras_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(first_cpt_kras== 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "KRASmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_kras_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(pfs_i_and_m_g_status == 1,
         first_cpt_kras== 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "KRASmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_kras_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ any_peritoneal_met ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(first_cpt_kras== 1,
           tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "KRASmut: Presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_kras_any_pm <-
  summary(
    coxph(
      Surv(
        time = adv_disease_cpt_mos,
        time2 = tt_os_adv_mos,
        event = os_adv_status
      ) ~ any_peritoneal_met,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(
          first_cpt_kras == 1,
          tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos
        )
    )
  )

logrank_pval_pfs_kras_any_pm <- left_trunc_pfs_kras_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_kras_any_pm <-
  tbl_merge(list(
    tbl_n_kras_any_pm_pfs,
    tbl_n_event_kras_any_pm_pfs,
    tbl_pfs_med_kras_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "KRASmut: Presence of PM",
    footnote = paste0("KRASmut p-value: ", round(logrank_pval_pfs_kras_any_pm, 3))
  )
```

```{r}
tbl_n_braf_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(first_cpt_braf== 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "BRAFmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_braf_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(pfs_i_and_m_g_status == 1,
         first_cpt_braf== 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "BRAFmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_braf_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ any_peritoneal_met ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(first_cpt_braf== 1,
           tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "BRAFmut: Presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_braf_any_pm <-
  summary(
    coxph(
      Surv(
        time = adv_disease_cpt_mos,
        time2 = tt_os_adv_mos,
        event = os_adv_status
      ) ~ any_peritoneal_met,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(
          first_cpt_braf == 1,
          tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos
        )
    )
  )

logrank_pval_pfs_braf_any_pm <- left_trunc_pfs_braf_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_braf_any_pm <-
  tbl_merge(list(
    tbl_n_braf_any_pm_pfs,
    tbl_n_event_braf_any_pm_pfs,
    tbl_pfs_med_braf_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "BRAFmut: Presence of PM",
    footnote = paste0("BRAFmut p-value: ", round(logrank_pval_pfs_braf_any_pm, 3))
  )
```

```{r}
tbl_n_nras_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(first_cpt_nras== 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "NRASmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_nras_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(pfs_i_and_m_g_status == 1,
         first_cpt_nras== 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "NRASmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_nras_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ any_peritoneal_met ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(first_cpt_nras== 1,
           tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "NRASmut: Presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_nras_any_pm <-
  summary(
    coxph(
      Surv(
        time = adv_disease_cpt_mos,
        time2 = tt_os_adv_mos,
        event = os_adv_status
      ) ~ any_peritoneal_met,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(
          first_cpt_nras == 1,
          tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos
        )
    )
  )

logrank_pval_pfs_nras_any_pm <- left_trunc_pfs_nras_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_nras_any_pm <-
  tbl_merge(list(
    tbl_n_nras_any_pm_pfs,
    tbl_n_event_nras_any_pm_pfs,
    tbl_pfs_med_nras_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)
```

```{r}
tbl_n_kras_braf_nras_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(first_cpt_kras_braf_nras== 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "RAS/BRAFmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_kras_braf_nras_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(pfs_i_and_m_g_status == 1,
         first_cpt_kras_braf_nras== 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "RAS/BRAFmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_kras_braf_nras_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ any_peritoneal_met ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(first_cpt_kras_braf_nras== 1,
           tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "RAS/BRAFmut: Presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_kras_braf_nras_any_pm <-
  summary(
    coxph(
      Surv(
        time = adv_disease_cpt_mos,
        time2 = tt_os_adv_mos,
        event = os_adv_status
      ) ~ any_peritoneal_met,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(
          first_cpt_kras_braf_nras == 1,
          tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos
        )
    )
  )

logrank_pval_pfs_kras_braf_nras_any_pm <- left_trunc_pfs_kras_braf_nras_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_kras_braf_nras_any_pm <-
  tbl_merge(list(
    tbl_n_kras_braf_nras_any_pm_pfs,
    tbl_n_event_kras_braf_nras_any_pm_pfs,
    tbl_pfs_med_kras_braf_nras_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "RAS/BRAFmut: Presence of PM",
    footnote = paste0("RAS/BRAFmut p-value: ", round(logrank_pval_pfs_kras_braf_nras_any_pm, 3))
  )
```

```{r}
tbl_n_pik3ca_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(first_cpt_pik3ca== 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "PIK3CAmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_pik3ca_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(pfs_i_and_m_g_status == 1,
         first_cpt_pik3ca== 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "PIK3CAmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_pik3ca_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ any_peritoneal_met ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(first_cpt_pik3ca== 1,
           tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "PIK3CAmut: Presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_pik3ca_any_pm <-
  summary(
    coxph(
      Surv(
        time = adv_disease_cpt_mos,
        time2 = tt_os_adv_mos,
        event = os_adv_status
      ) ~ any_peritoneal_met,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(
          first_cpt_pik3ca == 1,
          tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos
        )
    )
  )

logrank_pval_pfs_pik3ca_any_pm <- left_trunc_pfs_pik3ca_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_pik3ca_any_pm <-
  tbl_merge(list(
    tbl_n_pik3ca_any_pm_pfs,
    tbl_n_event_pik3ca_any_pm_pfs,
    tbl_pfs_med_pik3ca_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "PIK3CAmut: Presence of PM",
    footnote = paste0("PIK3CAmut p-value: ", round(logrank_pval_pfs_pik3ca_any_pm, 3))
  )
```

```{r}
tbl_n_apc_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(first_cpt_apc== 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "APCmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_apc_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(pfs_i_and_m_g_status == 1,
         first_cpt_apc== 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "APCmut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_apc_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ any_peritoneal_met ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(first_cpt_apc== 1,
           tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "APCmut: Presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_apc_any_pm <-
  summary(
    coxph(
      Surv(
        time = adv_disease_cpt_mos,
        time2 = tt_os_adv_mos,
        event = os_adv_status
      ) ~ any_peritoneal_met,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(
          first_cpt_apc == 1,
          tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos
        )
    )
  )

logrank_pval_pfs_apc_any_pm <- left_trunc_pfs_apc_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_apc_any_pm <-
  tbl_merge(list(
    tbl_n_apc_any_pm_pfs,
    tbl_n_event_apc_any_pm_pfs,
    tbl_pfs_med_apc_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "APCmut: Presence of PM",
    footnote = paste0("APCmut p-value: ", round(logrank_pval_pfs_apc_any_pm, 3))
  )
```

```{r}
tbl_n_tp53_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(first_cpt_tp53== 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "TP53mut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_tp53_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(pfs_i_and_m_g_status == 1,
         first_cpt_tp53== 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    type = list(any_peritoneal_met ~ "categorical"),
    label = list(any_peritoneal_met = "TP53mut: Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_tp53_any_pm <- survfit(
  Surv(
    time = adv_disease_cpt_mos,
    time2 = tt_os_adv_mos,
    event = os_adv_status
  ) ~ any_peritoneal_met ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(first_cpt_tp53== 1,
           tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos)
) %>%
  tbl_survfit(
    probs = .5,
    label = list(any_peritoneal_met = "TP53mut: Presence of PM"),
    label_header = "**Median overall survival (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_tp53_any_pm <-
  summary(
    coxph(
      Surv(
        time = adv_disease_cpt_mos,
        time2 = tt_os_adv_mos,
        event = os_adv_status
      ) ~ any_peritoneal_met,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(
          first_cpt_tp53 == 1,
          tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos
        )
    )
  )

logrank_pval_pfs_tp53_any_pm <- left_trunc_pfs_tp53_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_tp53_any_pm <-
  tbl_merge(list(
    tbl_n_tp53_any_pm_pfs,
    tbl_n_event_tp53_any_pm_pfs,
    tbl_pfs_med_tp53_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "TP53mut: Presence of PM",
    footnote = paste0("TP53mut p-value: ", round(logrank_pval_pfs_tp53_any_pm, 3))
  )
```

```{r}
## stack tbl_merge
tbl_pfs_genomics <- tbl_stack(
  list(
    tbl_merge_pfs_med_kras_any_pm,
    tbl_merge_pfs_med_braf_any_pm,
    tbl_merge_pfs_med_kras_braf_nras_any_pm,
    tbl_merge_pfs_med_pik3ca_any_pm,
    tbl_merge_pfs_med_apc_any_pm,
    tbl_merge_pfs_med_tp53_any_pm
  )
)

tbl_pfs_genomics
```

# PFS from first-line combination approved therapy by TP53mut vs TP53wt

```{r}
tbl_pfs_n_tp53 <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>% 
  select(first_cpt_tp53) %>%
  tbl_summary(
    type = list(first_cpt_tp53 ~ "categorical"),
    label = list(first_cpt_tp53 = "Full PFS cohort: TP53mut vs. TP53wt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_tp53 <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1) %>%
  select(first_cpt_tp53) %>%
  tbl_summary(
    type = list(first_cpt_tp53 ~ "categorical"),
    label = list(first_cpt_tp53 = "Full PFS cohort: TP53mut vs. TP53wt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_tp53 <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_tp53 ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_tp53 = "Full PFS cohort: TP53mut vs. TP53wt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_tp53 <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_tp53 ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
    timefix = FALSE
  ))

logrank_pval_pfs_tp53 <- left_trunc_pfs_tp53$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_tp53 <-
  tbl_merge(list(tbl_pfs_n_tp53,
                 tbl_pfs_n_event_tp53,
                 tbl_pfs_med_tp53)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "Full PFS cohort: TP53mut vs. TP53wt",
        footnote = paste0("Full PFS cohort p-value: ", round(logrank_pval_pfs_tp53, 3))
    )
```

```{r}
tbl_pfs_n_tp53_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "PM") %>% 
  select(first_cpt_tp53) %>%
  tbl_summary(
    type = list(first_cpt_tp53 ~ "categorical"),
    label = list(first_cpt_tp53 = "PM: TP53mut vs. TP53wt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_tp53_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         any_peritoneal_met == "PM") %>%
  select(first_cpt_tp53) %>%
  tbl_summary(
    type = list(first_cpt_tp53 ~ "categorical"),
    label = list(first_cpt_tp53 = "PM: TP53mut vs. TP53wt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_tp53_pm <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_tp53 ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "PM")) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_tp53 = "PM: TP53mut vs. TP53wt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_tp53_pm <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_tp53 ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             any_peritoneal_met == "PM")
  ))

logrank_pval_pfs_tp53_pm <- left_trunc_pfs_tp53_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_tp53_pm <-
  tbl_merge(list(tbl_pfs_n_tp53_pm,
                 tbl_pfs_n_event_tp53_pm,
                 tbl_pfs_med_tp53_pm)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "PM: TP53mut vs. TP53wt",
        footnote = paste0("PM p-value: ", round(logrank_pval_pfs_tp53_pm, 3))
    )
```

```{r}
tbl_pfs_n_tp53_pm_only <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM only") %>% 
  select(first_cpt_tp53) %>%
  tbl_summary(
    type = list(first_cpt_tp53 ~ "categorical"),
    label = list(first_cpt_tp53 = "PM only: TP53mut vs. TP53wt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_tp53_pm_only <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         pm_and_other_mets == "PM only") %>%
  select(first_cpt_tp53) %>%
  tbl_summary(
    type = list(first_cpt_tp53 ~ "categorical"),
    label = list(first_cpt_tp53 = "PM only: TP53mut vs. TP53wt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_tp53_pm_only <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_tp53 ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM only")) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_tp53 = "PM only: TP53mut vs. TP53wt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_tp53_pm_only <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_tp53 ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             pm_and_other_mets == "PM only")
  ))

logrank_pval_pfs_tp53_pm_only <- left_trunc_pfs_tp53_pm_only$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_tp53_pm_only <-
  tbl_merge(list(tbl_pfs_n_tp53_pm_only,
                 tbl_pfs_n_event_tp53_pm_only,
                 tbl_pfs_med_tp53_pm_only)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "PM only: TP53mut vs. TP53wt",
        footnote = paste0("PM only p-value: ", round(logrank_pval_pfs_tp53_pm_only, 3))
    )
```

```{r}
tbl_pfs_n_tp53_pm_others <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM and others") %>% 
  select(first_cpt_tp53) %>%
  tbl_summary(
    type = list(first_cpt_tp53 ~ "categorical"),
    label = list(first_cpt_tp53 = "PM and others: TP53mut vs. TP53wt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_tp53_pm_others <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         pm_and_other_mets == "PM and others") %>%
  select(first_cpt_tp53) %>%
  tbl_summary(
    type = list(first_cpt_tp53 ~ "categorical"),
    label = list(first_cpt_tp53 = "PM and others: TP53mut vs. TP53wt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_tp53_pm_others <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_tp53 ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM and others")) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_tp53 = "PM and others: TP53mut vs. TP53wt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_tp53_pm_others <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_tp53 ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             pm_and_other_mets == "PM and others")
  ))

logrank_pval_pfs_tp53_pm_others <- left_trunc_pfs_tp53_pm_others$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_tp53_pm_others <-
  tbl_merge(list(tbl_pfs_n_tp53_pm_others,
                 tbl_pfs_n_event_tp53_pm_others,
                 tbl_pfs_med_tp53_pm_others)) %>%
  modify_spanning_header(everything() ~ NA_character_)
```

```{r}
tbl_pfs_n_tp53_no_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "No PM") %>% 
  select(first_cpt_tp53) %>%
  tbl_summary(
    type = list(first_cpt_tp53 ~ "categorical"),
    label = list(first_cpt_tp53 = "No PM: TP53mut vs. TP53wt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_tp53_no_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         any_peritoneal_met == "No PM") %>%
  select(first_cpt_tp53) %>%
  tbl_summary(
    type = list(first_cpt_tp53 ~ "categorical"),
    label = list(first_cpt_tp53 = "No PM: TP53mut vs. TP53wt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_tp53_no_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_tp53 ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(
      tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
      any_peritoneal_met == "No PM"
    ),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_tp53 = "No PM: TP53mut vs. TP53wt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_tp53_no_pm <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_tp53 ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             any_peritoneal_met == "No PM"),
    timefix = FALSE
  ))

logrank_pval_pfs_tp53_no_pm <- left_trunc_pfs_tp53_no_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_tp53_no_pm <-
  tbl_merge(list(tbl_pfs_n_tp53_no_pm,
                 tbl_pfs_n_event_tp53_no_pm,
                 tbl_pfs_med_tp53_no_pm)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "No PM: TP53mut vs. TP53wt",
        footnote = paste0("No PM p-value: ", round(logrank_pval_pfs_tp53_no_pm, 3))
    )
```

*Note: a p-value for the PM and others group is omitted due to the low number of events.*

```{r}
tbl_stack(
  list(
    tbl_merge_pfs_med_tp53,
    tbl_merge_pfs_med_tp53_pm,
    tbl_merge_pfs_med_tp53_pm_only,
    tbl_merge_pfs_med_tp53_pm_others,
    tbl_merge_pfs_med_tp53_no_pm
  )
)
```

# PFS from first-line combination approved therapy by APCmut vs APCwt

```{r}
tbl_pfs_n_apc <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>% 
  select(first_cpt_apc) %>%
  tbl_summary(
    type = list(first_cpt_apc ~ "categorical"),
    label = list(first_cpt_apc = "Full PFS cohort: APCmut vs. APCwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_apc <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1) %>%
  select(first_cpt_apc) %>%
  tbl_summary(
    type = list(first_cpt_apc ~ "categorical"),
    label = list(first_cpt_apc = "Full PFS cohort: APCmut vs. APCwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_apc <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_apc ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_apc = "Full PFS cohort: APCmut vs. APCwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_apc <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_apc ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
    timefix = FALSE
  ))

logrank_pval_pfs_apc <- left_trunc_pfs_apc$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_apc <-
  tbl_merge(list(tbl_pfs_n_apc,
                 tbl_pfs_n_event_apc,
                 tbl_pfs_med_apc)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "Full PFS cohort: APCmut vs. APCwt",
        footnote = paste0("Full PFS cohort p-value: ", round(logrank_pval_pfs_apc, 3))
    )
```

```{r}
tbl_pfs_n_apc_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "PM") %>% 
  select(first_cpt_apc) %>%
  tbl_summary(
    type = list(first_cpt_apc ~ "categorical"),
    label = list(first_cpt_apc = "PM: APCmut vs. APCwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_apc_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         any_peritoneal_met == "PM") %>%
  select(first_cpt_apc) %>%
  tbl_summary(
    type = list(first_cpt_apc ~ "categorical"),
    label = list(first_cpt_apc = "PM: APCmut vs. APCwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_apc_pm <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_apc ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "PM")) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_apc = "PM: APCmut vs. APCwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_apc_pm <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_apc ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             any_peritoneal_met == "PM")
  ))

logrank_pval_pfs_apc_pm <- left_trunc_pfs_apc_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_apc_pm <-
  tbl_merge(list(tbl_pfs_n_apc_pm,
                 tbl_pfs_n_event_apc_pm,
                 tbl_pfs_med_apc_pm)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "PM: APCmut vs. APCwt",
        footnote = paste0("PM: ", round(logrank_pval_pfs_apc_pm, 3))
    )
```

```{r}
tbl_pfs_n_apc_pm_only <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM only") %>% 
  select(first_cpt_apc) %>%
  tbl_summary(
    type = list(first_cpt_apc ~ "categorical"),
    label = list(first_cpt_apc = "PM only: APCmut vs. APCwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_apc_pm_only <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         pm_and_other_mets == "PM only") %>%
  select(first_cpt_apc) %>%
  tbl_summary(
    type = list(first_cpt_apc ~ "categorical"),
    label = list(first_cpt_apc = "PM only: APCmut vs. APCwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_apc_pm_only <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_apc ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM only")) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_apc = "PM only: APCmut vs. APCwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_apc_pm_only <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_apc ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             pm_and_other_mets == "PM only")
  ))

logrank_pval_pfs_apc_pm_only <- left_trunc_pfs_apc_pm_only$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_apc_pm_only <-
  tbl_merge(list(tbl_pfs_n_apc_pm_only,
                 tbl_pfs_n_event_apc_pm_only,
                 tbl_pfs_med_apc_pm_only)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "PM only: APCmut vs. APCwt",
        footnote = paste0("PM only p-value: ", round(logrank_pval_pfs_apc_pm_only, 3))
    )
```

```{r}
tbl_pfs_n_apc_pm_others <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM and others") %>% 
  select(first_cpt_apc) %>%
  tbl_summary(
    type = list(first_cpt_apc ~ "categorical"),
    label = list(first_cpt_apc = "PM and others: APCmut vs. APCwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_apc_pm_others <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         pm_and_other_mets == "PM and others") %>%
  select(first_cpt_apc) %>%
  tbl_summary(
    type = list(first_cpt_apc ~ "categorical"),
    label = list(first_cpt_apc = "PM and others: APCmut vs. APCwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_apc_pm_others <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_apc ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM and others")) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_apc = "PM and others: APCmut vs. APCwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_apc_pm_others <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_apc ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             pm_and_other_mets == "PM and others")
  ))

logrank_pval_pfs_apc_pm_others <- left_trunc_pfs_apc_pm_others$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_apc_pm_others <-
  tbl_merge(list(tbl_pfs_n_apc_pm_others,
                 tbl_pfs_n_event_apc_pm_others,
                 tbl_pfs_med_apc_pm_others)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "PM and others: APCmut vs. APCwt",
        footnote = paste0("PM and others p-value: ", round(logrank_pval_pfs_apc_pm_others, 3))
    )
```

```{r}
tbl_pfs_n_apc_no_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "No PM") %>% 
  select(first_cpt_apc) %>%
  tbl_summary(
    type = list(first_cpt_apc ~ "categorical"),
    label = list(first_cpt_apc = "No PM: APCmut vs. APCwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_apc_no_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         any_peritoneal_met == "No PM") %>%
  select(first_cpt_apc) %>%
  tbl_summary(
    type = list(first_cpt_apc ~ "categorical"),
    label = list(first_cpt_apc = "No PM: APCmut vs. APCwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_apc_no_pm <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_apc ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "No PM"),
timefix = FALSE) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_apc = "No PM: APCmut vs. APCwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_apc_no_pm <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_apc ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             any_peritoneal_met == "No PM"),
    timefix = FALSE
  ))

logrank_pval_pfs_apc_no_pm <- left_trunc_pfs_apc_no_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_apc_no_pm <-
  tbl_merge(list(
    tbl_pfs_n_apc_no_pm,
    tbl_pfs_n_event_apc_no_pm,
    tbl_pfs_med_apc_no_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>%
  modify_table_styling(
    columns = label,
    rows = label == "No PM: APCmut vs. APCwt",
    footnote = paste0("No PM: ", round(logrank_pval_pfs_apc_no_pm, 3))
  )
```

```{r}
tbl_stack(
  list(
    tbl_merge_pfs_med_apc,
    tbl_merge_pfs_med_apc_pm,
    tbl_merge_pfs_med_apc_pm_only,
    tbl_merge_pfs_med_apc_pm_others,
    tbl_merge_pfs_med_apc_no_pm
  )
)
```

# PFS from first-line combination approved therapy by PIK3CAmut vs PIK3CAwt

```{r}
tbl_pfs_n_pik3ca <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>% 
  select(first_cpt_pik3ca) %>%
  tbl_summary(
    type = list(first_cpt_pik3ca ~ "categorical"),
    label = list(first_cpt_pik3ca = "Full PFS cohort: PIK3CAmut vs. PIK3CAwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_pik3ca <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1) %>%
  select(first_cpt_pik3ca) %>%
  tbl_summary(
    type = list(first_cpt_pik3ca ~ "categorical"),
    label = list(first_cpt_pik3ca = "Full PFS cohort: PIK3CAmut vs. PIK3CAwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_pik3ca <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_pik3ca ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_pik3ca = "Full PFS cohort: PIK3CAmut vs. PIK3CAwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_pik3ca <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_pik3ca ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
    timefix = FALSE
  ))

logrank_pval_pfs_pik3ca <- left_trunc_pfs_pik3ca$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_pik3ca <-
  tbl_merge(list(tbl_pfs_n_pik3ca,
                 tbl_pfs_n_event_pik3ca,
                 tbl_pfs_med_pik3ca)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "Full PFS cohort: PIK3CAmut vs. PIK3CAwt",
        footnote = paste0("Full PFS cohort p-value: ", round(logrank_pval_pfs_pik3ca, 3))
    )
```

```{r}
tbl_pfs_n_pik3ca_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "PM") %>% 
  select(first_cpt_pik3ca) %>%
  tbl_summary(
    type = list(first_cpt_pik3ca ~ "categorical"),
    label = list(first_cpt_pik3ca = "PM: PIK3CAmut vs. PIK3CAwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_pik3ca_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         any_peritoneal_met == "PM") %>%
  select(first_cpt_pik3ca) %>%
  tbl_summary(
    type = list(first_cpt_pik3ca ~ "categorical"),
    label = list(first_cpt_pik3ca = "PM: PIK3CAmut vs. PIK3CAwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_pik3ca_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_pik3ca ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           any_peritoneal_met == "PM")
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_pik3ca = "PM: PIK3CAmut vs. PIK3CAwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_pik3ca_pm <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_pik3ca ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             any_peritoneal_met == "PM")
  ))

logrank_pval_pfs_pik3ca_pm <- left_trunc_pfs_pik3ca_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_pik3ca_pm <-
  tbl_merge(list(tbl_pfs_n_pik3ca_pm,
                 tbl_pfs_n_event_pik3ca_pm,
                 tbl_pfs_med_pik3ca_pm)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
    modify_table_styling(
        columns = label,
        rows = label == "PM: PIK3CAmut vs. PIK3CAwt",
        footnote = paste0("PM p-value: ", round(logrank_pval_pfs_pik3ca_pm, 3))
    )
```

```{r}
tbl_pfs_n_pik3ca_pm_only <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM only") %>% 
  select(first_cpt_pik3ca) %>%
  tbl_summary(
    type = list(first_cpt_pik3ca ~ "categorical"),
    label = list(first_cpt_pik3ca = "PM only: PIK3CAmut vs. PIK3CAwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_pik3ca_pm_only <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         pm_and_other_mets == "PM only") %>%
  select(first_cpt_pik3ca) %>%
  tbl_summary(
    type = list(first_cpt_pik3ca ~ "categorical"),
    label = list(first_cpt_pik3ca = "PM only: PIK3CAmut vs. PIK3CAwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_pik3ca_pm_only <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_pik3ca ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM only")) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_pik3ca = "PM only: PIK3CAmut vs. PIK3CAwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_pik3ca_pm_only <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_pik3ca ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             pm_and_other_mets == "PM only")
  ))

logrank_pval_pfs_pik3ca_pm_only <- left_trunc_pfs_pik3ca_pm_only$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_pik3ca_pm_only <-
  tbl_merge(list(tbl_pfs_n_pik3ca_pm_only,
                 tbl_pfs_n_event_pik3ca_pm_only,
                 tbl_pfs_med_pik3ca_pm_only)) %>%
  modify_spanning_header(everything() ~ NA_character_)
```

```{r}
tbl_pfs_n_pik3ca_pm_others <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM and others") %>% 
  select(first_cpt_pik3ca) %>%
  tbl_summary(
    type = list(first_cpt_pik3ca ~ "categorical"),
    label = list(first_cpt_pik3ca = "PM and others: PIK3CAmut vs. PIK3CAwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_pik3ca_pm_others <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         pm_and_other_mets == "PM and others") %>%
  select(first_cpt_pik3ca) %>%
  tbl_summary(
    type = list(first_cpt_pik3ca ~ "categorical"),
    label = list(first_cpt_pik3ca = "PM and others: PIK3CAmut vs. PIK3CAwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_pik3ca_pm_others <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_pik3ca ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM and others")) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_pik3ca = "PM and others: PIK3CAmut vs. PIK3CAwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_pik3ca_pm_others <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_pik3ca ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             pm_and_other_mets == "PM and others")
  ))

logrank_pval_pfs_pik3ca_pm_others <- left_trunc_pfs_pik3ca_pm_others$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_pik3ca_pm_others <-
  tbl_merge(list(tbl_pfs_n_pik3ca_pm_others,
                 tbl_pfs_n_event_pik3ca_pm_others,
                 tbl_pfs_med_pik3ca_pm_others)) %>%
  modify_spanning_header(everything() ~ NA_character_)
```

```{r}
tbl_pfs_n_pik3ca_no_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "No PM") %>% 
  select(first_cpt_pik3ca) %>%
  tbl_summary(
    type = list(first_cpt_pik3ca ~ "categorical"),
    label = list(first_cpt_pik3ca = "No PM: PIK3CAmut vs. PIK3CAwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_pik3ca_no_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         any_peritoneal_met == "No PM") %>%
  select(first_cpt_pik3ca) %>%
  tbl_summary(
    type = list(first_cpt_pik3ca ~ "categorical"),
    label = list(first_cpt_pik3ca = "No PM: PIK3CAmut vs. PIK3CAwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_pik3ca_no_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_pik3ca ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           any_peritoneal_met == "No PM"),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_pik3ca = "No PM: PIK3CAmut vs. PIK3CAwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_pik3ca_no_pm <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_pik3ca ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             any_peritoneal_met == "No PM"),
    timefix = FALSE
  ))

logrank_pval_pfs_pik3ca_no_pm <- left_trunc_pfs_pik3ca_no_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_pik3ca_no_pm <-
  tbl_merge(
    list(
      tbl_pfs_n_pik3ca_no_pm,
      tbl_pfs_n_event_pik3ca_no_pm,
      tbl_pfs_med_pik3ca_no_pm
    )
  ) %>%
  modify_spanning_header(everything() ~ NA_character_) %>%
  modify_table_styling(
    columns = label,
    rows = label == "No PM: PIK3CAmut vs. PIK3CAwt",
    footnote = paste0("No PM p-value: ", round(logrank_pval_pfs_pik3ca_no_pm, 3))
  )
```

*Note: p-values for the PM only subgroup and PM and others subgroup are omitted due to the low number of events.*

```{r}
tbl_stack(
  list(
    tbl_merge_pfs_med_pik3ca,
    tbl_merge_pfs_med_pik3ca_pm,
    tbl_merge_pfs_med_pik3ca_pm_only,
    tbl_merge_pfs_med_pik3ca_pm_others,
    tbl_merge_pfs_med_pik3ca_no_pm
  )
)
```

# PFS from first-line combination approved therapy by KRAS/BRAF/NRASmut vs KRAS/BRAF/NRASwt

```{r}
tbl_pfs_n_kras_braf_nras <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>% 
  select(first_cpt_kras_braf_nras) %>%
  tbl_summary(
    type = list(first_cpt_kras_braf_nras ~ "categorical"),
    label = list(first_cpt_kras_braf_nras = "Full PFS cohort: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_kras_braf_nras <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1) %>%
  select(first_cpt_kras_braf_nras) %>%
  tbl_summary(
    type = list(first_cpt_kras_braf_nras ~ "categorical"),
    label = list(first_cpt_kras_braf_nras = "Full PFS cohort: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_kras_braf_nras <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_kras_braf_nras ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_kras_braf_nras = "Full PFS cohort: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_kras_braf_nras <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_kras_braf_nras ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
    timefix = FALSE
  ))

logrank_pval_pfs_kras_braf_nras <- left_trunc_pfs_kras_braf_nras$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_kras_braf_nras <-
  tbl_merge(list(tbl_pfs_n_kras_braf_nras,
                 tbl_pfs_n_event_kras_braf_nras,
                 tbl_pfs_med_kras_braf_nras)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "Full PFS cohort: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt",
        footnote = paste0("Full PFS cohort p-value: ", round(logrank_pval_pfs_kras_braf_nras, 3))
    )
```

```{r}
tbl_pfs_n_kras_braf_nras_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "PM") %>% 
  select(first_cpt_kras_braf_nras) %>%
  tbl_summary(
    type = list(first_cpt_kras_braf_nras ~ "categorical"),
    label = list(first_cpt_kras_braf_nras = "PM: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_kras_braf_nras_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         any_peritoneal_met == "PM") %>%
  select(first_cpt_kras_braf_nras) %>%
  tbl_summary(
    type = list(first_cpt_kras_braf_nras ~ "categorical"),
    label = list(first_cpt_kras_braf_nras = "PM: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_kras_braf_nras_pm <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_kras_braf_nras ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "PM")) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_kras_braf_nras = "PM: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_kras_braf_nras_pm <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_kras_braf_nras ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             any_peritoneal_met == "PM")
  ))

logrank_pval_pfs_kras_braf_nras_pm <- left_trunc_pfs_kras_braf_nras_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_kras_braf_nras_pm <-
  tbl_merge(list(tbl_pfs_n_kras_braf_nras_pm,
                 tbl_pfs_n_event_kras_braf_nras_pm,
                 tbl_pfs_med_kras_braf_nras_pm)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "PM: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt",
        footnote = paste0("PM p-value: ", round(logrank_pval_pfs_kras_braf_nras_pm, 3))
    )
```

```{r}
tbl_pfs_n_kras_braf_nras_pm_only <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM only") %>% 
  select(first_cpt_kras_braf_nras) %>%
  tbl_summary(
    type = list(first_cpt_kras_braf_nras ~ "categorical"),
    label = list(first_cpt_kras_braf_nras = "PM only: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_kras_braf_nras_pm_only <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         pm_and_other_mets == "PM only") %>%
  select(first_cpt_kras_braf_nras) %>%
  tbl_summary(
    type = list(first_cpt_kras_braf_nras ~ "categorical"),
    label = list(first_cpt_kras_braf_nras = "PM only: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_kras_braf_nras_pm_only <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_kras_braf_nras ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM only")) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_kras_braf_nras = "PM only: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_kras_braf_nras_pm_only <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_kras_braf_nras ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             pm_and_other_mets == "PM only")
  ))

logrank_pval_pfs_kras_braf_nras_pm_only <- left_trunc_pfs_kras_braf_nras_pm_only$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_kras_braf_nras_pm_only <-
  tbl_merge(list(tbl_pfs_n_kras_braf_nras_pm_only,
                 tbl_pfs_n_event_kras_braf_nras_pm_only,
                 tbl_pfs_med_kras_braf_nras_pm_only)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "PM only: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt",
        footnote = paste0("PM only p-value: ", round(logrank_pval_pfs_kras_braf_nras_pm_only, 3))
    )
```

```{r}
tbl_pfs_n_kras_braf_nras_pm_others <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM and others") %>% 
  select(first_cpt_kras_braf_nras) %>%
  tbl_summary(
    type = list(first_cpt_kras_braf_nras ~ "categorical"),
    label = list(first_cpt_kras_braf_nras = "PM and others: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_kras_braf_nras_pm_others <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         pm_and_other_mets == "PM and others") %>%
  select(first_cpt_kras_braf_nras) %>%
  tbl_summary(
    type = list(first_cpt_kras_braf_nras ~ "categorical"),
    label = list(first_cpt_kras_braf_nras = "PM and others: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_kras_braf_nras_pm_others <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_kras_braf_nras ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM and others")) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_kras_braf_nras = "PM and others: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_kras_braf_nras_pm_others <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_kras_braf_nras ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             pm_and_other_mets == "PM and others")
  ))

logrank_pval_pfs_kras_braf_nras_pm_others <- left_trunc_pfs_kras_braf_nras_pm_others$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_kras_braf_nras_pm_others <-
  tbl_merge(list(tbl_pfs_n_kras_braf_nras_pm_others,
                 tbl_pfs_n_event_kras_braf_nras_pm_others,
                 tbl_pfs_med_kras_braf_nras_pm_others)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "PM and others: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt",
        footnote = paste0("PM and others p-value: ", round(logrank_pval_pfs_kras_braf_nras_pm_others, 3))
    )
```

```{r}
tbl_pfs_n_kras_braf_nras_no_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "No PM") %>% 
  select(first_cpt_kras_braf_nras) %>%
  tbl_summary(
    type = list(first_cpt_kras_braf_nras ~ "categorical"),
    label = list(first_cpt_kras_braf_nras = "No PM: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_kras_braf_nras_no_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         any_peritoneal_met == "No PM") %>%
  select(first_cpt_kras_braf_nras) %>%
  tbl_summary(
    type = list(first_cpt_kras_braf_nras ~ "categorical"),
    label = list(first_cpt_kras_braf_nras = "No PM: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_kras_braf_nras_no_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_kras_braf_nras ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(
      tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
      any_peritoneal_met == "No PM"
    ),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_kras_braf_nras = "No PM: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_kras_braf_nras_no_pm <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_kras_braf_nras ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             any_peritoneal_met == "No PM"),
    timefix = FALSE
  ))

logrank_pval_pfs_kras_braf_nras_no_pm <- left_trunc_pfs_kras_braf_nras_no_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_kras_braf_nras_no_pm <-
  tbl_merge(list(tbl_pfs_n_kras_braf_nras_no_pm,
                 tbl_pfs_n_event_kras_braf_nras_no_pm,
                 tbl_pfs_med_kras_braf_nras_no_pm)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "No PM: KRAS/BRAF/NRASmut vs. KRAS/BRAF/NRASwt",
        footnote = paste0("No PM p-value: ", round(logrank_pval_pfs_kras_braf_nras_no_pm, 3))
    )
```

*Note: a p-value for the PM and others group is omitted due to the low number of events.*

```{r}
tbl_stack(
  list(
    tbl_merge_pfs_med_kras_braf_nras,
    tbl_merge_pfs_med_kras_braf_nras_pm,
    tbl_merge_pfs_med_kras_braf_nras_pm_only,
    tbl_merge_pfs_med_kras_braf_nras_pm_others,
    tbl_merge_pfs_med_kras_braf_nras_no_pm
  )
)
```

# PFS from first-line combination approved therapy by KRASmut vs KRASwt

```{r}
tbl_pfs_n_kras <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>% 
  select(first_cpt_kras) %>%
  tbl_summary(
    type = list(first_cpt_kras ~ "categorical"),
    label = list(first_cpt_kras = "Full PFS cohort: KRASmut vs. KRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_kras <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1) %>%
  select(first_cpt_kras) %>%
  tbl_summary(
    type = list(first_cpt_kras ~ "categorical"),
    label = list(first_cpt_kras = "Full PFS cohort: KRASmut vs. KRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_kras <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_kras ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_kras = "Full PFS cohort: KRASmut vs. KRASwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_kras <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_kras ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
    timefix = FALSE
  ))

logrank_pval_pfs_kras <- left_trunc_pfs_kras$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_kras <-
  tbl_merge(list(tbl_pfs_n_kras,
                 tbl_pfs_n_event_kras,
                 tbl_pfs_med_kras)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "Full PFS cohort: KRASmut vs. KRASwt",
        footnote = paste0("Full PFS cohort p-value: ", round(logrank_pval_pfs_kras, 3))
    )
```

```{r}
tbl_pfs_n_kras_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "PM") %>% 
  select(first_cpt_kras) %>%
  tbl_summary(
    type = list(first_cpt_kras ~ "categorical"),
    label = list(first_cpt_kras = "PM: KRASmut vs. KRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_kras_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         any_peritoneal_met == "PM") %>%
  select(first_cpt_kras) %>%
  tbl_summary(
    type = list(first_cpt_kras ~ "categorical"),
    label = list(first_cpt_kras = "PM: KRASmut vs. KRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_kras_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_kras ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           any_peritoneal_met == "PM")
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_kras = "PM: KRASmut vs. KRASwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_kras_pm <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_kras ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             any_peritoneal_met == "PM")
  ))

logrank_pval_pfs_kras_pm <- left_trunc_pfs_kras_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_kras_pm <-
  tbl_merge(list(tbl_pfs_n_kras_pm,
                 tbl_pfs_n_event_kras_pm,
                 tbl_pfs_med_kras_pm)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
    modify_table_styling(
        columns = label,
        rows = label == "PM: KRASmut vs. KRASwt",
        footnote = paste0("PM p-value: ", round(logrank_pval_pfs_kras_pm, 3))
    )
```

```{r}
tbl_pfs_n_kras_pm_only <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM only") %>% 
  select(first_cpt_kras) %>%
  tbl_summary(
    type = list(first_cpt_kras ~ "categorical"),
    label = list(first_cpt_kras = "PM only: KRASmut vs. KRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_kras_pm_only <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         pm_and_other_mets == "PM only") %>%
  select(first_cpt_kras) %>%
  tbl_summary(
    type = list(first_cpt_kras ~ "categorical"),
    label = list(first_cpt_kras = "PM only: KRASmut vs. KRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_kras_pm_only <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_kras ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM only")) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_kras = "PM only: KRASmut vs. KRASwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_kras_pm_only <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_kras ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             pm_and_other_mets == "PM only")
  ))

logrank_pval_pfs_kras_pm_only <- left_trunc_pfs_kras_pm_only$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_kras_pm_only <-
  tbl_merge(list(tbl_pfs_n_kras_pm_only,
                 tbl_pfs_n_event_kras_pm_only,
                 tbl_pfs_med_kras_pm_only)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "PM only: KRASmut vs. KRASwt",
        footnote = paste0("PM only p-value: ", round(logrank_pval_pfs_kras_pm_only, 3))
    )
```

```{r}
tbl_pfs_n_kras_pm_others <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM and others") %>% 
  select(first_cpt_kras) %>%
  tbl_summary(
    type = list(first_cpt_kras ~ "categorical"),
    label = list(first_cpt_kras = "PM and others: KRASmut vs. KRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_kras_pm_others <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         pm_and_other_mets == "PM and others") %>%
  select(first_cpt_kras) %>%
  tbl_summary(
    type = list(first_cpt_kras ~ "categorical"),
    label = list(first_cpt_kras = "PM and others: KRASmut vs. KRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_kras_pm_others <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_kras ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM and others")) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_kras = "PM and others: KRASmut vs. KRASwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_kras_pm_others <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_kras ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             pm_and_other_mets == "PM and others")
  ))

logrank_pval_pfs_kras_pm_others <- left_trunc_pfs_kras_pm_others$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_kras_pm_others <-
  tbl_merge(list(tbl_pfs_n_kras_pm_others,
                 tbl_pfs_n_event_kras_pm_others,
                 tbl_pfs_med_kras_pm_others)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "PM and others: KRASmut vs. KRASwt",
        footnote = paste0("PM and others p-value: ", round(logrank_pval_pfs_kras_pm_others, 3))
    )
```

```{r}
tbl_pfs_n_kras_no_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "No PM") %>% 
  select(first_cpt_kras) %>%
  tbl_summary(
    type = list(first_cpt_kras ~ "categorical"),
    label = list(first_cpt_kras = "No PM: KRASmut vs. KRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_kras_no_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         any_peritoneal_met == "No PM") %>%
  select(first_cpt_kras) %>%
  tbl_summary(
    type = list(first_cpt_kras ~ "categorical"),
    label = list(first_cpt_kras = "No PM: KRASmut vs. KRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_kras_no_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_kras ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           any_peritoneal_met == "No PM"),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_kras = "No PM: KRASmut vs. KRASwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_kras_no_pm <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_kras ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             any_peritoneal_met == "No PM"),
    timefix = FALSE
  ))

logrank_pval_pfs_kras_no_pm <- left_trunc_pfs_kras_no_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_kras_no_pm <-
  tbl_merge(
    list(
      tbl_pfs_n_kras_no_pm,
      tbl_pfs_n_event_kras_no_pm,
      tbl_pfs_med_kras_no_pm
    )
  ) %>%
  modify_spanning_header(everything() ~ NA_character_) %>%
  modify_table_styling(
    columns = label,
    rows = label == "No PM: KRASmut vs. KRASwt",
    footnote = paste0("No PM p-value: ", round(logrank_pval_pfs_kras_no_pm, 3))
  )
```

```{r}
tbl_stack(
  list(
    tbl_merge_pfs_med_kras,
    tbl_merge_pfs_med_kras_pm,
    tbl_merge_pfs_med_kras_pm_only,
    tbl_merge_pfs_med_kras_pm_others,
    tbl_merge_pfs_med_kras_no_pm
  )
)
```

# PFS from first-line combination approved therapy by BRAFmut vs BRAFwt

```{r}
tbl_pfs_n_braf <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>% 
  select(first_cpt_braf) %>%
  tbl_summary(
    type = list(first_cpt_braf ~ "categorical"),
    label = list(first_cpt_braf = "Full PFS cohort: BRAFmut vs. BRAFwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_braf <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1) %>%
  select(first_cpt_braf) %>%
  tbl_summary(
    type = list(first_cpt_braf ~ "categorical"),
    label = list(first_cpt_braf = "Full PFS cohort: BRAFmut vs. BRAFwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_braf <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_braf ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_braf = "Full PFS cohort: BRAFmut vs. BRAFwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_braf <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_braf ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
    timefix = FALSE
  ))

logrank_pval_pfs_braf <- left_trunc_pfs_braf$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_braf <-
  tbl_merge(list(tbl_pfs_n_braf,
                 tbl_pfs_n_event_braf,
                 tbl_pfs_med_braf)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "Full PFS cohort: BRAFmut vs. BRAFwt",
        footnote = paste0("Full PFS cohort p-value: ", round(logrank_pval_pfs_braf, 3))
    )
```

```{r}
tbl_pfs_n_braf_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "PM") %>% 
  select(first_cpt_braf) %>%
  tbl_summary(
    type = list(first_cpt_braf ~ "categorical"),
    label = list(first_cpt_braf = "PM: BRAFmut vs. BRAFwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_braf_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         any_peritoneal_met == "PM") %>%
  select(first_cpt_braf) %>%
  tbl_summary(
    type = list(first_cpt_braf ~ "categorical"),
    label = list(first_cpt_braf = "PM: BRAFmut vs. BRAFwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_braf_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_braf ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           any_peritoneal_met == "PM")
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_braf = "PM: BRAFmut vs. BRAFwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_braf_pm <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_braf ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             any_peritoneal_met == "PM")
  ))

logrank_pval_pfs_braf_pm <- left_trunc_pfs_braf_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_braf_pm <-
  tbl_merge(list(tbl_pfs_n_braf_pm,
                 tbl_pfs_n_event_braf_pm,
                 tbl_pfs_med_braf_pm)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
    modify_table_styling(
        columns = label,
        rows = label == "PM: BRAFmut vs. BRAFwt",
        footnote = paste0("PM p-value: ", round(logrank_pval_pfs_braf_pm, 3))
    )
```

```{r}
tbl_pfs_n_braf_pm_only <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM only") %>% 
  select(first_cpt_braf) %>%
  tbl_summary(
    type = list(first_cpt_braf ~ "categorical"),
    label = list(first_cpt_braf = "PM only: BRAFmut vs. BRAFwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_braf_pm_only <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         pm_and_other_mets == "PM only") %>%
  select(first_cpt_braf) %>%
  tbl_summary(
    type = list(first_cpt_braf ~ "categorical"),
    label = list(first_cpt_braf = "PM only: BRAFmut vs. BRAFwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_braf_pm_only <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_braf ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM only")) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_braf = "PM only: BRAFmut vs. BRAFwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_braf_pm_only <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_braf ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             pm_and_other_mets == "PM only")
  ))

logrank_pval_pfs_braf_pm_only <- left_trunc_pfs_braf_pm_only$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_braf_pm_only <-
  tbl_merge(list(tbl_pfs_n_braf_pm_only,
                 tbl_pfs_n_event_braf_pm_only,
                 tbl_pfs_med_braf_pm_only)) %>%
  modify_spanning_header(everything() ~ NA_character_)
```

```{r}
tbl_pfs_n_braf_pm_others <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM and others") %>% 
  select(first_cpt_braf) %>%
  tbl_summary(
    type = list(first_cpt_braf ~ "categorical"),
    label = list(first_cpt_braf = "PM and others: BRAFmut vs. BRAFwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_braf_pm_others <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         pm_and_other_mets == "PM and others") %>%
  select(first_cpt_braf) %>%
  tbl_summary(
    type = list(first_cpt_braf ~ "categorical"),
    label = list(first_cpt_braf = "PM and others: BRAFmut vs. BRAFwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_braf_pm_others <- survfit(Surv(
  time = reg_start_cpt_rep_mos,
  time2 = tt_pfs_i_and_m_g_mos,
  event = pfs_i_and_m_g_status
) ~ first_cpt_braf ,
data = pfs_surv_cohort_first_line_folfoxiri_cohort %>% 
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pm_and_other_mets == "PM and others")) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_braf = "PM and others: BRAFmut vs. BRAFwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_braf_pm_others <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_braf ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             pm_and_other_mets == "PM and others")
  ))

logrank_pval_pfs_braf_pm_others <- left_trunc_pfs_braf_pm_others$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_braf_pm_others <-
  tbl_merge(list(tbl_pfs_n_braf_pm_others,
                 tbl_pfs_n_event_braf_pm_others,
                 tbl_pfs_med_braf_pm_others)) %>%
  modify_spanning_header(everything() ~ NA_character_)
```

```{r}
tbl_pfs_n_braf_no_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "No PM") %>% 
  select(first_cpt_braf) %>%
  tbl_summary(
    type = list(first_cpt_braf ~ "categorical"),
    label = list(first_cpt_braf = "No PM: BRAFmut vs. BRAFwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_braf_no_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         any_peritoneal_met == "No PM") %>%
  select(first_cpt_braf) %>%
  tbl_summary(
    type = list(first_cpt_braf ~ "categorical"),
    label = list(first_cpt_braf = "No PM: BRAFmut vs. BRAFwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_braf_no_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_braf ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           any_peritoneal_met == "No PM"),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_braf = "No PM: BRAFmut vs. BRAFwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_braf_no_pm <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_braf ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             any_peritoneal_met == "No PM"),
    timefix = FALSE
  ))

logrank_pval_pfs_braf_no_pm <- left_trunc_pfs_braf_no_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_braf_no_pm <-
  tbl_merge(
    list(
      tbl_pfs_n_braf_no_pm,
      tbl_pfs_n_event_braf_no_pm,
      tbl_pfs_med_braf_no_pm
    )
  ) %>%
  modify_spanning_header(everything() ~ NA_character_) %>%
  modify_table_styling(
    columns = label,
    rows = label == "No PM: BRAFmut vs. BRAFwt",
    footnote = paste0("No PM p-value: ", round(logrank_pval_pfs_braf_no_pm, 3))
  )
```

*Note: p-values for the PM only subgroup and PM and others subgroup are omitted due to the low number of events.*

```{r}
tbl_stack(
  list(
    tbl_merge_pfs_med_braf,
    tbl_merge_pfs_med_braf_pm,
    tbl_merge_pfs_med_braf_pm_only,
    tbl_merge_pfs_med_braf_pm_others,
    tbl_merge_pfs_med_braf_no_pm
  )
)
```

# PFS from first-line combination approved therapy by NRASmut vs NRASwt

```{r}
tbl_pfs_n_nras <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>% 
  select(first_cpt_nras) %>%
  tbl_summary(
    type = list(first_cpt_nras ~ "categorical"),
    label = list(first_cpt_nras = "Full PFS cohort: NRASmut vs. NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_nras <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1) %>%
  select(first_cpt_nras) %>%
  tbl_summary(
    type = list(first_cpt_nras ~ "categorical"),
    label = list(first_cpt_nras = "Full PFS cohort: NRASmut vs. NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_nras <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_nras ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_nras = "Full PFS cohort: NRASmut vs. NRASwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_nras <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_nras ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
    timefix = FALSE
  ))

logrank_pval_pfs_nras <- left_trunc_pfs_nras$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_nras <-
  tbl_merge(list(tbl_pfs_n_nras,
                 tbl_pfs_n_event_nras,
                 tbl_pfs_med_nras)) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
        columns = label,
        rows = label == "Full PFS cohort: NRASmut vs. NRASwt",
        footnote = paste0("Full PFS cohort p-value: ", round(logrank_pval_pfs_nras, 3))
    )
```

```{r}
tbl_pfs_n_nras_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "PM") %>% 
  select(first_cpt_nras) %>%
  tbl_summary(
    type = list(first_cpt_nras ~ "categorical"),
    label = list(first_cpt_nras = "PM: NRASmut vs. NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_nras_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         any_peritoneal_met == "PM") %>%
  select(first_cpt_nras) %>%
  tbl_summary(
    type = list(first_cpt_nras ~ "categorical"),
    label = list(first_cpt_nras = "PM: NRASmut vs. NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_nras_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_nras ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           any_peritoneal_met == "PM")
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_nras = "PM: NRASmut vs. NRASwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_nras_pm <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_nras ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             any_peritoneal_met == "PM")
  ))

logrank_pval_pfs_nras_pm <- left_trunc_pfs_nras_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_nras_pm <-
  tbl_merge(list(tbl_pfs_n_nras_pm,
                 tbl_pfs_n_event_nras_pm,
                 tbl_pfs_med_nras_pm)) %>%
  modify_spanning_header(everything() ~ NA_character_)
```

```{r}
tbl_pfs_n_nras_no_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         any_peritoneal_met == "No PM") %>% 
  select(first_cpt_nras) %>%
  tbl_summary(
    type = list(first_cpt_nras ~ "categorical"),
    label = list(first_cpt_nras = "No PM: NRASmut vs. NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_n_event_nras_no_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1,
         any_peritoneal_met == "No PM") %>%
  select(first_cpt_nras) %>%
  tbl_summary(
    type = list(first_cpt_nras ~ "categorical"),
    label = list(first_cpt_nras = "No PM: NRASmut vs. NRASwt"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_nras_no_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ first_cpt_nras ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
           any_peritoneal_met == "No PM"),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(first_cpt_nras = "No PM: NRASmut vs. NRASwt"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_nras_no_pm <-
  summary(coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ first_cpt_nras ,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
             any_peritoneal_met == "No PM"),
    timefix = FALSE
  ))

logrank_pval_pfs_nras_no_pm <- left_trunc_pfs_nras_no_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_nras_no_pm <-
  tbl_merge(
    list(
      tbl_pfs_n_nras_no_pm,
      tbl_pfs_n_event_nras_no_pm,
      tbl_pfs_med_nras_no_pm
    )
  ) %>%
  modify_spanning_header(everything() ~ NA_character_) %>%
  modify_table_styling(
    columns = label,
    rows = label == "No PM: NRASmut vs. NRASwt",
    footnote = paste0("No PM p-value: ", round(logrank_pval_pfs_nras_no_pm, 3))
  )
```

```{r}
tbl_stack(
  list(
    tbl_merge_pfs_med_nras,
    tbl_merge_pfs_med_nras_pm,
    tbl_merge_pfs_med_nras_no_pm
  )
)
```

# PFS by alteration status x presence of PM

```{r}
tbl_n_kras_status_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(kras_any_pm) %>%
  tbl_summary(
    type = list(kras_any_pm ~ "categorical"),
    label = list(kras_any_pm = "KRAS status x presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_kras_status_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(pfs_i_and_m_g_status == 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(kras_any_pm) %>%
  tbl_summary(
    type = list(kras_any_pm ~ "categorical"),
    label = list(kras_any_pm = "KRAS status x presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_kras_status_any_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ kras_any_pm ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(kras_any_pm = "KRAS status x presence of PM"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x)
      gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_kras_status_any_pm <-
  summary(
    coxph(
      Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
    ) ~ kras_any_pm,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(
          tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos
        ),
    timefix = FALSE
    )
  )

logrank_pval_pfs_kras_status_any_pm <- left_trunc_pfs_kras_status_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_kras_status_any_pm <-
  tbl_merge(list(
    tbl_n_kras_status_any_pm_pfs,
    tbl_n_event_kras_status_any_pm_pfs,
    tbl_pfs_med_kras_status_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "KRAS status x presence of PM",
    footnote = paste0("KRAS p-value: ", round(logrank_pval_pfs_kras_status_any_pm, 3))
  )
```

```{r}
tbl_n_braf_status_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(braf_any_pm) %>%
  tbl_summary(
    type = list(braf_any_pm ~ "categorical"),
    label = list(braf_any_pm = "BRAF status x presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_braf_status_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(pfs_i_and_m_g_status == 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(braf_any_pm) %>%
  tbl_summary(
    type = list(braf_any_pm ~ "categorical"),
    label = list(braf_any_pm = "BRAF status x presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_braf_status_any_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ braf_any_pm ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(braf_any_pm = "BRAF status x presence of PM"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_braf_status_any_pm <-
  summary(
    coxph(
      Surv(
        time = reg_start_cpt_rep_mos,
        time2 = tt_pfs_i_and_m_g_mos,
        event = pfs_i_and_m_g_status
      ) ~ braf_any_pm,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(
          tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos
        ),
      timefix = FALSE
    )
  )

logrank_pval_pfs_braf_status_any_pm <- left_trunc_pfs_braf_status_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_braf_status_any_pm <-
  tbl_merge(list(
    tbl_n_braf_status_any_pm_pfs,
    tbl_n_event_braf_status_any_pm_pfs,
    tbl_pfs_med_braf_status_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "BRAF status x presence of PM",
    footnote = paste0("BRAF p-value: ", round(logrank_pval_pfs_braf_status_any_pm, 3))
  )
```

```{r}
tbl_n_nras_status_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(nras_any_pm) %>%
  tbl_summary(
    type = list(nras_any_pm ~ "categorical"),
    label = list(nras_any_pm = "NRAS status x presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_nras_status_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(pfs_i_and_m_g_status == 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(nras_any_pm) %>%
  tbl_summary(
    type = list(nras_any_pm ~ "categorical"),
    label = list(nras_any_pm = "NRAS status x presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_nras_status_any_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ nras_any_pm ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(nras_any_pm = "NRAS status x presence of PM"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_nras_status_any_pm <-
  summary(
    coxph(
      Surv(
        time = reg_start_cpt_rep_mos,
        time2 = tt_pfs_i_and_m_g_mos,
        event = pfs_i_and_m_g_status
      ) ~ nras_any_pm,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(
          tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos
        ),
      timefix = FALSE
    )
  )

logrank_pval_pfs_nras_status_any_pm <- left_trunc_pfs_nras_status_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_nras_status_any_pm <-
  tbl_merge(list(
    tbl_n_nras_status_any_pm_pfs,
    tbl_n_event_nras_status_any_pm_pfs,
    tbl_pfs_med_nras_status_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "NRAS status x presence of PM",
    footnote = paste0("NRAS p-value: ", round(logrank_pval_pfs_nras_status_any_pm, 3))
  )
```

```{r}
tbl_n_ras_braf_status_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(ras_braf_any_pm) %>%
  tbl_summary(
    type = list(ras_braf_any_pm ~ "categorical"),
    label = list(ras_braf_any_pm = "RAS/BRAF status x presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_ras_braf_status_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(pfs_i_and_m_g_status == 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(ras_braf_any_pm) %>%
  tbl_summary(
    type = list(ras_braf_any_pm ~ "categorical"),
    label = list(ras_braf_any_pm = "RAS/BRAF status x presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_ras_braf_status_any_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ ras_braf_any_pm ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(ras_braf_any_pm = "RAS/BRAF status x presence of PM"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_ras_braf_status_any_pm <-
  summary(
    coxph(
      Surv(
        time = reg_start_cpt_rep_mos,
        time2 = tt_pfs_i_and_m_g_mos,
        event = pfs_i_and_m_g_status
      ) ~ ras_braf_any_pm,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(
          tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos
        ),
      timefix = FALSE
    )
  )

logrank_pval_pfs_ras_braf_status_any_pm <- left_trunc_pfs_ras_braf_status_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_ras_braf_status_any_pm <-
  tbl_merge(list(
    tbl_n_ras_braf_status_any_pm_pfs,
    tbl_n_event_ras_braf_status_any_pm_pfs,
    tbl_pfs_med_ras_braf_status_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "RAS/BRAF status x presence of PM",
    footnote = paste0("RAS/BRAF p-value: ", round(logrank_pval_pfs_ras_braf_status_any_pm, 3))
  )
```

```{r}
tbl_n_pik3ca_status_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(pik3ca_any_pm) %>%
  tbl_summary(
    type = list(pik3ca_any_pm ~ "categorical"),
    label = list(pik3ca_any_pm = "PIK3CA status x presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_pik3ca_status_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(pfs_i_and_m_g_status == 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(pik3ca_any_pm) %>%
  tbl_summary(
    type = list(pik3ca_any_pm ~ "categorical"),
    label = list(pik3ca_any_pm = "PIK3CA status x presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_pik3ca_status_any_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ pik3ca_any_pm ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(pik3ca_any_pm = "PIK3CA status x presence of PM"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_pik3ca_status_any_pm <-
  summary(
    coxph(
      Surv(
        time = reg_start_cpt_rep_mos,
        time2 = tt_pfs_i_and_m_g_mos,
        event = pfs_i_and_m_g_status
      ) ~ pik3ca_any_pm,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(
          tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos
        ),
      timefix = FALSE
    )
  )

logrank_pval_pfs_pik3ca_status_any_pm <- left_trunc_pfs_pik3ca_status_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_pik3ca_status_any_pm <-
  tbl_merge(list(
    tbl_n_pik3ca_status_any_pm_pfs,
    tbl_n_event_pik3ca_status_any_pm_pfs,
    tbl_pfs_med_pik3ca_status_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "PIK3CA status x presence of PM",
    footnote = paste0("PIK3CA p-value: ", round(logrank_pval_pfs_pik3ca_status_any_pm, 3))
  )
```

```{r}
tbl_n_apc_status_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(apc_any_pm) %>%
  tbl_summary(
    type = list(apc_any_pm ~ "categorical"),
    label = list(apc_any_pm = "APC status x presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_apc_status_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(pfs_i_and_m_g_status == 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(apc_any_pm) %>%
  tbl_summary(
    type = list(apc_any_pm ~ "categorical"),
    label = list(apc_any_pm = "APC status x presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_apc_status_any_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ apc_any_pm ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(apc_any_pm = "APC status x presence of PM"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_apc_status_any_pm <-
  summary(
    coxph(
      Surv(
        time = reg_start_cpt_rep_mos,
        time2 = tt_pfs_i_and_m_g_mos,
        event = pfs_i_and_m_g_status
      ) ~ apc_any_pm,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(
          tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos
        ),
      timefix = FALSE
    )
  )

logrank_pval_pfs_apc_status_any_pm <- left_trunc_pfs_apc_status_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_apc_status_any_pm <-
  tbl_merge(list(
    tbl_n_apc_status_any_pm_pfs,
    tbl_n_event_apc_status_any_pm_pfs,
    tbl_pfs_med_apc_status_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "APC status x presence of PM",
    footnote = paste0("APC p-value: ", round(logrank_pval_pfs_apc_status_any_pm, 3))
  )
```

```{r}
tbl_n_tp53_status_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(tp53_any_pm) %>%
  tbl_summary(
    type = list(tp53_any_pm ~ "categorical"),
    label = list(tp53_any_pm = "TP53 status x presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_event_tp53_status_any_pm_pfs <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(pfs_i_and_m_g_status == 1,
         tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(tp53_any_pm) %>%
  tbl_summary(
    type = list(tp53_any_pm ~ "categorical"),
    label = list(tp53_any_pm = "TP53 status x presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N event**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_pfs_med_tp53_status_any_pm <- survfit(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ tp53_any_pm ,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_survfit(
    probs = .5,
    label = list(tp53_any_pm = "TP53 status x presence of PM"),
    label_header = "**Median PFS (months)**",
    estimate_fun = function(.x) gtsummary::style_sigfig(.x, digits = 4)
  )

left_trunc_pfs_tp53_status_any_pm <-
  summary(
    coxph(
      Surv(
        time = reg_start_cpt_rep_mos,
        time2 = tt_pfs_i_and_m_g_mos,
        event = pfs_i_and_m_g_status
      ) ~ tp53_any_pm,
      data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
        filter(
          tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos
        ),
      timefix = FALSE
    )
  )

logrank_pval_pfs_tp53_status_any_pm <- left_trunc_pfs_tp53_status_any_pm$sctest %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "rowname") %>% 
  filter(rowname == "pvalue") %>% 
  select(".") %>% 
  unlist()

tbl_merge_pfs_med_tp53_status_any_pm <-
  tbl_merge(list(
    tbl_n_tp53_status_any_pm_pfs,
    tbl_n_event_tp53_status_any_pm_pfs,
    tbl_pfs_med_tp53_status_any_pm
  )) %>%
  modify_spanning_header(everything() ~ NA_character_) %>% 
  modify_table_styling(
    columns = label,
    rows = label == "TP53 status x presence of PM",
    footnote = paste0("TP53 p-value: ", round(logrank_pval_pfs_tp53_status_any_pm, 3))
  )
```

```{r}
## stack tbl_merge
pfs_status_by_pm <- tbl_stack(
  list(
    tbl_merge_pfs_med_kras_status_any_pm,
    tbl_merge_pfs_med_braf_status_any_pm,
    ## commenting out NRAS comparison bc of small #s by subgroup
    # tbl_merge_pfs_med_nras_status_any_pm,
    tbl_merge_pfs_med_ras_braf_status_any_pm,
    tbl_merge_pfs_med_pik3ca_status_any_pm,
    tbl_merge_pfs_med_apc_status_any_pm,
    tbl_merge_pfs_med_tp53_status_any_pm
  )
)

pfs_status_by_pm
```

# PFS: Cox modeling

*The multivariable models below estimates the associations between covariates and PFS from initiation of 1L regimen.*

### Any PM vs. no PM 

```{r}
cox.zph(coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + any_peritoneal_met,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
))

tbl_n_pfs_any_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_pfs_event_any_pm <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_and_m_g_status == 1) %>%
  select(any_peritoneal_met) %>%
  tbl_summary(
    label = list(any_peritoneal_met = "Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_pfs_any_pm_cox <- coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + any_peritoneal_met,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(
      reg_start_cpt_rep_mos = "Months from reg start to CPT report",
      any_peritoneal_met = "Presence of PM"
    )
  ) %>%
  add_global_p() %>%
  bold_p()

tbl_univ_pfs_any_pm <-
  tbl_merge(list(
    tbl_n_pfs_any_pm,
    tbl_n_pfs_event_any_pm,
    tbl_univ_pfs_any_pm_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_pfs_any_pm
```

### PM only vs. PM and other mets vs. no PM 

```{r}
cox.zph(coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + pm_and_other_mets,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
))

tbl_n_pfs_pm_others <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_pfs_event_pm_others <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_and_m_g_status == 1) %>%
  select(pm_and_other_mets) %>%
  tbl_summary(
    label = list(pm_and_other_mets = "Presence of PM"),
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_pfs_pm_and_other_mets_cox <- coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + pm_and_other_mets,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(
      reg_start_cpt_rep_mos = "Months from reg start to CPT report",
      pm_and_other_mets = "Presence of PM"
    )
  ) %>%
  add_global_p() %>%
  bold_p()

tbl_univ_pfs_pm_and_other_mets <-
  tbl_merge(list(
    tbl_n_pfs_pm_others,
    tbl_n_pfs_event_pm_others,
    tbl_univ_pfs_pm_and_other_mets_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_pfs_pm_and_other_mets
```

### Stage IV vs. stage I-III with distant mets

```{r}
cox.zph(coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + pfs_cohort,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
))

tbl_n_pfs_stage <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(pfs_cohort) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_pfs_event_stage <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_and_m_g_status == 1) %>%
  select(pfs_cohort) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_pfs_stage_cox <- coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + pfs_cohort,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(
      reg_start_cpt_rep_mos = "Months from reg start to CPT report"
    )
  ) %>%
  add_global_p() %>%
  bold_p()

tbl_univ_pfs_stage <-
  tbl_merge(list(
    tbl_n_pfs_stage,
    tbl_n_pfs_event_stage,
    tbl_univ_pfs_stage_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_pfs_stage
```

### Tumor location

```{r}
cox.zph(coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + crc_type_derived,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
))

tbl_n_pfs_tumor_location <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(crc_type_derived) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_pfs_event_tumor_location <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_and_m_g_status == 1) %>%
  select(crc_type_derived) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_pfs_tumor_location_cox <- coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + crc_type_derived,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(
      reg_start_cpt_rep_mos = "Months from reg start to CPT report"
    )
  ) %>%
  add_global_p() %>%
  bold_p()

tbl_univ_pfs_tumor_location <-
  tbl_merge(list(
    tbl_n_pfs_tumor_location,
    tbl_n_pfs_event_tumor_location,
    tbl_univ_pfs_tumor_location_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_pfs_tumor_location
```

### Tumor side

```{r}
cox.zph(coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + crc_side_derived,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
))

tbl_n_pfs_tumor_side <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(crc_side_derived) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_pfs_event_tumor_side <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_and_m_g_status == 1) %>%
  select(crc_side_derived) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_pfs_tumor_side_cox <- coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + crc_side_derived,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(
      reg_start_cpt_rep_mos = "Months from reg start to CPT report"
    )
  ) %>%
  add_global_p() %>%
  bold_p()

tbl_univ_pfs_tumor_side <-
  tbl_merge(list(
    tbl_n_pfs_tumor_side,
    tbl_n_pfs_event_tumor_side,
    tbl_univ_pfs_tumor_side_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_pfs_tumor_side
```

### MSI status

```{r}
cox.zph(
  coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ reg_start_cpt_rep_mos + msi_status_derived_unk,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
      mutate(msi_status_derived_unk = relevel(factor(
        case_when(
          msi_status_derived == "Non-concordant" ~ as.character(NA),
          TRUE ~ msi_status_derived
        )
      ), ref = "MSI-L/MSS")),
    timefix = FALSE
  )
)

tbl_n_pfs_msi <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  mutate(msi_status_derived_unk = relevel(factor(
    case_when(
      msi_status_derived == "Non-concordant" ~ as.character(NA),
      TRUE ~ msi_status_derived
    )
  ), ref = "MSI-L/MSS")) %>%
  select(msi_status_derived_unk) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}"),
    label = list(msi_status_derived = "MSI status")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_pfs_event_msi <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1) %>%
  mutate(msi_status_derived_unk = relevel(factor(
    case_when(
      msi_status_derived == "Non-concordant" ~ as.character(NA),
      TRUE ~ msi_status_derived
    )
  ), ref = "MSI-L/MSS")) %>%
  select(msi_status_derived_unk) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}"),
    label = list(msi_status_derived = "MSI status")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_pfs_msi_cox <- coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + msi_status_derived_unk,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
    mutate(msi_status_derived_unk = relevel(factor(
      case_when(
        msi_status_derived == "Non-concordant" ~ as.character(NA),
        TRUE ~ msi_status_derived
      )
    ), ref = "MSI-L/MSS")),
  timefix = FALSE
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(
      reg_start_cpt_rep_mos = "Months from reg start to CPT report"
    )
  ) %>%
  add_global_p() %>%
  bold_p()

tbl_univ_pfs_msi <-
  tbl_merge(list(tbl_n_pfs_msi,
                 tbl_n_pfs_event_msi,
                 tbl_univ_pfs_msi_cox)) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_pfs_msi
```

### MMR status

```{r}
cox.zph(
  coxph(
    Surv(
      time = reg_start_cpt_rep_mos,
      time2 = tt_pfs_i_and_m_g_mos,
      event = pfs_i_and_m_g_status
    ) ~ reg_start_cpt_rep_mos + mmr_status_derived_factor,
    data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
      filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
      mutate(mmr_status_derived_factor = relevel(factor(case_when(
        mmr_status_derived == "Non-concordant" ~ as.character(NA),
        TRUE ~ mmr_status_derived
      )), ref = "pMMR")),
    timefix = FALSE
  )
)

tbl_n_pfs_mmr <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  mutate(mmr_status_derived_factor = relevel(factor(
    case_when(
      mmr_status_derived == "Non-concordant" ~ as.character(NA),
      TRUE ~ mmr_status_derived
    )
  ), ref = "pMMR")) %>%
  select(mmr_status_derived_factor) %>%
  tbl_summary(statistic = list(everything() ~ "{n}")) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_pfs_event_mmr <-
  pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos,
         pfs_i_and_m_g_status == 1) %>%
  mutate(mmr_status_derived_factor = relevel(factor(
    case_when(
      mmr_status_derived == "Non-concordant" ~ as.character(NA),
      TRUE ~ mmr_status_derived
    )
  ), ref = "pMMR")) %>%
  select(mmr_status_derived_factor) %>%
  tbl_summary(statistic = list(everything() ~ "{n}")) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_pfs_mmr_cox <- coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + mmr_status_derived_factor,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
    mutate(mmr_status_derived_factor = relevel(factor(
      case_when(
        mmr_status_derived == "Non-concordant" ~ as.character(NA),
        TRUE ~ mmr_status_derived
      )
    ), ref = "pMMR")),
  timefix = FALSE
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(reg_start_cpt_rep_mos = "Months from reg start to CPT report")
  ) %>%
  add_global_p() %>%
  bold_p()

tbl_univ_pfs_mmr <-
  tbl_merge(list(tbl_n_pfs_mmr,
                 tbl_n_pfs_event_mmr,
                 tbl_univ_pfs_mmr_cox)) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_pfs_mmr
```

### Presence of liver mets at diagnosis

```{r}
cox.zph(coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + any_liver_mets_at_dx,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
))

tbl_n_pfs_liver_mets <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_liver_mets_at_dx) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_pfs_event_liver_mets <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_and_m_g_status == 1) %>%
  select(any_liver_mets_at_dx) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_pfs_liver_mets_cox <- coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + any_liver_mets_at_dx,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(
      reg_start_cpt_rep_mos = "Months from reg start to CPT report"
    )
  ) %>%
  add_global_p() %>%
  bold_p()

tbl_univ_pfs_liver_mets <-
  tbl_merge(list(
    tbl_n_pfs_liver_mets,
    tbl_n_pfs_event_liver_mets,
    tbl_univ_pfs_liver_mets_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_pfs_liver_mets
```

### Presence of lung mets at diagnosis

```{r}
cox.zph(coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + any_lung_mets_at_dx,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
))

tbl_n_pfs_lung_mets <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(any_lung_mets_at_dx) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_pfs_event_lung_mets <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_and_m_g_status == 1) %>%
  select(any_lung_mets_at_dx) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_pfs_lung_mets_cox <- coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + any_lung_mets_at_dx,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(
      reg_start_cpt_rep_mos = "Months from reg start to CPT report"
    )
  ) %>%
  add_global_p() %>%
  bold_p()

tbl_univ_pfs_lung_mets <-
  tbl_merge(list(
    tbl_n_pfs_lung_mets,
    tbl_n_pfs_event_lung_mets,
    tbl_univ_pfs_lung_mets_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_pfs_lung_mets
```

### RAS/BRAF status

```{r}
cox.zph(coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + first_cpt_kras_braf_nras,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
))

tbl_n_pfs_ras_braf <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(first_cpt_kras_braf_nras) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_pfs_event_ras_braf <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_and_m_g_status == 1) %>%
  select(first_cpt_kras_braf_nras) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_pfs_ras_braf_cox <- coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + first_cpt_kras_braf_nras,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(
      reg_start_cpt_rep_mos = "Months from reg start to CPT report"
    )
  ) %>%
  add_global_p() %>%
  bold_p()

tbl_univ_pfs_ras_braf <-  
  tbl_merge(list(
    tbl_n_pfs_ras_braf,
    tbl_n_pfs_event_ras_braf,
    tbl_univ_pfs_ras_braf_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_pfs_ras_braf
```

### TP53

```{r}
cox.zph(coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + first_cpt_tp53,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
))

tbl_n_pfs_tp53 <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(first_cpt_tp53) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_pfs_event_tp53 <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_and_m_g_status == 1) %>%
  select(first_cpt_tp53) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_pfs_tp53_cox <- coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + first_cpt_tp53,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(
      reg_start_cpt_rep_mos = "Months from reg start to CPT report"
    )
  ) %>%
  add_global_p() %>%
  bold_p()

tbl_univ_pfs_tp53 <- 
  tbl_merge(list(
    tbl_n_pfs_tp53,
    tbl_n_pfs_event_tp53,
    tbl_univ_pfs_tp53_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_pfs_tp53
```

### APC

```{r}
cox.zph(coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + first_cpt_apc,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
))

tbl_n_pfs_apc <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(first_cpt_apc) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_pfs_event_apc <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_and_m_g_status == 1) %>%
  select(first_cpt_apc) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_pfs_apc_cox <- coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + first_cpt_apc,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(
      reg_start_cpt_rep_mos = "Months from reg start to CPT report"
    )
  ) %>%
  add_global_p() %>%
  bold_p()

tbl_univ_pfs_apc <-
  tbl_merge(list(
    tbl_n_pfs_apc,
    tbl_n_pfs_event_apc,
    tbl_univ_pfs_apc_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_pfs_apc
```

### PIK3CA

```{r}
cox.zph(coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + first_cpt_pik3ca,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
))

tbl_n_pfs_pik3ca <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos) %>%
  select(first_cpt_pik3ca) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_n_pfs_event_pik3ca <- pfs_surv_cohort_first_line_folfoxiri_cohort %>%
  filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos, 
         pfs_i_and_m_g_status == 1) %>%
  select(first_cpt_pik3ca) %>%
  tbl_summary(
    statistic = list(everything() ~ "{n}")
  ) %>%
  modify_header(update = list(stat_0 ~ "**N**")) %>%
  modify_footnote(update = everything() ~ NA)

tbl_univ_pfs_pik3ca_cox <- coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + first_cpt_pik3ca,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(
      reg_start_cpt_rep_mos = "Months from reg start to CPT report"
    )
  ) %>%
  add_global_p() %>%
  bold_p()

tbl_univ_pfs_pik3ca <- 
  tbl_merge(list(
    tbl_n_pfs_pik3ca,
    tbl_n_pfs_event_pik3ca,
    tbl_univ_pfs_pik3ca_cox
  )) %>%
  modify_spanning_header(everything() ~ NA_character_)

tbl_univ_pfs_pik3ca
```

```{r, include = FALSE}
tbl_univ_pfs_any_pm$table_body <- tbl_univ_pfs_any_pm$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "reg_start_cpt_rep_mos")

tbl_univ_pfs_pm_and_other_mets$table_body <- tbl_univ_pfs_pm_and_other_mets$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "reg_start_cpt_rep_mos")

tbl_univ_pfs_stage$table_body <- tbl_univ_pfs_stage$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "reg_start_cpt_rep_mos")

tbl_univ_pfs_tumor_location$table_body <- tbl_univ_pfs_tumor_location$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "reg_start_cpt_rep_mos")

tbl_univ_pfs_tumor_side$table_body <- tbl_univ_pfs_tumor_side$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "reg_start_cpt_rep_mos")

tbl_univ_pfs_msi$table_body <- tbl_univ_pfs_msi$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "reg_start_cpt_rep_mos")

tbl_univ_pfs_liver_mets$table_body <- tbl_univ_pfs_liver_mets$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "reg_start_cpt_rep_mos")

tbl_univ_pfs_lung_mets$table_body <- tbl_univ_pfs_lung_mets$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "reg_start_cpt_rep_mos")

tbl_univ_pfs_ras_braf$table_body <- tbl_univ_pfs_ras_braf$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "reg_start_cpt_rep_mos")

tbl_univ_pfs_tp53$table_body <- tbl_univ_pfs_tp53$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "reg_start_cpt_rep_mos")

tbl_univ_pfs_apc$table_body <- tbl_univ_pfs_apc$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "reg_start_cpt_rep_mos")

tbl_univ_pfs_pik3ca$table_body <- tbl_univ_pfs_pik3ca$table_body %>% 
  as.data.frame() %>% 
  filter(variable != "reg_start_cpt_rep_mos")

tbl_univ_pfs_cox <- tbl_stack(
  list(
    tbl_univ_pfs_any_pm,
    tbl_univ_pfs_pm_and_other_mets,
    tbl_univ_pfs_stage,
    tbl_univ_pfs_tumor_location,
    tbl_univ_pfs_tumor_side,
    tbl_univ_pfs_msi,
    tbl_univ_pfs_liver_mets,
    tbl_univ_pfs_lung_mets,
    tbl_univ_pfs_ras_braf,
    tbl_univ_pfs_tp53,
    tbl_univ_pfs_apc,
    tbl_univ_pfs_pik3ca
  )
)

tbl_univ_pfs_cox
```

### Multiple covariates

```{r}
cox.zph(coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + any_liver_mets_at_dx + first_cpt_apc,
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
))

tbl_multi_pfs_cox <- coxph(
  Surv(
    time = reg_start_cpt_rep_mos,
    time2 = tt_pfs_i_and_m_g_mos,
    event = pfs_i_and_m_g_status
  ) ~ reg_start_cpt_rep_mos + factor(any_liver_mets_at_dx) + factor(first_cpt_apc),
  data = pfs_surv_cohort_first_line_folfoxiri_cohort %>%
    filter(tt_pfs_i_and_m_g_mos > reg_start_cpt_rep_mos),
  timefix = FALSE
) %>%
  tbl_regression(
    exponentiate = TRUE,
    label = list(reg_start_cpt_rep_mos = "Months from advanced dx to CPT report")
  ) %>%
  add_nevent(location = "level") %>% 
  add_n(location = "level") %>% 
  add_global_p() %>% 
  bold_p()

tbl_multi_pfs_cox
```
